---
title: "Language and cognitive phenotypes in children with sex chromosome trisomies"
author: "Dorothy V M Bishop, Annie Brookman-Byrne,  Nicola Gratton, Elaine Gray, Georgina Holt, Louise Morgan, Sarah Morris, Eleanor Paine, Holly Thornton, and Paul A. Thompson"
date: 
output:
  word_document: default
  html_document: default
---

`r Sys.time()`


```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='Figs/',
                      echo=FALSE, include=FALSE, warning=FALSE, message=FALSE)
```

```{r load_packages, include = FALSE, echo=FALSE}
#################################################################################

#devtools::install_github("rubenarslan/formr")
require(formr) #to handle conflicts with 'select' function in dplyr (may not be needed)
require(here) #for reproducible pathnames: see https://www.tidyverse.org/articles/2017/12/workflow-vs-script/

library(tidyverse)
library(reshape2)  #for melt
library(knitr) #for kable - allows nice tables in Word output - but can be temperamental
library(gridExtra)
library(grid)
library(Hmisc)
library(doBy)
library(lme4) #linear model
library(lmerTest) #Adding this extra package provides p-values (if you want them) for your fixed effects.
library(ggridges) #for joyplot
library(car)
require(psych)
library(beeswarm) #particular plot type
library(beanplot) #ditto
library(yarrr) #for pirate plot - another plot type
library(lavaan) #for SEM to extract language factor
library(semPlot) #used with Lavaan
library(standardize)
library(ggpubr)
library(reshape2)
library(ggcorrplot)
library(DiagrammeR)
library(DiagrammeRsvg)
library(magrittr)
library(svglite)
library(rsvg)
library(png)
library(stringr)
library(car)
library(kableExtra)
library(flextable)
library(officer)
library(stargazer)
library(mvnormtest) #check assumptions for MANOVA
testbattery<-read.csv('testbattery.csv') #used for Table 1 in word doc

#################################################################################

```

```{r explan,include=FALSE}
#Need to create a main analysis file in which we have a code that corresponds to group.
#Have col for trisomy which corresponds to XXX, XXY, XYY, and NA: NA will be twins
#Makes sense to use randomtwininclude to select just one from each pair
#For trisomies, have a col that specifies whether highbias or lowbias

#Have a col that determines if meeting criteria for DLD - regardless of whether twin or not.
#This can be same criterion as for doppler paper : this was:
#<i>Results from the language/cognitive test battery were used to categorise children as having DLD if they scored more than 1 SD below population norms on two or more out of 13 language/literacy measures, and as TD if they scored below this threshold on no more than one measure. In previous studies, we have excluded children who met this criterion solely on literacy measures (TOWRE and NARA-II); 
#Children were excluded from the sample if they met any of the following criteria: WASI nonverbal ability (performance IQ) more than two SDs below the population mean; diagnosis of autism spectrum disorder (ASD) in one or both twins; sensorineural hearing loss or failure of a hearing test on the day of testing; and brain injury or a serious medical condition affecting one or both twins.</i>

#nb index of multiple deprivation from postcode (English)
#http://imd-by-postcode.opendatacommunities.org/

#Also see re: CCC-2 nonresponse rates - https://www.aapor.org/About-Us/History/Presidential-Addresses/2004-Presidential-Address.aspx
```


```{r readredcap, include=FALSE}
#########################################################################################
## Step 1
#Read in latest SCT from redcap.
#Do the same for the twins.
#Select variables to be used.

#redcap.dir <-"c:/Users/pthompson/Dropbox/project SCT analysis/Data from Redcap/"
redcap.dir <-"~/Dropbox/ERCadvanced/project SCT analysis/Data from Redcap/"

#Use latest version - now includes milestones
sct.redcap <- 'SCTDATA_DATA_2018-08-13_1234.csv'
#updated to include code indicating whether participated in prior study
twin.redcap <- 'TwinsData_DATA_2018-08-19_1139.csv' #updated to include ccc2_consistent variable
#also now has pheno_include, which selects individual twins for inclusion
sct.data <- read.csv(paste0(redcap.dir,sct.redcap),stringsAsFactors = FALSE)
twin.data <- read.csv(paste0(redcap.dir,twin.redcap),stringsAsFactors = FALSE)
names(sct.data)[1]<-"record_id"
names(twin.data)[1]<-"record_id"
#fix any columns with discrepant names
w<-which(colnames(twin.data)=='neurodev_diagnosis')
colnames(twin.data)[w] <-'neurodev_diag'
# Coded from parental interview only: 0, none, 1, ADHD
# 2, APD, # 3, ASD, # 4, behav, # 5, dyscalc, # 6, dyslexia
# 7, dyspraxia, # 8, SLI/DLD, # 9, ID

#Nb Fam position: Coded so if you subtract 1 you get N older sibs: disregards birth order of twins
w<-which(colnames(sct.data)=='position_in_family')
colnames(sct.data)[w] <-'fam_position'

#NB this dld_rd categorisation not currently used
twin.data$dyslexia<-twin.data$dld_rd%%10 #modulus 10 , ie last digit
twin.data$lang_disorder<-round((twin.data$dld_rd-4.5)/10) #first digit is dld code

#add gender info to sct data so it merges properly
sct.data$female<-0
w<-which(sct.data$trisomy==1)
sct.data$female[w]<-1 #XXX cases are female

sct.data$fam_id<-NA #dummy column to ensure merges ok
#########################################################################################
#----------------------------------------------------------------------------------
```


```{r colselect, include=FALSE}
#########################################################################################


#Specify columns to retain that are common to both sct and twin
#NB we are using oromotor_ss2, which is renormed against the TD twin group
keepcols <- c('record_id','fam_id','female','age_at_test','partial_testing','pass_hearing','wasi_matrices_ss','wasi_block_design_ss','wasi_vocab_ss','wdck_jhsn_ss',
              'sent_rep_ss','nonword_rep_ss','oromotor_ss_2','nara_acc_ss',      'nara_comp_ss','nara_rate_ss','towre_words_ss','towre_nonwords_ss','phab_pic_ss','phab_digit_ss','srs_t_score','slt','splang_conc','schooling','piq','neurodev_diag','dyslexia','lang_disorder',
              'hyperkinetic_icd_r1','adhd_comb_dsm_r1','adhd_hyp_dsm_r1','adhd_inatt_dsm_r1',
              'conduct_dsm_r1','asd_dsm_r1','autism_icd_r1','srs_t_score',
              'gcc','scdi','ccc_a','ccc_b','ccc_c','ccc_d','ccc_e','ccc_f',
              'ccc_g','ccc_h','ccc_i','ccc_j','ccc2_consistent','mo_educ','fa_educ','deprivation_index','fam_structure','fam_position','wordcode','sentcode','walkcode','drycode',
              'nepsy_oromotor_seq_raw')

#NB asd_dsm_r1 does not always agree with parental report. Use either to exclude. 
#Only definite diagnosis (1) on DSM is used (4 is PDDNOS)

twin.short <- select(twin.data,keepcols)
sct.short <- select(sct.data,keepcols) #NB 'select' is ambiguous between packages but should work ok if we use formr (otherwise need dplyr::select)
#########################################################################################

```

```{r wrangledata, include=FALSE}
#----------------------------------------------------------------------------------
#OROMOTOR SCORES!
#NB see SCT_twin_langpheno_orig.Rmd for script re creation of oromotor_ss_2.
#Basically, we took the TD twins as norms, and used regression of raw score on Age to create residualised scaled scores

#myintercept<-summary(mymod)$coefficients[1,1] gave value of 35.754
#myb<-summary(mymod)$coefficients[2,1] gave value of 0.07612
#myse<-summary(mymod)$sigma gave value of 9.1667
#Then formula is:

#z.oro<-100+15*(langcog$Oromotor_raw-(myintercept+langcog$Age*myb))/myse

#The original script was tweaked on 19/8/18 to add values of oromotor_ss_2 for the 2nd twin.
###################################################################################################
#Add columns that are specific to twins or SCTs so all aligned
#Select data for short file


sct.short$randomtwininclude <- 0 #this specifies child is not a twin but a trisomy
sct.short$zygosity <- NA
twin.short$randomtwininclude <- twin.data$randomtwininclude+1 #twins subdivided
#randomly into 1 and 2 for replication sample
twin.short$zygosity <- twin.data$zygosity
#2 ASD, 3 ASDsib, 4 fail hearing screen (35dB plus mean for better ear), 5 other (rare condition)
#The previous include variable was used for Doppler paper and had more stringent hearing criteria.

sct.short$trisomy <- sct.data$trisomy
sct.short$pre_postnatal_diag <- sct.data$pre_postnatal_diag
sct.short$why_tested <-sct.data$why_tested
sct.short$orig_famcode<-0
sct.short$orig_famcode[sct.data$orig_famcode>0]<-1 #if seen before this is 1
twin.short$trisomy <- NA
twin.short$pre_postnatal_diag <- NA
twin.short$why_tested <- NA
twin.short$orig_famcode<-NA
twin.short$splang_conc<-twin.data$splang_conc
sct.short$splang_conc<-NA
sct.short$pheno_include<-1
twin.short$pheno_include<-twin.data$pheno_include #this codes reason for exclusion as:
#2 ASD, 3 ASD-sib, 4-hearing, 5- other
#----------------------------------------------------------------------------------
#Bolt SCT and twin files together
all.short <-rbind(sct.short,twin.short)

#remove case of isochromosome
w<-which(all.short$trisomy==9)
all.short<-all.short[-w,]

all.short<-all.short[all.short$pheno_include==1,] #removes those with ASD, ASD-sib,hearing, other

#rename age variable
w<-which(colnames(all.short)=='age_at_test')
colnames(all.short)[w]<-'Age'

#add a column identifying age range: 1 = <6, 2 = 6-11;11, 3 = 12+
all.short$ageband<-2
w<-which(all.short$Age<72)
all.short$ageband[w]<-1

w<-which(all.short$Age>143)
all.short$ageband[w]<-3

#exclude cases with DSM ASD diagnosis
w<-which(all.short$asd_dsm_r1==1)
all.short$pheno_include[w]<-2

#########################################################################################
```

```{r missingcheck, include=FALSE}
##Step 2
#Deal with missing data
#Codes are

#999 – test not given
#998 – result not valid (maladministration)
#997 – result not valid (child refusal etc.)
#996 – child too old/young for norms

#Check for nonword rep - in some cases OK to substitute low score, if child unable to attempt task. 

all.short$pass_hearing[all.short$pass_hearing==9]<-NA #change 9 to NA for pass_hearing
#NB these include both refusal and equipment failure

#missing data has codes of 996-999 for language tests
w <- which(all.short$nonword_rep_ss>900)
print('Cases with missing nonword rep data: ')
all.short$record_id[w]
#These cases all checked : all SCT cases who either were not tested because v limited spoken language, or who refused spoken tests or had low scores on other language tests. All these cases assigned a scaled score of 3.
all.short$nonword_rep_ss[w] <- 3

#999 is code where child was not tested bcs too low-functioning
w <- which(all.short$wasi_vocab_ss==999)
print('Cases with missing Vocab data (999) reassigned to floor: ')
all.short$record_id[w]
#These assigned SS of 2.33 SD below mean (equivalent to 3 on NEPSY scale)
all.short$wasi_vocab_ss[w] <- 25

w <- which(all.short$wdck_jhsn_ss==999)
print('Cases with missing Woodcock_J data (999) reassigned to floor: ')
all.short$record_id[w]
#These assigned SS of 2.33 SD below mean (equivalent to 3 on NEPSY scale)
all.short$wdck_jhsn_ss[w] <- 55

w <- which(all.short$sent_rep_ss==999)
print('Cases with missing Sent rep data (999) reassigned to floor:: ')
all.short$record_id[w]
#These assigned SS of 2.33 SD below mean (equivalent to 3 on NEPSY scale)
all.short$sent_rep_ss[w] <- 3

w <- which(all.short$oromotor_ss_2==999)
print('Cases with missing oromotor data (999) reassigned to floor:: ')
all.short$record_id[w]
#These assigned SS of 55 , as this corresponds to floor in this sample
all.short$oromotor_ss_2[w] <- 55

#Now count how many tests have missing data
mycols<-colnames(all.short)
#Identify the range of columns where 900+ is missing data code
mc <-which(mycols=='wasi_matrices_ss')
mc0<-which(mycols=='nonword_rep_ss') #end of range for spoken language
mc1<-which(mycols=='phab_digit_ss') #end of range of reading related
mc2<-which(mycols=='ccc_a') #end of range of reading related
mc3<-which(mycols=='ccc_j') #end of range of reading related

all.short$nmissinglang<-0
for(i in 1:nrow(all.short)){
  mytests<-all.short[i,mc:mc0]
 nna<- length(which(is.na(mytests)))
 n9<- length(which(mytests>900))
  all.short$nmissinglang[i]<-(nna+n9)
  
}
table(all.short$nmissinglang,all.short$ageband)

all.short$nmissingread<-0
for(i in 1:nrow(all.short)){
  mytests<-all.short[i,(mc0+1):mc1]
   nna<- length(which(is.na(mytests)))
 n9<- length(which(mytests>900))
  all.short$nmissingread[i]<-(nna+n9)
  
}
table(all.short$nmissingread,all.short$ageband)


#Put in NA for missing data codes all lang/read/CCC-2
for (i in c(mc:mc1,mc2:mc3)){
  w <- which(all.short[,i]>900)
  all.short[w,i]<-NA
  
  ncol<-length(colnames(all.short)) #define ncol here
  
}

#Parental educ variables, missing is 9
w<-which(all.short$mo_educ==9)
all.short$mo_educ[w]<-NA
w<-which(all.short$fa_educ==9)
all.short$fa_educ[w]<-NA


```

```{r rescalelang, include=FALSE}
#Rescale all variables to mean 100 and SD 15 and save in new columns
shortlangnames<-c('Matrices','Blocks','Vocab','Comprehnsn','SentRep','NwdRep','Oromotor',
                  'ReadAcc','ReadComp','ReadRate','TOWREwd','TOWREnwd',
                  'PicName','DigitName')

#convert all variables to same scale: mean 100 and SD 15
#This has to be hard coded because different tests on different scales
langcogcols<-c("wasi_matrices_ss" ,"wasi_block_design_ss", "wasi_vocab_ss" ,
               "wdck_jhsn_ss" , "sent_rep_ss" , "nonword_rep_ss",  "oromotor_ss_2",
               "nara_acc_ss" ,"nara_comp_ss","nara_rate_ss" ,"towre_words_ss",
               "towre_nonwords_ss","phab_pic_ss","phab_digit_ss")
#Note we now are using the oromotor_ss_2 variable: normed against TD twins

langcols<-which(colnames(all.short)%in%langcogcols)
#NB this finds columns corresponding to langcogcols, but they will be in the order they occur in file
#start by just duplicating original scaled scores
nnew<-length(shortlangnames)
nurange<-(ncol+1):(ncol+nnew) #column range for ss values, NB ncol defined in prev block
all.short[,nurange]<-all.short[,langcols]
colnames(all.short)[nurange]<-shortlangnames
#First 3 are normed with mean 50 and SD 10
for (i in 1:3){
  oldmean<-50
  oldsd<-10
  all.short[,(i+ncol)]<- 100+15*(all.short[,langcols[i]]-oldmean)/oldsd
}

for (i in c(5,6)){
  oldmean<-10
  oldsd<-3
  all.short[,(i+ncol)]<- 100+15*(all.short[,langcols[i]]-oldmean)/oldsd
}

```

```{r interpolatemissing, include=FALSE}
#If just one value is missing in lang or reading groups, substitute the mean for others
langrange<-(ncol+1):(ncol+7)
readrange<-(ncol+8):(ncol+13)
#Start with language range
w<-which(all.short$nmissinglang==1) #find those rows with just one missing value
for(s in 1:length(w)){
  thisrow<-w[s] #row with missing value
  v<-which(is.na(all.short[thisrow,langrange])) #which value is missing?
  mycol<-langrange[v] #column with missing data
  all.short[thisrow,langrange[v]]<-round(rowMeans(all.short[thisrow,langrange],na.rm=TRUE),0) #mean of other values substituted
}

print(paste0('Mean substituted for cases with one missing datapoint: Interpolated ', length(w),' missing values from ',nrow(all.short)*length(langrange),
             ' datapoints in ',length(langrange),' nonverbal and language tests'))

w<-which(all.short$nmissingread==1) #find those rows with just one missing value
for(s in 1:length(w)){
  thisrow<-w[s] #row with missing value
  v<-which(is.na(all.short[thisrow,readrange])) #which value is missing?
  mycol<-readrange[v]
  all.short[thisrow,readrange[v]]<-round(rowMeans(all.short[thisrow,readrange],na.rm=TRUE),0) #mean of other values substituted
}
print(paste0('Mean substituted for cases with one missing datapoint: Interpolated ', length(w),' missing values from ',nrow(all.short)*length(readrange),
             ' datapoints in ',length(readrange),' reading and rapidnaming tests'))




```


```{r ncheck, include=FALSE}
## Step 3

# #Check that Ns match those in the protocol
# Should have three groups with N = 142 SCT, 194 twin1 and 194 twin2 (these are randomly assigned twin 1 and 2), - but Ns less for twins because of exclusions
# Numbers are greater than for Newbury et al, as some children did not feature in that study because of lack of DNA.
# 
# Twins recruited in relation to splang_conc, which is coded as
# 0, never
# 1, preschool only
# 2, continuing, mild
# 3, continuing, severe
# 4, reading concerns only
# 9, unclear
# We will be using this categorisation to subdivide twins; categories 2 and 3 distinguished.
# 
# Trisomies - divided according to reason for diagnosis
# why tested used to make bias column: coded as: 
# 0, maternal age
# 1, medical concerns
# 2, behavioural concerns
# 3, neurodevelopmental concerns
# 4, family history of genetic problems
# 9, no information
###################################################################################
twintab <- table(all.short$randomtwininclude)
names(twintab)<-c('SCT','twin1','twin2')

all.short$bias<-'NA'
w<-which(all.short$randomtwininclude==0)
all.short$bias[w]<-0
w<-c(which(all.short$why_tested==2),which(all.short$why_tested==3))
all.short$bias[w]<-1
scttab <- table(all.short$trisomy,all.short$bias)
rownames(scttab)<-c('XXX','XXY','XYY')
colnames(scttab)<-c('low bias','high bias','NA')
#Show tables
twintab
tab_sct<-scttab
scttab
rowSums(scttab)

all.short$parentconc<-NA #parental concern re language
w<-which(all.short$randomtwininclude>0)
all.short$parentconc[w]<-0
w<-which(all.short$splang_conc%in%c(2,3))
all.short$parentconc[w]<-1

parcontab<-table(all.short$parentconc,all.short$randomtwininclude)
parcontab<-parcontab[,2:3]
rownames(parcontab)<-c('No concern','Concern')
colnames(parcontab)<-c('twin1','twin2')
parcontab
print('Age range (min, max, average) for SCT')
min(sct.data$age_at_test)
max(sct.data$age_at_test)
mean(sct.data$age_at_test)
print('Age range (min, max, average) for twin')
min(twin.data$age_at_test)
max(twin.data$age_at_test)
mean(twin.data$age_at_test)

#count how many SCT cases were in the earlier 2009 study (Bishop et al 2011 paper)
print('')
print('In 2009 study')
w<-which(all.short$orig_famcode==1)
origstudy<-table(all.short$bias[w],all.short$trisomy[w])
colnames(origstudy)<-c('XXX','XXY','XYY')
rownames(origstudy)<-c('Low bias','High bias')
origstudy

#Count how many cases available for MANOVAs
#Must have full data on lang or reading (5 yr olds excluded on this basis)
w<-which(all.short$nmissinglang<2) #we have retained those with one missing value
tempshort<-all.short[w,]
scttab1 <- table(tempshort$trisomy,tempshort$bias)
rownames(scttab1)<-c('XXX','XXY','XYY')
colnames(scttab1)<-c('low bias','high bias','NA')
#Show tables
print('Cases with full data for nv/lang analysis')

scttab1
rowSums(scttab1)

w<-which(all.short$nmissingread<2) #we have retained those with one missing value
tempshort<-all.short[w,]
scttab2 <- table(tempshort$trisomy,tempshort$bias)
rownames(scttab2)<-c('XXX','XXY','XYY')
colnames(scttab2)<-c('low bias','high bias','NA')
#Show tables
print('Cases with full data for reading/naming analysis')

scttab2
rowSums(scttab2)
###################################################################################
```



```{r groupcreate, include=FALSE}
##Create 8 groups:
# 1. XXX-nobias
# 2. XXY-nobias
# 3. XYY-nobias
# 4. XXX-hibias
# 5. XXY-hibias
# 6. XYY-hibias
# 7. twin_noconcern
# 8. twin_concern
# NB as currently defined, not taking into account co-twin status. So a twin_noconcern could have cotwin with concern, ie have family risk.

#####################################################################################
#first code SLT into 0, 1 or 2;  2 indicates SALT beyond 4 yr, 1 is assessment or preschool only
all.short$sltcode<-0
w<-c(which(all.short$slt==1),which(all.short$slt==8))
all.short$sltcode[w]<-1
w<-c(which(all.short$slt==2),which(all.short$slt==3))
all.short$sltcode[w]<-2


all.short<-filter(all.short,randomtwininclude<2) #remove twin2
all.short$group8<-all.short$trisomy
w<-which(all.short$bias==1)
all.short$group8[w]<-all.short$group8[w]+3 #convert 1 2 3 to 4 5 6
w<-which(all.short$randomtwininclude>0) #just focus on twins
all.short$group8[w]<-7
w<-which(all.short$parentconc==1) #coded as NA for SCTs, so only twins coded here
all.short$group8[w]<-8

#add twins with SLT code 2 to group 8, even if no parental concern
w<-intersect(which(all.short$group8==7),which(all.short$sltcode==2))
all.short$group8[w]<-8

#Tried removing those with affected cotwin, but it has little effect other than to reduce N.
#This is therefore not done at present, though code retained.
remove.by.cotwin<-0
if(remove.by.cotwin==1){
#now remove from group 7 those who have an affected cotwin: either SLT or concerns
cotwins<-filter(twin.short,randomtwininclude==2)
f<-which(all.short$group8==7)
cotwinremove<-vector()
for (i in 1:length(f)){
  thisrow<-f[i]
  thisfam<-all.short$fam_id[thisrow]
  g<-which(cotwins$fam_id==thisfam)
  thisslt<-cotwins$slt[g]
  thislangconcern<-cotwins$splang_conc[g]
  if(thisslt==2 || thisslt==3 || thislangconcern==2 ||thislangconcern==3)
  {cotwinremove<-c(cotwinremove,thisrow)}
}
all.short<-all.short[-cotwinremove,]
}

all.short$group8level<-all.short$group8#useful to retain numeric code before converting to factor
all.short$group8<-as.factor(all.short$group8)
levels(all.short$group8)<-c('XXX.lowbias','XXY.lowbias','XYY.lowbias',
                            'XXX.hibias','XXY.hibias','XYY.hibias',
                            'Twin: No Concerns','Twin: Lang concerns')

#Add code for thse with NA data on the language tests
#For MANOVA, only those with complete data are included
all.short$langdatacomplete<-1 #default: updated after interpolating values if only one missing
mycols<-colnames(all.short)
mcx1 <-which(mycols=='Matrices')
mcx2<-which(mycols=='DigitName')
for (i in mcx1:mcx2){
  w <- which(is.na(all.short[,i]))
  all.short$langdatacomplete[w]<-0
  
}

#Now recode all SCTs together in one group for contrast with twins
all.short$trisomyvstwin<-1
all.short$trisomyvstwin[all.short$group8level<7]<-2

```

```{r get.covariates, include=FALSE}
##Covariates
# a)	Educational level of mother and father, transformed into an ordinal scale based on age at leaving full-time education/qualifications obtained, with points of 0 (prior to age 16 years), 1 (16 years/did GCSE or O-levels), 2 (18 years/ did A-levels), 3 (21 years, degree), 4 (postgraduate study). 
# 
# b)	An index of multiple deprivation based on postcode was obtained for those living in England from the website http://imd-by-postcode.opendatacommunities.org/. This uses local statistics from the Department for Communities and Local Government to rank over 32,000 postcodes on the basis of a weighted sum based on income, employment, education, health, crime, housing and living environment. The rank score was converted to a z-score to give a normally-distributed variable.  
# 
# c)	Living with single parent, coded as a binary variable (0 or 1), based on status at the time of the initial interview.
# 
# d)	Number of older siblings. For twins, the co-twin was not counted.
# 
# N.B. The finding that the lowbias SCT children are more likely to have older sibs is probably related to the fact that pre-natal testing is associated with older parents. We do have data on parental ages and could look at that, but not sure it's necessary. It does, though, make me concerned that there may be problems in using N older sibs as a covariate, as it could look causal when it is not. It might be reasonable to use it just within the twin group.

#Parental educational level
par.ed <-aggregate(all.short$mo_educ, by=list(all.short$group8),
                   FUN=mean, na.rm=TRUE)
pirateplot(mo_educ~group8,data=all.short)
pirateplot(fa_educ~group8,data=all.short)

#Deprivation index
#pirateplot(deprivation_index~group8,data=all.short,main='Deprivation index: high is good')

#Can generate deciles, but better to normalise deprviation index as below
#depcut<-3275*c(1,2,3,4,5,6,7,8,9)
# print('Deprivation index decile cutpoints are:')
# print(depcut)
# pirateplot(deprivation_index~mo_educ,data=all.short,main='Deprivation index: high is good')

#Use ranks to convert to normal distribution
#Rw deprivation index is a rank out of 32750
all.short$deprivz<-qnorm(all.short$deprivation_index/32750)
w<-which(is.na(all.short$deprivz))
all.short$deprivz[w]<-0
#Assign zscore of zero to those outside england where no dep index

#NB rows below 143 are trisomies
print(paste('N SCT with missing dep index (assigned 0) = ',length(which(w<143))))
print(paste('N twin with missing dep index (assigned 0) = ',length(which(w>142))))
all.short$deprivz[w]<-0
dopirate<-0 #toggle to 1 to see pirateplot (not really needed here, but useful as eg of plot labeling)
if (dopirate==1){
pirateplot(deprivz~group8,data=all.short,xaxt='n')
text(1,3.2,'XXX',cex=.8)
text(2,3.2,'XXY',cex=.8)
text(3,3.2,'XYY',cex=.8)
text(4,3.2,'XXX',cex=.8)
text(5,3.2,'XXY',cex=.8)
text(6,3.2,'XYY',cex=.8)
text(7,3.4,'Twin\nno concerns',cex=.8)
text(8,3.2,'Twin\nlanguage\nproblems ',cex=.8)
text(2,-3,'<-------Low bias------->')
text(5,-3,'<-------High bias------->')
}
all.short$singlepar<-0
all.short$singlepar[all.short$fam_structure<3]<-1
singlepar.tab<-table(all.short$group8,all.short$singlepar)
colnames(singlepar.tab)<-c('2 parents','single parent')
prop.table(singlepar.tab,1)

print('N older siblings')
all.short$oldersibs<-all.short$fam_position-1
nsib.tab<-table(all.short$group8,all.short$oldersibs)
round(prop.table(nsib.tab,1),3)

ncol<-length(colnames(all.short)) #we will use this in later block as start column when adding new columns with all scaled on same scale


```


```{r DLDcategories, include=FALSE}
#Proportions meeting criteria for DLD
# And proportions having SALT
# Coded as: 
# 0 never
# 1 preschool
# 2 beyond 4 yr
# 3 ongoing
# 8 assessed only
# 9 no info

#now recode according N language tests low
#use the language tests scaled to 100 (cutoff 85), exclude reading, include GCC (cutoff 55)
v<-which(colnames(all.short)=='Vocab')
g<-which(colnames(all.short)=='gcc')
all.short$Ntestlow<-0
all.short$Ntestdone<-6
for(i in 1:5){
  thiscol<-i+v-1
  w<-which(all.short[,thiscol]<85)
  all.short$Ntestlow[w]<-all.short$Ntestlow[w]+1
  w<-which(is.na(all.short[,thiscol]))
  all.short$Ntestdone[w]<-all.short$Ntestdone[w]-1
}
#now add gcc
  w<-which(all.short[,g]<55)
  all.short$Ntestlow[w]<-all.short$Ntestlow[w]+1
  w<-which(is.na(all.short[,g]))
  all.short$Ntestdone[w]<-all.short$Ntestdone[w]-1

  #table of whole range of low test
 lowtab<- table(all.short$group8,all.short$Ntestlow)
 round(prop.table(lowtab,1),2)
 
   #binarise Ntestlow into 0/1 and 2+
 all.short$dldcat<-0
 w<-which(all.short$Ntestlow>1)
 all.short$dldcat[w]<-1
 #need to also add exclusionary categories: lowIQ,Autism,Hearing
 w<-which(all.short$pass_hearing==0)
 all.short$dldcat[w]<-4
 
 #asd use rater 1 dsm5, DAWBA gives 1 unsure, 2 ASD, 3 Asperger and 4 other. We have used the 4 = other for cases such as SCD where only 2 domains are met
  w<-c(which(all.short$asd_dsm_r1==2),which(all.short$asd_dsm_r1==3)) #categories 2-3 treated as Asd
 all.short$dldcat[w]<-3
 #lowIQ
 w<-which(all.short$piq<70)
 all.short$dldcat[w]<-2
 
 all.short$dldlevel<-all.short$dldcat #retain numeric level for easy of filtering later
 
 all.short$dldcat<-as.factor(all.short$dldcat)
 levels(all.short$dldcat)<-c('TD','DLD','lowIQ','ASD','hearing')
 dldcat<-table(all.short$dldcat,all.short$group8)
 
 round(prop.table(dldcat,2),2)
 
round(table(all.short$dldcat,all.short$sltcode),2) #check how SALT relates to DLD status
```

```{r make.subfiles, include=FALSE}
##Analyse language test and reading data
# First select the data 
# 
# Then make shorter sctfile that excludes the high bias group

# Make table for creating sums for Participants section
excludetab<-table(all.short$dldcat,all.short$bias,all.short$trisomy) #can index cells as DLDcat,bias,trisomy

#######################################################
# exclude those with ASD, low IQ or hearing exclusion #
#######################################################
all.short$group8xlevel<-all.short$group8level #easiest to make new grouping variable that has NA for excluded
w<-which(all.short$dldlevel>1)
all.short$group8xlevel[w]<-NA

#assign some values needed for participant report
ageband.group.tab<-table(all.short$ageband,all.short$group8xlevel)
N5yr1<-sum(ageband.group.tab[1,1:3]) #N 5 yr olds after exclusions - lobias
N5yr2<-sum(ageband.group.tab[1,4:6]) #N 5 yr olds after exclusions - hibias
missing.tab<-table(all.short$langdatacomplete,all.short$group8xlevel)
Nmisslang1<-sum(missing.tab[1,1:3]) #N all missing - lobias
Nmisslang2<-sum(missing.tab[1,4:6])#N all missing - hibias
cccNtab<-table(all.short$ccc2_consistent,all.short$group8xlevel,useNA='ifany')
Nmissccc<-sum(cccNtab[3,1:6])
Nbadccc<-sum(cccNtab[1,1:6])
lobias.sct<-filter(all.short,group8xlevel%in%c(1:3,8)) #we're using numeric code for factor level to make this easier

lobias.sct$Group<-as.factor(lobias.sct$group8xlevel)
levels(lobias.sct$Group)<-c('XXX','XXY','XYY','Twin')


#exclude those with incomplete data as they won't feature in MANOVA
lobias.sct2<-filter(lobias.sct,langdatacomplete==1) 



#exclude those outside twin age band - this is a new file so we can compute medians for those separately
lobias.sct3<-filter(lobias.sct2,ageband==2)

twin.TD<-filter(all.short,group8level==7) #we're using numeric code for factor level to make this easier
twin.TD$Group<-'TD.twin'

trisomy.only<-filter(all.short,group8xlevel<7)  #NB with 8x we omit the ASD etc exclusionary cases

#create table for reporting demographics
#age done using aggregate because something odd happens when using summarise!
agemean.tab <-aggregate(all.short$Age, by=list(all.short$group8xlevel),
                   FUN=mean, na.rm=TRUE)
agesd.tab <-aggregate(all.short$Age, by=list(all.short$group8xlevel),
                   FUN=sd, na.rm=TRUE)

mybkgtab <- data.frame(all.short %>%
  group_by(group8xlevel) %>%
  summarise(N = length(Age),
            Age   = mean(Age),
            Age.SD   = sd(Age),
              MotherEd   = mean(mo_educ, na.rm = TRUE),
            MotherEd.SD   = sd(mo_educ, na.rm = TRUE),
            Deprivation = mean(deprivz, na.rm = TRUE),
             Depriv.SD = sd(deprivz, na.rm = TRUE)))
mybkgtab2<-mybkgtab[1:8,c(1,2,3,5,7)]
mybkgtab2[,1]<-c('XXX: Low bias',
                'XXY: Low bias',
                 'XYY: Low bias',
                  'XXX: High bias',
                'XXY: High bias',
                 'XYY: High bias',
                 'Twin: No concerns',
                 'Twin: Language concerns')
mybkgtab2[,3]<-paste0(round(mybkgtab[1:8,3],0),' (',round(agesd.tab[,2],1),')')
mybkgtab2[,4]<-paste0(round(mybkgtab[1:8,5],1),' (',round(mybkgtab[1:8,6],2),')')
mybkgtab2[,5]<-paste0(round(mybkgtab[1:8,7],1),' (',round(mybkgtab[1:8,8],2),')')
colnames(mybkgtab2)<-c('Group','N','Age (mo)','Mo. educ.','Depriv. index')

```



```{r beeswarmgrid, include=FALSE}
#Now do a grid of beeswarm plots
longname<-c('Matrices','Block Design','Vocabulary','Comprehension','Sentence Repetition','Nonword Repetition','Oromotor',
            'Reading Accuracy','Reading Comprehension','Reading Rate','Speeded word reading','Speeded nonword reading',
            'Rapid Naming: Pictures','Rapid Naming: Digits')

png_lang_bees<-'beeswarm_lang.png' #name to save png file
png(png_lang_bees,width=1000,height=1000)
par(mfrow=c(5,3)) #5 row and 3 columns
par(mar=c(5.1,4.1,4.1,0.2))
#beeswarm jittered values (last value in jitter statement determines amount of jitter)

pointcol<-c(10,16,1) #use open circle, filled circle, circle +, for ageband 1-3
agebit<-pointcol[lobias.sct2$ageband]
#alldatabit<-pointcol[(1+lobias.sct$langdatacomplete)] #orig values are 0 and 1, so add 1
for (i in 1:14){
  mytitle<-longname[i]
  beeswarm(jitter(lobias.sct2[,nurange[i]],3)~Group , data = lobias.sct2, xlab='Group',ylab='Scaled score',spacing=.8,
           pch = 16,pwpch=agebit,cex=1.7,col=c(2,2,2,4),main=mytitle,ylim=c(55,135)) 
#can use bxplot to add medians (probs=.5), but as we are analysing means, use segments instead to add means
  
  #bxplot(lobias.sct2[,nurange[i]]~Group , data = lobias.sct2, probs = 0.5, col = 1,lty=3,add=TRUE) #adds medians for all cases with dotted line
#   bxplot(lobias.sct3[,nurange[i]]~Group , data = lobias.sct3, probs = 0.5, col = 1,add=TRUE) #adds medians just for those in Twin age range, with continuous line
  
  #add means
  for (g in 1:4){
    myg<-g
    if (g==4){myg<-8}
  mm<-mean(lobias.sct2[lobias.sct2$group8xlevel==myg,nurange[i]],na.rm=TRUE)
  segments(g-.4,mm,g+.4,mm,col = 1,lty=2,lwd=2)
  shortrange<-intersect(which(lobias.sct$group8xlevel==myg),which(lobias.sct$ageband==2))
  mm2<-mean(lobias.sct2[shortrange,nurange[i]],na.rm=TRUE)
  segments(g-.4,mm2,g+.4,mm2,col = 1,lty=1,lwd=2)
 
  }
  abline(a=100,b=0,col='darkgray',lty=1)
  abline(a=85,b=0,col='darkgray',lty=2)
  abline(a=70,b=0,col='darkgray',lty=2)
  
  mean_TD<-mean(twin.TD[,nurange[i]],na.rm=T)
  sd_TD<-sd(twin.TD[,nurange[i]],na.rm=T)
  
  up.lim<-mean_TD+sd_TD
  low.lim<-mean_TD-sd_TD
  
  polygon(c(0,0,5,5), c(up.lim, low.lim, low.lim, up.lim),
          col=adjustcolor("yellow",alpha.f=0.3), border = NA)
  
}
dev.off()
```





```{r lang.manova, include=FALSE}
#create dataframe to hold results from all Manovas
manova.df<-data.frame(matrix(NA,nrow=3,ncol=7))
colnames(manova.df)<-c('Comparison','Effect','Wilks','F','df1','df2','p')
manova.ccc.df<-manova.df #just copying basic file
#First just do groups 1-3
#NB cases with missing data are excluded, so doesn't matter if you use lobias.sct or lobias.sct2
#Result will be the same, but use lobias.sct2 to get numbers to report
gp3<-filter(lobias.sct2,group8level<4)
sct.matrix <- as.matrix(gp3[,nurange])
print('Check multivariate normality')
mshapiro.test(t(sct.matrix))
#Oh dear! not multivariate normal....

gp3.fit <- manova(sct.matrix ~ gp3$Group+gp3$mo_educ+gp3$deprivz)
mymanova3<-summary(gp3.fit, test="Wilks")
print('Manova on the 14 language measures for the 3 Low Bias SCT groups only')
mytab<-table(gp3$trisomy)
names(mytab)<-c('XXX','XXY','XYY')
print('Ns for analysis:')
print(mytab)
######################################################################################
#define function for populating manova.df here
manova.function <- function(mycomparison,effectnames,offset,manova.stats,manova.df){
  for (i in 1:length(effectnames)){
      j<-i+offset
    if(i==1){
      manova.df[j,1]<-mycomparison
    }
 
  manova.df[j,2]<-effectnames[i]
  manova.df[j,3]<-round(manova.stats[i,2],3)
  manova.df[j,4]<-round(manova.stats[i,3],2)
    manova.df[j,5]<-round(manova.stats[i,4],0)
    manova.df[j,6]<-round(manova.stats[i,5],0)
    manova.df[j,7]<-round(manova.stats[i,6],3)
}
return(manova.df)
}
######################################################################################

mycomparison<-paste0('A. Low bias: XXX (',mytab[1],
                      ') vs XXY (',mytab[2],
                       ') vs XYY (',mytab[3],')')

manova.stats<-mymanova3$stats
effectnames<-c('Group','Mo. education','Depriv. index')
offset<-0
manova.df<-manova.function(mycomparison,effectnames,offset,manova.stats,manova.df)
#summary.aov(gp3.fit) - if you want to inspect individual language measures

print('ANOVA on N tests low')
lowtest.aov<-aov(gp3$Ntestlow~gp3$Group)
summary(lowtest.aov)

#for comparison with twin group, need to restrict to those aged 6-11
w<-c(which(lobias.sct2$ageband==2))
lobias.agematch<-lobias.sct2[w,]
gp2.matrix <- as.matrix(lobias.agematch[,nurange])
mshapiro.test(t(gp2.matrix))

gp2.fit <- manova(gp2.matrix ~ lobias.agematch$trisomyvstwin+lobias.agematch$mo_educ+lobias.agematch$deprivz)
mymanova2<-summary(gp2.fit, test="Wilks")
mytab<-table(lobias.agematch$trisomyvstwin)
names(mytab)<-c('Trisomy','Twin')
print('Ns for analysis:')
print(mytab)
print('Manova on the 14 language measures for the SCT vs twin (language disordered) group, aged 6-11;11')
mymanova2

mycomparison<-paste0('B. Trisomies (',mytab[1],') vs. Lang concern twins (',mytab[2],'): aged 6-11')
manova.stats<-mymanova2$stats
offset<-nrow(manova.df)
manova.df<-manova.function(mycomparison,effectnames,offset,manova.stats,manova.df)
#summary.aov(gp3.fit) - if you want to inspect individual language measures

print('ANOVA on N tests low')
lowtest2.aov<-aov(lobias.agematch$Ntestlow~lobias.agematch$Group)
summary(lowtest2.aov)

#This plot not really adding anything, now we have coded age in the other beeswarm
png('png_2gp_bees.png',width=1000,height=1000)
par(mfrow=c(4,4)) #4 row and 4 columns
par(mar=c(5.1,4.1,4.1,0.2))
for (i in 1:14){
  mytitle<-longname[i]
  beeswarm(jitter(lobias.sct2[,nurange[i]],3)~trisomyvstwin , data = lobias.sct2, xlab='Group',ylab='Scaled score',
           pch = 16,cex=1,col=c(2,4),main=mytitle,ylim=c(50,140))
  bxplot(lobias.sct2[,nurange[i]]~trisomyvstwin , data = lobias.sct2, probs = 0.5, col = 1,add=TRUE) #adds medians
}
dev.off()

#NB correlation with mother's education, so compute this to check direction of effect

w<-which(colnames(lobias.sct2)=='Matrices')
w2<-w+13
templang<-rowMeans(lobias.sct2[w:w2])
cor(lobias.sct2$mo_educ,templang)
moedreg<-lm(templang~lobias.sct2$mo_educ)
plot(lobias.sct2$mo_educ,templang)
mymoedreg<-summary(moedreg)
my.moedregb<-round(mymoedreg$coefficients[2],1)

#lang concern twin only
wt<-which(lobias.sct2$group8xlevel==8)
moedregt<-lm(templang[wt]~lobias.sct2$mo_educ[wt])

```


```{r heatmapcorr, include=FALSE}
##Inspect correlations with heat map
# Problems combining heatmaps into same plot?
# These plots no longer needed, given that no group effect - previously, before excluding older SCT cases, there was a group effect, which seemed driven by different associations between reading/language measures. Can still see a suggestion of this - lighter red for lang/read correlations in twins. But as NS, not worth exploring further.
 png_heat<-'heat_plot_sct.png'
 png(png_heat,width=1000,height=500)
cormat_SCT <- round(cor(lobias.sct2[lobias.sct2$mycontrast==1,nurange],use="na.or.complete"),2)
ggcorrplot(cormat_SCT,title='SCT')
dev.off()
 png_heat<-'heat_plot_twin.png'
 png(png_heat,width=1000,height=500)
cormat_twin <- round(cor(lobias.sct2[lobias.sct2$mycontrast==8,nurange],use="na.or.complete"),2)
ggcorrplot(cormat_twin,title='Twin')
dev.off()
```


```{r lowhi.sct, include=FALSE}
##Comparison of low and high bias groups
#Use trisomy.only group for this

png_lang_bees<-'beeswarm_lang_bias.png' #name to save png file
png(png_lang_bees,width=1000,height=1000)
par(mfrow=c(5,3)) #5 row and 3 columns
par(mar=c(5.1,4.1,4.1,0.2))
#beeswarm jittered values (last value in jitter statement determines amount of jitter)

trisomy.only<-filter(trisomy.only,ageband>1) #remove youngest bcs missing data
trisomy.only<-filter(trisomy.only,langdatacomplete==1)
pointcol<-c(10,16,1) #use open circle, filled circle, circle +, for ageband 1-3
agebit<-pointcol[trisomy.only$ageband]
#alldatabit<-pointcol[(1+lobias.sct$langdatacomplete)] #orig values are 0 and 1, so add 1

for (i in 1:14){
  mytitle<-longname[i]
  beeswarm(jitter(trisomy.only[,nurange[i]],3)~group8xlevel , data = trisomy.only, xlab='Group',ylab='Scaled score',spacing=.8,labels='',
           pch = 16,pwpch=agebit,cex=1.7,col=c(2,2,2,1,1,1),main=mytitle,
           ylim=c(55,140))
  bxplot(trisomy.only[,nurange[i]]~group8xlevel, data = trisomy.only, probs = 0.5, 
         col = 1,lty=1,add=TRUE) #adds medians for all cases with dotted line
   abline(a=100,b=0,col='darkgray',lty=1)
  abline(a=85,b=0,col='darkgray',lty=2)
  abline(a=70,b=0,col='darkgray',lty=2)
  #same settings for polygon as before: shows TD twin range
  polygon(c(0,0,7,7), c(up.lim, low.lim, low.lim, up.lim),
          col=adjustcolor("yellow",alpha.f=0.3), border = NA)
  text(1,133,'XXX',cex=1)
text(2,133,'XXY',cex=1)
text(3,133,'XYY',cex=1)
text(4,133,'XXX',cex=1)
text(5,133,'XXY',cex=1)
text(6,133,'XYY',cex=1)
text(2,140,'<---------Low bias--------->')
text(5,140,'<---------High bias--------->')

}
dev.off()
```

```{r manova2, include=FALSE}


#check that each bias group is multivariate normal
b<-which(trisomy.only$bias==1)
sctb.matrix <- as.matrix(trisomy.only[b,nurange])
 #transpose cols and rows for mshapiro.test !
print('bias cases')
mshapiro.test(t(sctb.matrix))
sctn.matrix <- as.matrix(trisomy.only[-b,nurange])
 #transpose cols and rows for mshapiro.test !
print('No bias cases')
mshapiro.test(t(sctn.matrix))

sct2.matrix <- as.matrix(trisomy.only[,nurange])
gp6.fit <- manova(sct2.matrix ~ trisomy.only$bias*trisomy.only$trisomy+trisomy.only$mo_educ+trisomy.only$deprivz)
mymanova6<-summary(gp6.fit, test="Wilks")
print('Manova on the 14 language measures for the 3 Low Bias SCT groups vs 3 High Bias')
mytab<-table(trisomy.only$group8xlevel)

print('Ns for analysis:')
print(mytab)
mymanova6

mycomparison<-'C. 3 trisomy x 2 bias group'
effectnames<-c('Bias','Trisomy','Mo. education','Depriv. index','Bias x Trisomy')
offset<-nrow(manova.df)
manova.stats<-mymanova6$stats
manova.df<-manova.function(mycomparison,effectnames,offset,manova.stats,manova.df)
manova.df[(2+offset),1]<-paste0('- Ns Low bias: XXX (',mytab[1],'), XXY (',mytab[2],'), XYY (',mytab[3],')')
manova.df[(3+offset),1]<-paste0('- Ns High bias: XXX (',mytab[4],'), XXY (',mytab[5],'), XYY (',mytab[6],')')  
#summary.aov(gp6.fit) #- if you want to inspect individual language measures

print('ANOVA on N tests low: Vocab, Comp, Sent Rep, Nwrep, Oromotor + GCC')
lowtest6.aov<-aov(trisomy.only$Ntestlow~trisomy.only$trisomy*trisomy.only$bias)
summary(lowtest6.aov)
lowdld6.aov<-aov(trisomy.only$dldlevel~trisomy.only$trisomy*trisomy.only$bias)
summary(lowdld6.aov)
cutofflang.tab<-table(trisomy.only$Ntestlow,trisomy.only$group8xlevel)
cutofflang.tab

```


```{r ccc2, include=FALSE}
##CCC-2 analysis
# Consider how CCC-2 profiles for SCTs compare with those in our previous paper. Also include twins with concerns as comparison group.  Plot subscales; also analyse the GCC and SIDC.

w<-which(colnames(all.short)=='ccc_a')
x<-which(colnames(all.short)=='group8')
y<-which(colnames(all.short)=='group8xlevel')
z<-which(colnames(all.short)=='trisomy')
z1<-which(colnames(all.short)=='bias')
z2<-which(colnames(all.short)=='dldlevel')
z3<-which(colnames(all.short)=='mo_educ')
z4<-which(colnames(all.short)=='deprivz')
#make file with ID,  and CCC results, as well as group8; gender, age added at end
cccbit<-all.short[,c(1,(w-2):(w+10),x,y,z,z1,z2,z3,z4,2,3)]#2 before ccc_a is GCC and SIDC

cccbit<-filter(cccbit,dldlevel<2) #remove ASD, low IQ etc
w<-which(is.na(cccbit$ccc2_consistent))
cccbit$ccc2_consistent[w]<--1 #we need a count of missing data, so convert NA to -1
consistent.ccc<-table(cccbit$group8,cccbit$ccc2_consistent)

colnames(consistent.ccc)<-c('No data','Inconsistent','Consistent')
consistent.ccc
print('Proportions of non-response to CCC-2')
print(paste('All low bias SCT:',sum(consistent.ccc[1:3,1])/sum(consistent.ccc[1:3,])))
print(paste('All high bias SCT:',sum(consistent.ccc[4:6,1])/sum(consistent.ccc[4:6,])))
print(paste('Twin: no concerns:',sum(consistent.ccc[7,1])/sum(consistent.ccc[7,])))
print(paste('Twin: language concerns:',sum(consistent.ccc[8,1])/sum(consistent.ccc[8,])))

#total responses
print('Overall response rates:')
print(paste('All SCT:',1-sum(consistent.ccc[1:6,1])/sum(consistent.ccc[1:6,])))
print(paste('Twin:',1-sum(consistent.ccc[7:8,1])/sum(consistent.ccc[7:8,])))

```
```{r logreg.cccresponse, include=FALSE}
##Analysis to explore factor affecting response rates

all.short2<-all.short
for(i in 1:length(all.short2[,1]))
{
all.short2$response[i]<-ifelse(is.na(all.short2$gcc[i]),0,1)
all.short2$SCT[i]<-ifelse(is.na(all.short2$trisomy[i]),0,1)
}

#Make factor scores for predicting based on severity
model.f5a <- ' f1 =~ wasi_vocab_ss + wdck_jhsn_ss + sent_rep_ss + oromotor_ss_2 
              wasi_vocab_ss ~~ wdck_jhsn_ss'     
fit.mod.E2 <- cfa(model.f5a, data = all.short2,estimator = "ML",missing = "ML")
lbls<-c("Vocabulary", "Woodcock\nJohnson", "Sentence\nRepetition", "Oromotor","Language")

#adjust scores to make suitable for logistic regression
all.short2$lang_severity <- as.numeric( predict(fit.mod.E2)) #factor scores

all.short2$mo_educ<-car::recode(all.short2$mo_educ,"0=1") #recode 0 as 1.

all.short2$mo_educ<-as.factor(all.short2$mo_educ) #make factor for GLM
#run logistic regression
NR1 <- glm(response ~ lang_severity + mo_educ + singlepar + SCT,family=binomial(link='logit'),data=all.short2)
summary(NR1)

```
```{r ccc.analyse, include=FALSE}
##Now analyse CCC-2 for 8 groups

#First remove the cases with no response or inconsistent
ccc.data<-filter(cccbit,ccc2_consistent>0)


#create dataframe for summary results
ccc.df<-data.frame(matrix(NA,nrow=8,ncol=26))
ccc.df[,1]<-levels(ccc.data$group8)
ccc.df[,2]<- table(ccc.data$group8)
thiscol<- 1
for (i in 1:12){
  thiscol<-thiscol+2
  thisdat<-ccc.data[,i+1]
  ag <- aggregate(thisdat~ group8, ccc.data, function(x) c(mean = mean(x), se = sd(x)/sqrt(length(x))))
  ccc.df[,(thiscol:(thiscol+1))]<-ag[,2]
}
colnames(ccc.df)<-c('Group','N','GCC.mean','GCC.se','SIDC.mean','SIDC.se','A.mean','A.se','B.mean','B.se','C.mean','C.se','D.mean','D.se',
                    'E.mean','E.se','F.mean','F.se','G.mean','G.se','H.mean','H.se',
                    'I.mean','I.se','J.mean','J.se')
```

```{r makedata.cccplot, include=FALSE}
#Need data in long form for ggplot2
#set up the long form df with data from scale A
ccc.dflong<-ccc.df[1:8,c(1,2,2,7:8)]
colnames(ccc.dflong)[3:5]<-c('scale','mean','se')
ccc.dflong$scale<-1
mybase<-ccc.dflong

#Now bolt on the other scales
for (j in 2:10){
  #rows to write to in new long df
  startrow<-(j-1)*8+1
  endrow<-(j*8)
  #cols to read from in old df
  startcol<-7+(j-1)*2
  ccc.dflong<-rbind(ccc.dflong,mybase) #just duplicate first 8 rows before overwriting them
  ccc.dflong$scale[startrow:endrow]<-j
  ccc.dflong[startrow:endrow,4:5]<-ccc.df[,startcol:(startcol+1)]
  
}
ccc.dflong$scale<-as.factor(ccc.dflong$scale)
levels(ccc.dflong$scale)<-c('A','B','C','D','E','F','G','H','I','J')
linetypes<-c(1,1,1,2,2,2,1,1)
ccc.dflong$lines<-as.factor(linetypes)
ccc.dflong$plotgroup<-rep(c('XXX','XXY','XYY','XXX','XXY','XYY','twin_typical','twin_concerns'),10)
```

```{r plotccc,echo=FALSE}

png_ccc<-'cccplot.png' #name to save png file
png(png_ccc,width=800,height=600)
#par(mar=c(5.1,4.1,4.1,0.2)) #if need to alter margins
#Plot the means
ymax <- 12
ymin <- 0
cccplot<-ggplot(ccc.dflong, aes(x=scale, y=mean, colour=plotgroup,group=Group)) +
  geom_line(aes(linetype=lines)) + geom_point(shape=21, fill="white") + 
  ylim(ymin,ymax) +ylab('Mean scaled score')
cccplot+geom_errorbar(width=.1, aes(ymin=mean-se, ymax=mean+se))
dev.off()
```

```{r ccc2.manova, include=FALSE}
##Manova for CCC-2
#Follow https://www.statmethods.net/stats/anova.html
#First just do groups 1-3 on the 10 subscales
ccc.gp3<-filter(ccc.data,group8xlevel<4)
colA<-which(colnames(ccc.gp3)=='ccc_a')

Y <- as.matrix(ccc.gp3[,colA:(colA+9)])
fit.cccsct <- manova(Y ~ ccc.gp3$group8+ccc.gp3$mo_educ+ccc.gp3$deprivz)
summary(fit.cccsct, test="Wilks")
mytab<-table(ccc.gp3$group8)
mycomparison<-paste0('A. XXX (',mytab[1],'), XXY (',mytab[2],'),XYY (',mytab[3],')')
manova.stats<-summary(fit.cccsct, test="Wilks")$stats
offset<-0
effectnames<-c('Group','Mo. education','Depriv. index')
manova.ccc.df<-manova.function(mycomparison,effectnames,offset,manova.stats,manova.ccc.df)



#Now add twin ld group
temp<-filter(ccc.data,group8xlevel==8)
nsct<-nrow(ccc.gp3)
ccc.gp2<-rbind(ccc.gp3,temp)
ccc.gp2$group2<-1
ccc.gp2$group2[1:nsct]<-2
Y <- as.matrix(ccc.gp2[,colA:(colA+9)])
fitccc2 <- manova(Y ~ ccc.gp2$group2+ccc.gp2$mo_educ+ccc.gp2$deprivz)
summary(fitccc2, test="Wilks")
manova.stats<-summary(fitccc2, test="Wilks")$stats
offset<-nrow(manova.ccc.df)
mycomparison<-'B. All SCT vs Lang concern twins'
effectnames<-c('Group','Mo. education','Depriv. index')
manova.ccc.df<-manova.function(mycomparison,effectnames,offset,manova.stats,manova.ccc.df)


#Now do full 2 way analysis on hi and low bias trisomies
temp1<-filter(ccc.data,group8xlevel<7)
Y6 <- as.matrix(temp1[,colA:(colA+9)])
fitccc6 <- manova(Y6 ~ temp1$trisomy*temp1$bias+temp1$mo_educ+temp1$deprivz)
summary(fitccc6, test="Wilks")
#summary.aov(fitccc6) #- if you want to inspect individual language measures
manova.stats<-summary(fitccc6, test="Wilks")$stats
offset<-nrow(manova.ccc.df)
mycomparison<-'C. Bias x Trisomy'
effectnames<-c('Bias','Trisomy','Mo. education','Depriv. index','Bias x Trisomy')
manova.ccc.df<-manova.function(mycomparison,effectnames,offset,manova.stats,manova.ccc.df)



#GCC and SIDC - just do anova
aov.gcc<-aov(temp1$gcc~temp1$trisomy*temp1$bias)
summary(aov.gcc)

aov.scdi<-aov(temp1$scdi~temp1$trisomy*temp1$bias)
summary(aov.scdi)
```

```{r beeswarm.gcc, include=FALSE}
png_gcc_bees<-'beeswarm_gcc.png' #name to save png file
png(png_gcc_bees,width=400,height=300)
beeswarm(temp1$gcc~temp1$trisomy*temp1$bias, xlab='Group',ylab='GCC' ,
           pch = 16,cex=.9,col=c(2,2,2,1,1,1),ylim=c(0,120))
  bxplot(temp1$gcc~temp1$trisomy*temp1$bias, probs = 0.5, col = 1,lty=1,add=TRUE) #adds medians for all cases with dotted line
  
    
  mean_TD<-mean(twin.TD$gcc,na.rm=T)
  sd_TD<-sd(twin.TD$gcc,na.rm=T)
  
  up.lim<-mean_TD+sd_TD
  low.lim<-mean_TD-sd_TD
  
  polygon(c(0,0,8,8), c(up.lim, low.lim, low.lim, up.lim),
          col=adjustcolor("yellow",alpha.f=0.3), border = NA)
abline(h=55,lty=3)
dev.off()
```



```{r flow_sct, include=FALSE}
## Flow diagrams

all.short$exclude<-car::recode(all.short$dldlevel,"0=0;1=0;2=1;3=1;4=1")
all.short$ccc2_data<-car::recode(all.short$ccc2_consistent,"1=1;0=0;NA=0")


  y1<-table(all.short$randomtwininclude)[[1]]
  
  biasLow <- table(all.short$bias)[[1]]
  biasHigh <- table(all.short$bias)[[2]]
  #--------------------------------------------------------------------------------#
  
  mytab<-table(all.short$trisomy,all.short$bias,all.short$exclude)
  
  #--------------------------------------------------------------------------------#
    mytab1<-table(all.short$langdatacomplete,all.short$group8xlevel)
    #those with complete language data in row 2
  
  #--------------------------------------------------------------------------------#
  
  all.short.age<-all.short[all.short$ageband==2,]
  
  mytab2<-table(all.short.age$trisomy,all.short.age$bias,all.short.age$exclude)
  
  
  #--------------------------------------------------------------------------------#
  
  all.short.ccc<-all.short[all.short$ccc2_data==1,]
  
  mytab3<-table(all.short.ccc$trisomy,all.short.ccc$bias,all.short.ccc$exclude)
  
  #--------------------------------------------------------------------------------#

 excl3 <- sum(mytab[1:3,1,2])
 excl3a <- mytab[1,1,2]
 excl3b <- mytab[2,1,2]
 excl3c <- mytab[3,1,2]
 
 excl4 <- sum(mytab[1:3,2,2])
 excl4a <- mytab[1,2,2]
 excl4b <- mytab[2,2,2]
 excl4c <- mytab[3,2,2]
 
 xxx_a0<-mytab[1,1,1]
 xxx_a1<-mytab1[2,1]
 xxx_a2<-mytab2[1,1,1]
 xxx_a3<-mytab3[1,1,1]
 
  xxy_a0<-mytab[2,1,1]
 xxy_a1<-mytab1[2,2]
 xxy_a2<-mytab2[2,1,1]
 xxy_a3<-mytab3[2,1,1]
   
 xyy_a0<-mytab[3,1,1]
  xyy_a1<-mytab1[2,3]
 xyy_a2<-mytab2[3,1,1]
 xyy_a3<-mytab3[3,1,1]

 xxx_b0<-mytab[1,2,1]
 xxx_b1<-mytab1[2,4]
 xxx_b2<-mytab2[1,2,1]
 xxx_b3<-mytab3[1,2,1]
   
 xxy_b0<-mytab[2,2,1]
 xxy_b1<-mytab1[2,5]
 xxy_b2<-mytab2[2,2,1]
 xxy_b3<-mytab3[2,2,1]
   
 xyy_b0<-mytab[3,2,1]
 xyy_b1<-mytab1[2,6]
 xyy_b2<-mytab2[3,2,1]
 xyy_b3<-mytab3[3,2,1]
   
  #now create flow chart ; TB denotes top to bottom
  #Need to add labels along the side: top row 'Reason for testing or Time of testing'
  #Then trisomy, then with DAWBA data

  flow1<-DiagrammeR::grViz("
              digraph a_nice_graph {
              
              # node definitions with substituted label text
              node [shape = plaintext, fontname = Helvetica]
              X[label='@@1']
              Y[label= 'Ascertainment\\nbias']
              Z[label = 'Trisomy']
              
              node [shape=square]
              A[label='@@2',fillcolor=lightBlue]
              
              C[label='@@3']
              D[label='@@4']

              E[label='@@5']
              F[label='@@6']
              G[label='@@7']
              H[label='@@8']

              I[label='@@9']
              J[label='@@10']
              K[label='@@11']
              L[label='@@12']

              node [shape = plaintext]
              Q[label='* With complete language data, ** Complete language: Aged 6 to 11 yr, *** With useable CCC-2 data']
              
 
              
              # edge definitions with the node IDs
              A -> {C,D}
              {rank=same ;C -> H}
              C -> {rank=same ;E,F,G}
              {rank=same ;D -> L}
              D -> {rank=same ;I,J,K}
             
              
              
              X -> Y [alpha=0,color='white']
              Y -> Z [alpha=0,color='white']
              F -> Q [alpha=0,color='white']
              }
              
              [1]: paste0(' ')
              [2]: paste0('SCT:\\n N = ',y1)
              [3]: paste0('Low bias,\\n N = ',biasLow)
              [4]: paste0('High bias, \\nN = ',biasHigh)
              [5]: paste0('XXX','\\n', 'N = ',xxx_a0,'\\n','N* = ',xxx_a1,'\\n', 'N** = ',xxx_a2,'\\n', 'N*** = ',xxx_a3)
              [6]: paste0('XXY',' \\n', 'N = ',xxy_a0,' \\n','N* = ',xxy_a1,' \\n', 'N** = ',xxy_a2,' \\n', 'N*** = ',xxy_a3)
              [7]: paste0('XYY',' \\n','N = ',xyy_a0,' \\n', 'N* = ',xyy_a1,' \\n', 'N** = ',xyy_a2,' \\n', 'N*** = ',xyy_a3)
              [8]: paste0('Excluded',' \\n', '(N = ',excl3,')\\n',excl3a,' XXX','\\n',excl3b,' XXY','\\n',excl3c,' XYY')
              [9]: paste0('XXX',' \\n', 'N = ',xxx_b0,' \\n','N* = ',xxx_b1,' \\n', 'N** = ',xxx_b2,' \\n', 'N*** = ',xxx_b3)
              [10]: paste0('XXY',' \\n', 'N = ',xxy_b0,' \\n','N* = ',xxy_b1,' \\n', 'N** = ',xxy_b2,' \\n', 'N*** = ',xxy_b3)
              [11]: paste0('XYY',' \\n','N = ',xyy_b0,' \\n', 'N* = ',xyy_b1,' \\n', 'N** = ',xyy_b2,' \\n', 'N*** = ',xyy_b3)
              [12]: paste0('Excluded',' \\n', '(N = ',excl4,')\\n',excl4a,' XXX','\\n',excl4b,' XXY','\\n',excl4c,' XYY')
              ")
  
  

flow1 %>% export_svg %>% charToRaw %>% rsvg_png("SCT_flow.png")

```



```{r flow_twins, include=FALSE}

#Now do plot for twin data

y1<-table(all.short$randomtwininclude)[[2]]

exc1<-table(all.short$exclude[all.short$randomtwininclude==1])[[2]]
inc1<-table(all.short$exclude[all.short$randomtwininclude==1])[[1]]

all.short$ccc2_data<-car::recode(all.short$ccc2_consistent,"1=1;0=0;NA=0")

all.short.exc<-all.short[all.short$exclude==0 & all.short$randomtwininclude==1,]

noconcernM<-table(all.short.exc$parentconc[all.short.exc$female==0])[[1]]
concernM<-table(all.short.exc$parentconc[all.short.exc$female==0])[[2]]

noconcernF<-table(all.short.exc$parentconc[all.short.exc$female==1])[[1]]
concernF<-table(all.short.exc$parentconc[all.short.exc$female==1])[[2]]

CCC_LI_Y <- table(all.short.exc$ccc2_data[all.short.exc$parentconc==1])[[2]]
CCC_LI_N <- table(all.short.exc$ccc2_data[all.short.exc$parentconc==1])[[1]]

CCC_TD_Y <- table(all.short.exc$ccc2_data[all.short.exc$parentconc==0])[[2]]
CCC_TD_N <- table(all.short.exc$ccc2_data[all.short.exc$parentconc==0])[[1]]

#now create flow chart ; TB denotes top to bottom

label3<-' '

flow2<-DiagrammeR::grViz("
            digraph twinflow {
            
            # node definitions with substituted label text
            node [shape = plaintext, fontname = Helvetica]
            X[label='@@1']
             Y[label= 'Exclude:\\n ASD, IQ, hearing']
            Z[label= 'Total N twin children']
            Q[label='CCC-2 data\\n available']
            
            node [shape=square]
            I[label='@@10']
            A[label='@@2',fillcolor=lightBlue]
            B[label='@@3']
            
            C[label='@@4']
            D[label='@@5']
            E[label='@@6']
            F[label='@@7']
            G[label='@@8']
            H[label='@@9']
            
            
            
            # edge definitions with the node IDs
            I -> {A B}
            B -> {G H}
            G -> {C D}
            H -> {E F}
            
            
            
            X -> Y [alpha=0,color='white']
            Y -> Z [alpha=0,color='white']
            Z -> Q [alpha=0,color='white']
            }
            
            [1]: paste0(label3, ':\\n ',' ')
            [2]: paste0('Excluded:\\n N = ',exc1)
            [3]: paste0('Included:\\n N = ',inc1)
            [4]: paste0('with CCC-2:\\nN = ',CCC_LI_Y)
            [5]: paste0('without CCC-2:\\nN = ',CCC_LI_N)
            [6]: paste0('with CCC-2:\\nN = ',CCC_TD_Y)
            [7]:  paste0('without CCC-2:\\nN = ',CCC_TD_N)
            [8]: paste0('LI: \\n Parental concern \\n or SALT,\\n N (male) = ',concernM,'\\n N (female) = ',concernF)
            [9]: paste0('TD: \\n No concern \\n No SALT,\\n N (male) = ',noconcernM,'\\n N (female) = ',noconcernF)
            [10]: paste0(y1,' twin pairs, \\n 1 selected at \\n random')
            
            ")

flow2 %>% export_svg %>% charToRaw %>% rsvg_png("twins_flow.png")

```



```{r sessinfo, include=FALSE}
sessionInfo()
```




Chromosome trisomies arise when an error of cell division leads to an egg or sperm that contains two rather than one copy of the chromosome. Trisomies that affect one of the autosomes are usually lethal, and in survivors they lead to marked physical and mental abnormalities. Trisomies of the sex chromosomes, however, have milder impacts, and often go undetected.  This complicates the study of the consequences of sex chromosome trisomies in two ways; first, it can be difficult to recruit large samples of cases, and second, those that are studied may not be representative of the population. In particular, there is a danger of overestimating the severity of impairment if we include cases where discovery of the trisomy was prompted by genetic testing to investigate developmental abnormalities.  
In the 1960s, a multicentre project was initiated with the aim of evaluating the impact of sex chromosome trisomies in samples identified on newborn screening that were free from ascertainment bias (Robinson et al., 1979). The three kinds of trisomy - trisomy X (47,XXX), Klinefelter’s syndrome (47,XXY) and 47,XYY karotypes were all found to be associated with neurodevelopmental problems, particularly affecting language and motor functions (Leggett et al, 2010).  
To our knowledge, there have been no more newborn screening studies initiated in the past 50 years. One reason is practical: each of the three trisomy types has a prevalence ranging from 1 in 600 to 1 in 1000, so many thousands of cases need to be screened to identify even a small sample. Second, ethical issues are raised when a screening procedure identifies a trisomy in a newborn: while knowledge of the trisomy could be useful in helping parents take steps to ensure early intervention, it can also be damaging by creating anxiety (Valentine, 1979).  
It is now possible to detect sex chromosome trisomies prenatally, in the course of prenatal screening. Prenatal screening for Down syndrome was developed in the 1980s and led to cases of sex chromosome trisomy being detected as an incidental finding (Cuckle & Maymon, 2016). Because prenatal screening is offered to older mothers, there is some bias in samples identified this way, but detection of the trisomy is not dependent on the child’s developmental outcome.   
A study of neurodevelopmental of outcomes of prenatally-identified children was reported by Bishop et al (2011). Outcome measures were based on standardized parental report, as it was deemed unethical to conduct direct testing of children who might not be aware of their trisomy. This study was consistent with the earlier newborn screening studies in finding a high rate of language difficulties in all three trisomies, but in addition, there was an elevated risk of autism spectrum disorder diagnosis in boys with both XXY and XYY karyotypes. This had not been reported in the earlier prospective studies of newborns, but it seems likely that this was because diagnostic criteria for autism were far more stringent in the 20th century. A second group of children with trisomies detected postnatally was found to have a similar profile, but levels of all neurodevelopmental problems were higher, as would be expected in a group where there was ascertainment bias. In the course of this study, we established that 43% of children had been told about the trisomy. This suggested that a study that involved direct neurodevelopmental assessment of children would be feasible.  
The association with language problems is of interest because there are few genetic aetiologies that selectively impair language development. There are two reasons why it is of interest to compare the language phenotype of children with sex chromosome trisomies with that seen in children with developmental language disorder (DLD) of unknown origin. First, a comparative study could also help determine whether interventions that have been devised for children with DLD are also likely to be effective for children with a sex chromosome trisomy (see Bishop et al., 2017). Second, if the similarities are not just superficial, then studying how an extra chromosome affects language development could help us understand aetiology of DLD (Bishop & Scerif, 2011).  Studies to date have suggested that there may be different profiles of impairment in the three types of trisomy, which could be related to the different genetic and hormonal impacts of an extra X or Y chromosome, but we still have only a very limited body of data from samples free from ascertainment bias.   
In 2011 we embarked on data collection for a new study of children with sex chromosome trisomies, including both pre- and post-natally identified cases. The goals were, first, to document the nature and range of language and behavioural difficulties in the three types of trisomy, and second, to test whether individual variation in outcomes could be predicted by variation in specific genetic variants. Here we report descriptive data relating to the first question, with a particular emphasis on language characteristics.  Behavioural and psychiatric findings will be covered in a companion paper. To date we have not found any associations of phenotypes with variants in CNTNAP2 and NRXN1 genes; these analyses are reported elsewhere (Newbury et al, 2018).  
Our focus here is both on documenting whether there are reliable differences in language and cognitive outcomes for the three types of trisomy, and how far the pattern of language problems resembles that seen in children with developmental language disorder (DLD) who do not have any known genetic abnormalities. To minimize ascertainment bias, these analyses were restricted to children whose trisomy was either detected prenatally, or as an incidental finding during other medical investigations (referred to as a ‘low bias’ group). To address the latter question, we compared the results of trisomic children with those of children who had participated in a twin study of language development and disorders, and who had been given the same test battery. Finally, we considered how far trisomy outcomes are influenced by ascertainment bias, by comparing the low bias group with a high bias group whose trisomies were discovered in the course of investigation of behavioural or neurodevelopmental problems.  
The specific questions we considered were:  
1.	Is there a distinctive language profile for each of the three types of trisomy?  
2.	Is the cognitive profile of children with sex chromosome trisomies similar to that of children with language problems who have the usual XY or XX karyotype?  
3.	How far does ascertainment bias affect the language and cognitive profiles of children with sex chromosome trisomies?  

## Participants  
### Sex chromosome trisomy cases  
We recruited `r tab_sct[1,1]+tab_sct[1,2]` girls with XXX, `r tab_sct[2,1]+tab_sct[2,2]` boys with XXY and `r tab_sct[3,1]+tab_sct[3,2]` boys with XYY to the study, in the age range from 5 years 0 months to 15 years 11 months. `r round(100*sum(origstudy)/sum(tab_sct),0)`% of the children had participated in the Bishop et al (2011) study. Recruitment was via NHS Clinical Genetics centres, two support groups (Unique: the Rare Chromosome Support Group, and the Klinefelter Syndrome Association) and self-referral through social media. To be included in the study, the child had to be aware of their trisomy status.  

**Ascertainment bias**. During the initial telephone interviews, caregivers were asked how their child was diagnosed, in particular whether this followed postnatal testing motivated by neurodevelopmental/behavioural problems. The phenotype of such children may be more severe, potentially biasing the sample, and so, as shown in Figure 1, we grouped them prospectively as a high-risk-of-bias (‘High Bias’) subgroup, comprised of `r sum(excludetab[,2,1])` XXX, `r sum(excludetab[,2,2])` XXY and `r sum(excludetab[,2,3])` XYY cases. All other children formed the low-risk-of-bias (‘Low Bias’) subgroup, with `r sum(excludetab[,1,1])` XXX, `r sum(excludetab[,1,2])` XXY and `r sum(excludetab[,1,3])` XYY cases.  

**Exclusionary criteria**. Given our current focus on language disorder, we excluded children who had a nonverbal ability scaled score more than 2 SD below the mean (`r sum(excludetab[3,1,])` Low Bias and `r sum(excludetab[3,2,])` High Bias), those who met criteria for autism spectrum disorder (ASD) (`r sum(excludetab[4,1,])` Low Bias and `r sum(excludetab[4,2,])` High Bias) and those who failed a hearing screen (`r sum(excludetab[5,1,])` Low Bias and `r sum(excludetab[5,2,])` High Bias). This enabled us to make a direct comparison with the twin sample, which also excluded children on this basis, and to see how far language problems were seen in children who did not have accompanying conditions that are likely to affect language function. However, note that this means that our conclusions are based on a selected sample of trisomy cases. ASD and other diagnoses in children with sex chromosome trisomies will be the focus of a companion paper.  

**Missing data**. Where a child had missing data on just one of the language/cognitive tests, that value was prorated from the mean of that child on the other measures. Likewise, when there was just one missing measure on reading-related tests, the value as prorated from other reading measures. `r sum(Nmisslang1,Nmisslang2)` children had more missing data than this, and were excluded from the current analyses. This included `r N5yr1+N5yr2` five-year-olds, for whom standard scores were not computable as they fell below the age range of norms. All but `r Nmisslang1-N5yr1` cases of missing data in children aged over 5 years came from the High Bias group, reflecting refusal or inability to attempt some tests. Norms for the CCC-2 extend across the full age range covered here, but `r Nmissccc+Nbadccc` children had missing data because parents either did not complete the CCC-2 (N = `r Nmissccc`), or because they failed the consistency check (N = `r Nbadccc`), which is suggestive of invalid responses.  

**Age range**. When comparing test profiles of trisomy cases with those of twins with language concerns, we restricted consideration to children covering the same age range, i.e. 6 yr 0 months to 11 yr 11 months. 


The numbers included in different analyses are shown in the flowchart in Figure 1.

![Figure 1. Numbers of children with sex chromosome trisomies included in different analyses.](SCT_flow.png)

### Comparison group  

The comparison sample consisted of twins recruited via fliers sent to primary schools around the UK, advertisements on our group's website and via twins' clubs. The age range for this sample was narrower than for the SCT cases. We aimed to recruit families with twin children aged between 6 years 0 months and 11 years 11 months, with over-representation of those where one or both twins had language or literacy problems. Further details are provided in Wilson and Bishop (2018).   
Our second question involved comparing the language profile of children with trisomies with that of children with language disorders of unknown cause. Usually, language disorder would be diagnosed on the basis of language test scores, but if we took that approach, the result would be a foregone conclusion, in that we would use the same measures to define the independent variable (language status) and the dependent variables. To avoid this circularity, twin children were subdivided according to parental concern about language and history of speech and language therapy (SALT), rather than by language test scores, which were treated as dependent variables. Parental concern was coded from the initial interview, and used to divide the twin sample into a ‘language concerns’ group with ongoing parental concerns about oral language (mild or severe) and the remainder (which included some cases where there had been transient concerns in preschool that had resolved, or where the concern affected only reading).  The latter is referred to as the ‘no concerns’ group. In addition, children who had received speech and language therapy after the age of 4 years were included in the ‘language concerns’ group. 
We followed the CATALISE criteria for DLD and excluded twin pairs if either twin had a pre-existing diagnosis of autism spectrum disorder (ASD), failed the hearing screen, had a performance IQ score below 70, or had a serious long-term illness; other diagnoses, such as dyslexia, ADHD, or dyspraxia, were not grounds for exclusion (Bishop et al., 2017).   
One twin from each pair was selected at random to avoid dependencies in the data. The final sample consisted of `r concernM+concernF` children in the language concerns group, and `r noconcernM+noconcernF` children with no concerns.  The latter group were used to derive a normative range against which to evaluate the other groups on language measures. Figure 2 shows the number of twin children included in the study, and the number for whom useable CCC-2 data were available.

![Figure 2. Numbers of twin children with and without language concerns.](twins_flow.png)

## Methods

### Language, literacy and cognitive assessments###

The assessment battery administered to the child is shown in Table 1. 

```{r table1, echo=FALSE}
#usual problems with getting tables to print in Word. OK with updated pandoc 2.0
#https://davidgohel.github.io/flextable/articles/overview.html
#autofit(regulartable(testbattery))
ft<-regulartable(testbattery)
ft <- width(ft, width = 3)
ft
```
Parents were asked to complete the Children’s Communication Checklist – 2 (CCC-2: Bishop, 2003) and Social Responsiveness Scales (SRS: Constantino, 2005) and return them by mail; the response rate was 86% for the trisomy families and 83% for the twin families.  The SRS data and results from an online interview covering behavioural and psychiatric characteristics will be described in a companion paper. 
Results from the assessments in table 1 were converted to age-scaled scores on a common scale with mean 100 and SD 15. All tests had published norms covering the age range of the twin sample (6 to 11 years), but some did not extend outside this range. As explained in Newbury et al. (2018), where feasible norms were extrapolated based on data from other samples encompassing the wider age range. For oromotor skills, however, extrapolated norms gave scaled scores that were well below the range of other tests. For this test, therefore the No Concerns twin group was used to derive norms, using the regression of total correct on age to compute a standardized residual, which was then scaled to mean 100 and SD 15.  

## Covariates

Two measures of socio-economic variables that have been associated with language status were included in the analysis. These were:  
a)	Educational level of mother, transformed into an ordinal scale based on age at leaving full-time education/qualifications obtained, with points of 0 (prior to age 16 years), 1 (16 years/did GCSE or O-levels), 2 (18 years/ did A-levels), 3 (21 years, degree), 4 (postgraduate study).   

b)	An index of multiple deprivation based on postcode was obtained for those living in England from the website http://imd-by-postcode.opendatacommunities.org/. This uses local statistics from the Department for Communities and Local Government to rank over 32,000 postcodes on the basis of a weighted sum based on income, employment, education, health, crime, housing and living environment. The rank score was converted to a z-score to give a normally-distributed variable, by dividing by 32750 before applying the qnorm function in R.  A value of zero was assigned to 13 SCT cases and 4 twin pairs from Wales, Scotland and Northern Ireland.


## Procedure

Children were seen for an individual assessment at home or school. The language, reading and nonverbal ability tests (Table 1) were given in an initial session lasting around 90 minutes, followed by an assessment of laterality, the results of which are described elsewhere (Wilson & Bishop, 2018).  

## Analysis

Table 2 shows the characteristics of the trisomy and twin groups on background variables: age, mother's educational level and standardized deprivation index.

**Table 2: Mean (SD) for age, mother's educational level and standardized deprivation index for SCT and twin groups**

```{r table2, echo=FALSE}

ft2<-regulartable(mybkgtab2)
ft2 <- width(ft2, width = c(2.2,1,.8,.8,.8))
ft2
```

##Statistical comparisons
For each type of variable (Psychometric tests and parent report on CCC-2) three comparisons were conducted, corresponding to the three research questions, A) Comparison of the three trisomies for Low Bias group only; B) Comparison of all Low Bias trisomy cases aged 6-11 yr with the Language Concerns comparison group; C) Comparison of Low Bias and High Bias trisomy groups.

**Psychometric tests**  
A.	**Comparison of three trisomy groups** (Low Bias group only)  

As a first step, data from the psychometric test variables were plotted to see the range and distribution of scores for the three Low Bias trisomy groups, as shown in the red points in Figure 3. These plots exclude children with missing data on any test. The continuous horizontal bars show the median for each group.  All data have been scaled to mean of 100 and SD 15. The yellow shaded area shows the range covered by mean +/- 1 SD for the No Concerns twins. The distribution of blue points, corresponding to the Language Concerns twins will be discussed below (analysis B).  

An initial impression from these plots is that there is variation from test to test in severity of impairment, but on all measures, the most striking feature is a wide spread of scores, with some children scoring above average and others well below. Although the median test scores vary from trisomy to trisomy, the within-group variation is much greater than between group variation. 

![Figure 3: Distributions of scores on psychometric tests for the Low Bias trisomy groups and the Language Concerns Comparison group. Filled circles show cases in age range 6 to 11 yr. Continous line is mean for 6-11 yr olds, dotted line is mean for whole sample. Yellow band is mean +/- 1 SD for No Concern comparison group.](beeswarm_lang.png)

MANOVA was used to test whether there were reliable differences between the three trisomy groups on the 14 test measures, after taking covariates (Mother's educational level and normalised deprivation index) into account.  No effect of trisomy type was found (see Table 3).  

**Table 3. Results of MANOVA comparisons on psychometric tests**
```{r table3, echo=FALSE}
ft3<-regulartable(manova.df)
ft3 <- width(ft3, width = c(2,1,.8,.8,.8,.8,.8))
ft3

```

B.	**Comparison of combined trisomy cases aged 6-11 yr (Low Bias) with Language Concern comparison group**  
Table 3 also shows the MANOVA result for this comparison, again revealing no overall group effect of group. The only factor affecting scores was mother's education, with each additional point on the scale associated with an increase of `r my.moedregb` points in average score on psychometric tests.

C.	**Comparison of effect of trisomy type and bias group: trisomy cases only**  
For this analysis, we considered two main effects: trisomy type and bias group, as well as the interaction between them. As shown in table 3 also shows the MANOVA result for this comparison, again revealing no overall group effect. The effect of bias group showed a nonsignificant trend for poorer scores to be associated with high ascertainment bias.

![Figure 4: Distributions of scores on psychometric tests for the Low Bias vs High Bias trisomy groups. Filled circles show cases in age range 6 to 11 yr, unfilled circles show older children. Continous lines are group means. Yellow band is mean +/- 1 SD for No Concern comparison group. ](beeswarm_lang_bias.png)

**Children's Communication Checklist-2**  

A.	**Comparison of three trisomy groups** (Low Bias group only)  

Mean scores on the CCC-2 subscales are shown in Figure 5. As expected, the No Concerns comparison group has mean scores close to the population mean of 10. As with the psychometric tests, three MANOVAs were conducted, corresponding to the three research questions.

![Figure 5: Mean scores on the 10 CCC-2 subscales for trisomy and comparison groups. For trisomies, continuous lines show Low Bias group, and dotted lines show High Bias group.](cccplot.png)
**Table 4. Results of MANOVA comparisons on CCC-2 subscales**

```{r table4, echo=FALSE}
ft4<-regulartable(manova.ccc.df)
ft4 <- width(ft4, width = c(2,1,.8,.8,.8,.8,.8))
ft4

```

Categorical analysis - DLD or not
The lack of effect of bias group may seem surprising, given the visual evidence of group differences in some of the plots. An important factor in the statistical outcome is the wide spread of scores in all groups. To demonstrate this, we ran an alternative analysis in which ANOVA was conducted with the number of test scores more than 1 SD below the normative mean as dependent measure. This showed a


???would be worth redoing MANOVA just with the key language measures with data for 5 yr olds??




## Acknowledgements
We offer warmest thanks to the families who took part in the study, and school staff who helped facilitate assessment arrangements. We greatly appreciate the help we had with recruitment from Unique (the Rare Chromosome Disorder Support Group), the Klinefelter’s Syndrome Association, and staff from Clinical Genetics Centres in the following NHS Trusts: Royal Devon and Exeter, The Northwest London Hospitals, University Hospitals Bristol, Sheffield Children’s, Liverpool Women’s, University Hospital Southampton, Birmingham Women’s Hospital, Great Ormond Street Hospital for Children, Leeds Teaching Hospitals, Newcastle upon Tyne Hospitals, Oxford University Hospitals, Central Manchester University Hospitals, University Hospitals of Leicester, Nottingham University Hospitals, Belfast Health and Social Care Trust and the NIHR Clinical Research Network. We extend special thanks to Dr Gillian Borthwick, National Research Coordinator NIHR CRN: Genetics, for helping us navigate the complexities of working with multiple NHS sites under the NIHR UK Rare Genetic Disease Research Consortium Agreement. The study would not have been possible without the generous support of Prof Simon Fisher, Director of the Max Planck Institute for Psycholinguistics, who offered facilities for analysis of genetic material during the data collection phase of this study. We thank Katerina Kucera and Arianna Vino for their substantial work in processing DNA samples.






