---
title             : "Autism spectrum disorder and social anxiety in children with sex chromosome trisomies"
shorttitle        : "ASD and social anxiety in trisomies"
author: 
  - name          : "Alexander Wilson"
    affiliation   : "1"
    corresponding : no    # Define only one corresponding author
    address       : "Department of Experimental Psychology, Anna Watts Building, Woodstock Road, Oxford, UK, OX2 6GG"
    email         : "dorothy.bishop@psy.ox.ac.uk"
  - name          : "Judith King"
    affiliation   : "1"
  - name          : "Dorothy V. M. Bishop"
    affiliation   : "1"


affiliation:
  - id            : "1"
    institution   : "University of Oxford"

author_note: |

keywords          : "Autism spectrum disorder, social anxiety, sex chromosome trisomy, trisomy X, Klinefelter syndrome, XYY syndrome, DAWBA, SRS, SDQ"
wordcount         : "X"



figsintext        : no
figurelist        : no
tablelist         : no
footnotelist      : no
lineno            : yes

lang              : "english"
class             : "man"
output            : papaja::apa6_word
bibliography      :  ["/Users/dorothybishop/endnotehome/!bishoprefs2019.enl"]
always_allow_html: yes
---
```{r set-options, include=FALSE}
rm(list = ls())
knitr::opts_chunk$set(echo = TRUE, cache = FALSE, error = TRUE)
thistab<-0 #counter for tables
thisfig<-0 #counter for figures
```

```{r load_packages, include = FALSE}
#devtools::install_github("crsh/papaja")
library(papaja)
library(tidyverse)
library(knitr) #for kable
library(pander)
library(gridExtra)
library(grid)
library(Hmisc)
library(doBy)
library(beeswarm)
library(DiagrammeR) #Needed for flowcharts
library(DiagrammeRsvg)#Needed for flowcharts
library(magrittr)#Needed for flowcharts
library(svglite)#Needed for flowcharts
library(rsvg)#Needed for flowcharts
library(car)
library(lavaan) #for language factor extraction
library(semPlot)
library(kableExtra)
library(flextable)
library(ggfortify) #see https://www.r-bloggers.com/dimensionality-reduction-for-visualization-and-prediction/
library(cluster)
library(ggplot2)
library(ggridges) #for joyplot
library(mvnormtest) #check assumptions for MANOVA
library(ggpubr)
library(yarrr) #for pirateplot
library(lsr) #for correlations
library(pwr) #for power calculation

dopng <- 0 #toggle for png figures
numnames<-c('no','one','two','three','four','five','six','seven','eight','nine','ten','eleven','twelve')
#numnames used when reporting low Ns in text.
options(scipen=999)
```


```{r, echo=FALSE} 
# Seed for random number generation
#set.seed(42)

```

```{r readredcap, include=FALSE}
#########################################################################################
## Step 1 - this is identical to script for language phenotype paper
#Read in latest SCT from redcap.
#Do the same for the twins. 

#N.B. Original idea was to use twins as comparison group for psychiatric symptoms.
#This is difficult, though, because they have ASD as exclusion, and so are not very comparable.
#For the present, the twin data are still read in and combined in case they may be useful, but the main analysis is just with SCT

#Select variables to be used.

#redcap.dir <-"c:/Users/pthompson/Dropbox/project SCT analysis/Data from Redcap/"
redcap.dir <-"~/Dropbox/ERCadvanced/project SCT analysis/Data from Redcap/"

#Use latest version - this includes consensus DAWBA diagnosis for SCTs
#Automated diagnosis added 30th Nov, with Soc Aptitude Scale data
sct.redcap <- 'SCTData_DATA_2019-01-05_0947.csv'
#updated to include code indicating whether participated in prior study, with prior diagnosis if so
twin.redcap <- 'TwinsData_DATA_2019-01-05_0818.csv' #updated to include ccc2_consistent variable
#also now has pheno_include, which selects individual twins for inclusion
sct.data <- read.csv(paste0(redcap.dir,sct.redcap),stringsAsFactors = FALSE)
twin.data <- read.csv(paste0(redcap.dir,twin.redcap),stringsAsFactors = FALSE)
twin.data <- twin.data[twin.data$pheno_include==1,]

names(sct.data)[1]<-"record_id"
names(twin.data)[1]<-"record_id"
#fix any columns with discrepant names
w<-which(colnames(twin.data)=='neurodev_diagnosis')
colnames(twin.data)[w] <-'neurodev_diag'
# Coded from parental interview only: 0, none, 1, ADHD
# 2, APD (ignored), # 3, ASD, # 4, behav, # 5, dyscalc, # 6, dyslexia
# 7, dyspraxia, # 8, SLI/DLD, # 9, ID

#Nb Fam position: Coded so if you subtract 1 you get N older sibs: disregards birth order of twins
w<-which(colnames(sct.data)=='position_in_family')
colnames(sct.data)[w] <-'fam_position'
w<-which(colnames(sct.data)=='fh_langprobs')
colnames(sct.data)[w] <-'fh_langprob' #1 if 1 aff relative, 2 if more than one
#NB this dld_rd categorisation not currently used
twin.data$dyslexia<-twin.data$dld_rd%%10 #modulus 10 , ie last digit
twin.data$lang_disorder<-round((twin.data$dld_rd-4.5)/10) #first digit is dld code

#add gender info to sct data so it merges properly
sct.data$female<-0
w<-which(sct.data$trisomy==1)
sct.data$female[w]<-1 #XXX cases are female

sct.data$fam_id<-NA #dummy column to ensure merges ok

#only one rater for DAWBA in twins, but need to have columns for the SCTs
twin.data$asd_dsm_r2<-NA
twin.data$autism_icd_r2 <-NA
twin.data$social_anx_dsm_r2<-NA
twin.data$social_anx_icd_r2<-NA
twin.data$cgas_r2<-NA

# Twins recruited in relation to splang_conc, which is coded as
# 0, never
# 1, preschool only
# 2, continuing, mild
# 3, continuing, severe
# 4, reading concerns only
# 9, unclear
# We will be using this categorisation to subdivide twins; categories 2 and 3 distinguished.
# 
sct.data$parent_conc<-NA

twin.data$rater1<-'DB'
twin.data$rater2<-NA
twin.data$parent_conc<-0
w<-which(twin.data$splang_conc%in%c(2,3))
twin.data$parent_conc[w]<-1

#########################################################################################
#----------------------------------------------------------------------------------
```

```{r ccc2009, include=FALSE}
#just checking correspondence between CCC-2 time 1 and time 2 for those seen twice
sct.data$gcc[sct.data$gcc==999]<-NA #convert 999 to missing data code for gcc
cor(sct.data$gcc,sct.data$ccc_gcc_2009,use='complete.obs')
plot(sct.data$gcc,sct.data$ccc_gcc_2009,pch=(sct.data$trisomy-1),col=(1+sct.data$neurodev_diag))
#xxx is square, xxy is circle, xyy is triangle
#I checked the ones with big changes to ensure we have correct cases.
#Nothing all that obvious to explain the little cluster who have declined. Not explained by age


```

```{r colselect, include=FALSE}
#########################################################################################
#Focus on columns to be used in this analysis 
#(includes DAWBA etc, cf. language pheno paper)

#Also now reading in DAWBA bands: Goodman, A., Heiervang, E., Collishaw, S., & Goodman, R. (2010). The 'DAWBA bands' as an ordered-categorical measure of child mental health: description and validation in British and Norwegian samples. Social Psychiatry and Psychiatric Epidemiology, 46, 521-532. doi:10.1007/s00127-010-0219-x

#Specify columns to retain that are common to both sct and twin
#NB we are using oromotor_ss2, which is renormed against the TD twin group
#NB also include SRS items that form short scale

#NB for ASD and Socphobia from DAWBA, we now have a variable called 'asd_dsm_agree',
#and 'socphob_dsm_agree'. This is consensus rating: if R1 and R2 agreed, we used that,
#otherwise DVMB reviewed the DAWBA and made a consensus rating.
#We will use this for main diagnostic analysis.

keepcols <- c('record_id','fam_id','female','age_at_test','parent_conc','partial_testing','pass_hearing','wasi_matrices_ss','wasi_block_design_ss','wasi_vocab_ss','wdck_jhsn_ss',
              'sent_rep_ss','nonword_rep_ss','oromotor_ss_2','nara_acc_ss',      'nara_comp_ss','nara_rate_ss','towre_words_ss','towre_nonwords_ss','phab_pic_ss','phab_digit_ss','srs_t_score','slt','splang_conc','schooling','piq','neurodev_diag','dyslexia','lang_disorder',
              'mo_educ','fa_educ','deprivation_index','fam_structure','fh_langprob',
              'gcc','scdi','ccc_a','ccc_b','ccc_c','ccc_d','ccc_e','ccc_f',
              'ccc_g','ccc_h','ccc_i','ccc_j','ccc2_consistent',
              'srs_item10', 'srs_item14', 'srs_item16', 'srs_item18', 'srs_item24', 
              'srs_item28', 'srs_item29', 'srs_item33', 'srs_item35', 'srs_item37',
              'srs_item44', 'srs_item50', 'srs_item51', 'srs_item56', 'srs_item58', 'srs_item65',
              'socaw_ss','soccomm_ss','soccog_ss','socmot_ss','autfeat_ss','srs_t_score', 
              'asd_dsm_agree','socphob_dsm_agree','rater1','rater2','asd_dsm_r1', 'adhd_comb_dsm_r1', 'adhd_hyp_dsm_r1', 'adhd_inatt_dsm_r1', 'generalized_anx_dsm_r1',
              'social_anx_dsm_r1', 'separation_anx_dsm_r1', 'other_anx_dsm_r1', 'phobia_dsm_r1',
              'depression_dsm_r1', 'opp_def_dsm_r1', 'conduct_dsm_r1', 'chronic_tic_dsm_r1', 
              'tourette_dsm_r1', 'anorexia_dsm_r1', 'autism_icd_r1', 'hyperkinetic_icd_r1',
              'generalized_anx_icd_r1', 'social_anx_icd_r1', 'separation_anx_icd_r1', 'other_anx_icd_r1',
              'phobia_icd_r1', 'minor_depression_icd_r1', 'mod_depression_icd_r1', 'sev_depression_icd_r1',
              'opp_def_icd_r1', 'soc_conduct_icd_r1', 'unsoc_conduct_icd_r1', 'fam_conduct_icd_r1',
              'chronic_tic_icd_r1', 'tourette_icd_r1', 'anorexia_icd_r1',
              'cgas_r1', 'cgas_r2','honosca_disruptive_r1', 
              'honosca_adhd_symptoms_r1', 'honosca_self_injury_r1', 'honosca_sub_misuse_r1',
              'honosca_sholastic_lang_r1', 'honosca_phys_illness_r1', 'honosca_pschotic_r1',
              'honosca_somatic_r1', 'honosca_emotional_r1', 'honosca_peers_r1', 'honosca_selfcare_ind_r1',
              'honosca_family_r1', 'honosca_school_att_r1',
              'dawba_id', 'emotional', 'conduct', 'hyperactivity', 'peer', 'prosocial', 'impact',
              'dawba_sdq_complete', 'sepanx_concern', 'loss_af', 'taken_from_af', 'school_ref', 
              'sleep_alone', 'sleep_af', 'sleep_away', 'room_alone', 'home_alone', 'sep_nightmare',
              'somatic_sep', 'sep_antic_anx', 'sep_symptoms_1mo', 'sep_age_of_onset', 'sep_distress',
              'sep_impact_family', 'sep_impact_friends', 'sep_impact_learning', 'sep_impact_leisure',
              'sep_burden', 'socanx_concerns', 'new_people', 'many_people', 'soc_eating', 'soc_speaking',
              'soc_reading', 'soc_writing', 'soc_vs_sep', 'adults_kids', 'can_socialise', 'cause_fear',
              'cause_litprobs', 'socanx_age_onset', 'socanx_dur_mo', 'socfear_level', 'socfear_freq',
              'avoid_soc', 'socanx_impact', 'fear_excessive', 'upset_socfear', 'socanx_burden',
              'generous', 'lively', 'keen2learn', 'affectionate', 'reliable', 'easygoing', 'fun', 'interested',
              'caring', 'bounces', 'grateful', 'indepdt', 'helps', 'gets_on', 'homework', 'creative', 'family',
              'appearance', 'good_school', 'polite', 'good_sport', 'tidy', 'good_friends', 'good_behav',
              'sas_laugh', 'sas_chat', 'sas_flexible', 'sas_defuse', 'sas_lose', 'sas_ease', 'sas_tom',
              'sas_apolog', 'sas_leader', 'sas_socaware', 'sas_total', 'dawba_positive_complete',
              'makefriends', 'keepfriends', 'nfriends', 'share_interest', 'joint', 'confide', 'gen_reason',
              'mentalage', 'langlevel', 'langage', 'langcope', 'speech3yr', 'soc3yr', 'play3yr', 'ritual3yr',
              'ment3yr', 'continued_diffs', 'words2yr', 'phrase3yr', 'toddlergesture', 'toddlergames',
              'shareenjoy', 'repetitive', 'unusual', 'imagplay', 'adjustsplay', 'turntaking', 'obsession',
              'topic', 'dominates', 'dominateconv', 'interferes', 'goodconv', 'goodsustain', 'chats',
              'adjustsconv', 'nonvbl_blank', 'can_read_nv', 'abnormal_eye', 'echoing', 'repquest', 
              'repcliche', 'strongroutine', 'changeroutine', 'flapping', 'parconcern', 'distress',
              'impactfam', 'impactfriend', 'impactlearn', 'impactleisur', 'asd_burden',
              'suddenonset', 'ageonset','asd_dsm_r2','autism_icd_r2','social_anx_dsm_r2',
              'social_anx_icd_r2',
              'sepaband',  'spphband', 'sophband',  'panband', 'agoband',  'ptsdband', 'ocdband',  'genaband', 'depband',  'adhdbandd', 'adhdtype',  'oddband', 'cdband',  'eatband',  'ticbandd',  'asdband')


twin.short <- select(twin.data,keepcols)
sct.short <- select(sct.data,keepcols) #NB 'select' is ambiguous between packages but should work ok if we use formr (otherwise need dplyr::select)

#########################################################################################

```

```{r combinefiles, include=FALSE}
###################################################################################################
#Add columns that are specific to twins or SCTs so all aligned
#Select data for short file


sct.short$is_twin <- 0 #this specifies child is not a twin but a trisomy
sct.short$zygosity <- NA
twin.short$is_twin <- twin.data$randomtwininclude+1 #twins subdivided
#randomly into 1 and 2 for replication sample


twin.short$zygosity <- twin.data$zygosity
#at current time have only one rater for twins, so treat that as consensus
twin.short$asd_dsm_agree<-twin.short$asd_dsm_r1
twin.short$socphob_dsm_agree<-twin.short$social_anx_dsm_r1

sct.short$trisomy <- sct.data$trisomy
sct.short$pre_postnatal_diag <- sct.data$pre_postnatal_diag
sct.short$why_tested <-sct.data$why_tested
sct.short$orig_famcode<-0
sct.short$orig_famcode[sct.data$orig_famcode>0]<-1 #if seen before this is 1
twin.short$trisomy <- 0
twin.short$pre_postnatal_diag <- NA
twin.short$why_tested <- NA
twin.short$orig_famcode<-NA
twin.short$splang_conc<-twin.data$splang_conc
sct.short$splang_conc<-NA
sct.short$pheno_include<-1
twin.short$pheno_include<-twin.data$pheno_include #this codes reason for exclusion as:
#2 ASD, 3 ASD-sib, 4-hearing, 5- other
#exclude twins with DSM ASD diagnosis: nb in this analysis ASD case removed for twins only
w<-which(twin.short$asd_dsm_r1==1)
twin.short$pheno_include[w]<-2
twin.short<-twin.short[twin.short$pheno_include==1,] #removes those with ASD, ASD-sib,hearing, other
#remove one twin at random
w <- which(twin.short$is_twin==1)
twin.short<-twin.short[w,]
#----------------------------------------------------------------------------------
#Bolt SCT and twin files together
all.short <-rbind(sct.short,twin.short)

#remove case of isochromosome
w<-which(all.short$trisomy==9)
all.short<-all.short[-w,]



#rename age variable
w<-which(colnames(all.short)=='age_at_test')
colnames(all.short)[w]<-'Age'

#add a column identifying age range: 1 = <6, 2 = 6-11;11, 3 = 12+
all.short$ageband<-2
w<-which(all.short$Age<72)
all.short$ageband[w]<-1

w<-which(all.short$Age>143)
all.short$ageband[w]<-3

#age band for SDQ norms makes distinction between 5-10 yr and 11-16
all.short$ageSDQ<-2
w<-which(all.short$Age<132)
all.short$ageSDQ[w]<-1

```

```{r sdqnorms, include=FALSE}
#create norms for SDQ and SAS
#we have centiles by age band and sex for sdq
#we can turn these into T scores, so same scale as SAS
#But for comparability with SRS, will make high score indicate problems
#So need to switch the SAS scores!
sdqnorm<- read.csv('dawba diagnosis/sdqcentiles_longform_forR.csv')
sasnorm<- read.csv('dawba diagnosis/sas_norms.csv')
#sas norms are T-scores: raw scores seem reasonably normal
#See Liddle2008 article for more info - these norms from
#dawba.info/SAS/psastscore.pdf



sdqscales<-c('emotional','conduct','hyperactivity','peer','prosocial','sas_total')
sdqscalesT<-paste0(sdqscales,'.T')

addsdq<-data.frame(matrix(NA,nrow=nrow(all.short),ncol=6))
colnames(addsdq)<-sdqscalesT
colnames(addsdq)[6]<-'socapt.T'

  for (thisrow in 1:nrow(all.short)){
    thissex<-all.short$female[thisrow]
    thisage<-all.short$ageSDQ[thisrow]
    for (subscale in 1:5){
  thiscol<-which(colnames(all.short)==sdqscales[subscale])
   thisscore<-all.short[thisrow,thiscol]
    if(!is.na(thisscore)){
    myfilt<-filter(sdqnorm,female==thissex,ageband==thisage,scale==sdqscales[subscale],raw==thisscore)
    mycumprob<-myfilt$cumulative/100
    if(subscale==5){mycumprob<-1-mycumprob} #switch direction for prosocial
    if(mycumprob==0){mycumprob<-.01}
    if(mycumprob==1){mycumprob<-.99}
    addsdq[thisrow,subscale]<-round(50+10*qnorm(mycumprob),0)
    }
    }
    #now add SAS t scores from table
      thiscol<-which(colnames(all.short)=='sas_total')
   sasrow<-all.short[thisrow,thiscol]+1
   myage<-as.integer(all.short$Age[thisrow]/12)
   readcol<-myage-3
   addsdq[thisrow,6]<-100-sasnorm[sasrow,readcol]
   #subtract from 100 to invert the scale so high is bad

}

all.short<-cbind(all.short,addsdq)


```

```{r missingcheck, include=FALSE}
#----------------------------------------------------------------------------------
# These next chunks are taken direct from the lang-pheno script

##Step 2
#Deal with missing data
#Codes are

#999 – test not given
#998 – result not valid (maladministration)
#997 – result not valid (child refusal etc.)
#996 – child too old/young for norms

#Check for nonword rep - in some cases OK to substitute low score, if child unable to attempt task. 

all.short$pass_hearing[all.short$pass_hearing==9]<-NA #change 9 to NA for pass_hearing
#NB these include both refusal and equipment failure

all.short$piq[all.short$piq==999]<-NA

#missing data has codes of 996-999 for language tests
w <- which(all.short$nonword_rep_ss>900)
print('Cases with missing nonword rep data: ')
all.short$record_id[w]
#These cases all checked : all SCT cases who either were not tested because v limited spoken language, or who refused spoken tests or had low scores on other language tests. All these cases assigned a scaled score of 3.
all.short$nonword_rep_ss[w] <- 3

#999 is code where child was not tested bcs too low-functioning
w <- which(all.short$wasi_vocab_ss==999)
print('Cases with missing Vocab data (999) reassigned to floor: ')
all.short$record_id[w]
#These assigned SS of 2.33 SD below mean (equivalent to 3 on NEPSY scale)
all.short$wasi_vocab_ss[w] <- 25

w <- which(all.short$wdck_jhsn_ss==999)
print('Cases with missing Woodcock_J data (999) reassigned to floor: ')
all.short$record_id[w]
#These assigned SS of 2.33 SD below mean (equivalent to 3 on NEPSY scale)
all.short$wdck_jhsn_ss[w] <- 55

w <- which(all.short$sent_rep_ss==999)
print('Cases with missing Sent rep data (999) reassigned to floor:: ')
all.short$record_id[w]
#These assigned SS of 2.33 SD below mean (equivalent to 3 on NEPSY scale)
all.short$sent_rep_ss[w] <- 3

w <- which(all.short$oromotor_ss_2==999)
print('Cases with missing oromotor data (999) reassigned to floor:: ')
all.short$record_id[w]
#These assigned SS of 55 , as this corresponds to floor in this sample
all.short$oromotor_ss_2[w] <- 55

#Now count how many tests have missing data
mycols<-colnames(all.short)
#Identify the range of columns where 900+ is missing data code
mc <-which(mycols=='wasi_matrices_ss')
mc0<-which(mycols=='nonword_rep_ss') #end of range for spoken language
mc1<-which(mycols=='phab_digit_ss') #end of range of reading related
mc2<-which(mycols=='ccc_a') #end of range of reading related
mc3<-which(mycols=='ccc_j') #end of range of reading related

all.short$n_missinglang<-0
for(i in 1:nrow(all.short)){
  mytests<-all.short[i,mc:mc0]
  n.na<- length(which(is.na(mytests)))
  n9<- length(which(mytests>900))
  all.short$n_missinglang[i]<-(n.na+n9)
  
}
table(all.short$n_missinglang,all.short$ageband)

all.short$n_missingread<-0
for(i in 1:nrow(all.short)){
  mytests<-all.short[i,(mc0+1):mc1]
  n.na<- length(which(is.na(mytests)))
  n9<- length(which(mytests>900))
  all.short$n_missingread[i]<-(n.na+n9)
  
}
table(all.short$n_missingread,all.short$ageband)


#Put in NA for missing data codes all lang/read/CCC-2
for (i in c(mc:mc1,mc2:mc3)){
  w <- which(all.short[,i]>900)
  all.short[w,i]<-NA
}

n.col<-length(colnames(all.short)) #define n.col here

#Parental educ variables, missing is 9
w<-which(all.short$mo_educ==9)
all.short$mo_educ[w]<-NA
w<-which(all.short$fa_educ==9)
all.short$fa_educ[w]<-NA
w<-which(all.short$fh_langprob==9)
all.short$fh_langprob[w]<-NA

```

```{r langfactor, include=FALSE}
#Compute language factor, using script from Appendix 2 from Newbury et al 2018
#Aim: to get a single robust measure of language ability 


model.f5a <- ' f1 =~ wasi_vocab_ss + wdck_jhsn_ss + sent_rep_ss + oromotor_ss_2 
              wasi_vocab_ss ~~ wdck_jhsn_ss'     
fit.mod.E2 <- cfa(model.f5a, data = all.short,estimator = "ML",missing = "ML")
lbls<-c("Vocabulary", "Woodcock\nJohnson", "Sentence\nRepetition", "Oromotor","Language")
#semPaths(fit.mod.E2, "std", title = TRUE, curvePivot = TRUE, edge.label.cex = 1.2,width=10,height=5,nodeLabels=lbls,intercepts = FALSE,sizeMan = 10, sizeLat = 10)
all.short$langfactor <- as.vector(predict(fit.mod.E2)) #factor scores
```

```{r rescalelang, include=FALSE}
#Rescale all variables to mean 100 and SD 15 and save in new columns
#This is taken from language phenotypes Rmd script; probably not needed here
short.langnames<-c('Matrices','Blocks','Vocab','Comprehnsn','SentRep','NwdRep','Oromotor',
                   'ReadAcc','ReadComp','ReadRate','TOWREwd','TOWREnwd',
                   'PicName','DigitName')

#convert all variables to same scale: mean 100 and SD 15
#This has to be hard coded because different tests on different scales
langcog.cols<-c("wasi_matrices_ss" ,"wasi_block_design_ss", "wasi_vocab_ss" ,
                "wdck_jhsn_ss" , "sent_rep_ss" , "nonword_rep_ss",  "oromotor_ss_2",
                "nara_acc_ss" ,"nara_comp_ss","nara_rate_ss" ,"towre_words_ss",
                "towre_nonwords_ss","phab_pic_ss","phab_digit_ss")
#Note we now are using the oromotor_ss_2 variable: normed against TD twins

lang.cols<-which(colnames(all.short)%in%langcog.cols)
#NB this finds columns corresponding to langcog.cols, but they will be in the order they occur in file
#start by just duplicating original scaled scores
n.new<-length(short.langnames)
n.col<-dim(all.short)[2] #find final column: will append new ones after this
nurange<-(n.col+1):(n.col+n.new) #column range for ss values, NB n.col defined in prev block
all.short[,nurange]<-all.short[,lang.cols]
colnames(all.short)[nurange]<-short.langnames
#First 3 are normed with mean 50 and SD 10
for (i in 1:3){
  oldmean<-50
  oldsd<-10
  all.short[,(i+n.col)]<- 100+15*(all.short[,lang.cols[i]]-oldmean)/oldsd
}

for (i in c(5,6)){
  oldmean<-10
  oldsd<-3
  all.short[,(i+n.col)]<- 100+15*(all.short[,lang.cols[i]]-oldmean)/oldsd
}


```

```{r interpolatemissing, include=FALSE}
#Again, this taken from language phenotypes Rmd.
#If just one value is missing in lang or reading clusters, substitute the mean of others
langrange<-(n.col+1):(n.col+7)
readrange<-(n.col+8):(n.col+13)
#Start with language range
w<-which(all.short$n_missinglang==1) #find those rows with just one missing value
for(s in 1:length(w)){
  thisrow<-w[s] #row with missing value
  v<-which(is.na(all.short[thisrow,langrange])) #which value is missing?
  mycol<-langrange[v] #column with missing data
  all.short[thisrow,langrange[v]]<-round(rowMeans(all.short[thisrow,langrange],na.rm=TRUE),0) #mean of other values substituted
}

print(paste0('Mean substituted for cases with one missing datapoint: Interpolated ', length(w),' missing values from ',nrow(all.short)*length(langrange),
             ' datapoints in ',length(langrange),' nonverbal and language tests'))

w<-which(all.short$n_missingread==1) #find those rows with just one missing value
for(s in 1:length(w)){
  thisrow<-w[s] #row with missing value
  v<-which(is.na(all.short[thisrow,readrange])) #which value is missing?
  mycol<-readrange[v]
  all.short[thisrow,readrange[v]]<-round(rowMeans(all.short[thisrow,readrange],na.rm=TRUE),0) #mean of other values substituted
}
print(paste0('Mean substituted for cases with one missing datapoint: Interpolated ', length(w),' missing values from ',nrow(all.short)*length(readrange),
             ' datapoints in ',length(readrange),' reading and rapidnaming tests'))


#create an averaged literacy measure to avoid floor effects: this is more normal
all.short$literacy<-rowMeans(all.short[,(n.col+8):(n.col+14)])
all.short$genlang<-rowMeans(all.short[,(n.col+3):(n.col+4)])
all.short$verbmem<-rowMeans(all.short[,(n.col+5):(n.col+7)])
#create an averaged verbalmem measure 
```


```{r ncheck, include=FALSE}
## Step 3

# #Check that Ns match those in the protocol
# Should have three groups with N = 142 SCT, 194 twin1 and 194 twin2 (these are randomly assigned twin 1 and 2), - but Ns less for twins because of exclusions
# Numbers are greater than for Newbury et al, as some children did not feature in that study because of lack of DNA.
# 
# Twins recruited in relation to splang_conc, which is coded as
# 0, never
# 1, preschool only
# 2, continuing, mild
# 3, continuing, severe
# 4, reading concerns only
# 9, unclear
# We will be using this categorisation to subdivide twins; categories 2 and 3 distinguished.
# 
# Trisomies - divided according to reason for diagnosis
# why tested used to make bias column: coded as: 
# 0, maternal age
# 1, medical concerns
# 2, behavioural concerns
# 3, neurodevelopmental concerns
# 4, family history of genetic problems
# 9, no information
###################################################################################
twin.tab <- table(all.short$is_twin)
names(twin.tab)<-c('SCT','twin')

all.short$bias<-NA
w<-which(all.short$is_twin==0)
all.short$bias[w]<-0
w<-c(which(all.short$why_tested==2),which(all.short$why_tested==3))
all.short$bias[w]<-1
sct.tab <- table(all.short$trisomy,all.short$bias)
rownames(sct.tab)<-c('twin','XXX','XXY','XYY')
colnames(sct.tab)<-c('Low Bias','High Bias')
#Show tables
twin.tab
tab_sct<-sct.tab
sct.tab
rowSums(sct.tab)

all.short$parent_conc<-NA #parental concern re language
w<-which(all.short$is_twin>0)
all.short$parent_conc[w]<-0
w<-which(all.short$splang_conc%in%c(2,3))
all.short$parent_conc[w]<-1

parcon.tab<-table(all.short$parent_conc,all.short$is_twin)
rownames(parcon.tab)<-c('No concern','Concern')
colnames(parcon.tab)<-c('null','twin')
parcon.tab[,2]
print('Age range (min, max, average) for SCT')
min(sct.data$age_at_test)
max(sct.data$age_at_test)
mean(sct.data$age_at_test)
print('Age range (min, max, average) for twin')
min(twin.data$age_at_test)
max(twin.data$age_at_test)
mean(twin.data$age_at_test)



```



```{r groupcreate, include=FALSE}
##Create 8 groups:
# 1. XXX-nobias
# 2. XXY-nobias
# 3. XYY-nobias
# 4. XXX-hibias
# 5. XXY-hibias
# 6. XYY-hibias
# 7. twin_noconcern
# 8. twin_concern

#####################################################################################
#first code SLT into 0, 1 or 2;  2 indicates SALT beyond 4 yr, 1 is assessment or preschool only
all.short$slt.code<-0
w<-c(which(all.short$slt==1),which(all.short$slt==8))
all.short$slt.code[w]<-1
w<-c(which(all.short$slt==2),which(all.short$slt==3))
all.short$slt.code[w]<-2


all.short$group8<-all.short$trisomy
w<-which(all.short$bias==1)
all.short$group8[w]<-all.short$group8[w]+3 #convert 1 2 3 to 4 5 6
w<-which(all.short$is_twin>0) #just focus on twins
all.short$group8[w]<-7
w<-which(all.short$parent_conc==1) #coded as NA for SCTs, so only twins coded here
all.short$group8[w]<-8

#add twins with SLT code 2 to group 8, even if no parental concern
w<-intersect(which(all.short$group8==7),which(all.short$slt.code==2))
all.short$group8[w]<-8


#Create factor for overall group
all.short$group8level<-all.short$group8#useful to retain a column with numeric code before converting to factor (converting factor back to number is a nightmare!)
all.short$group8<-as.factor(all.short$group8)
levels(all.short$group8)<-c('XXX.lowbias','XXY.lowbias','XYY.lowbias',
                            'XXX.hibias','XXY.hibias','XYY.hibias',
                            'Twin: No Concerns','Twin: Lang concerns')


```


```{r get.covariates, include=FALSE}
## Again - this is from language phenotypes rmd.
##Covariates
# a)	Educational level of mother and father, transformed into an ordinal scale based on age at leaving full-time education/qualifications obtained, with points of 0 (prior to age 16 years), 1 (16 years/did GCSE or O-levels), 2 (18 years/ did A-levels), 3 (21 years, degree), 4 (postgraduate study). 
# 
# b)	An index of multiple deprivation based on postcode was obtained for those living in England from the website http://imd-by-postcode.opendatacommunities.org/. This uses local statistics from the Department for Communities and Local Government to rank 32,844 postcodes on the basis of a weighted sum based on income, employment, education, health, crime, housing and living environment. The rank score was converted to a z-score to give a normally-distributed variable.  
# 
# c)	Living with single parent, coded as a binary variable (0 or 1), based on status at the time of the initial interview.
# 
# d)	Number of older siblings. For twins, the co-twin was not counted.
# 
# e) Family history of language problems: added after reading Boada et al, who suggest this could explain heterogeneity.

# N.B. The finding that the lowbias SCT children are more likely to have older sibs is probably related to the fact that pre-natal testing is associated with older parents. We do have data on parental ages and could look at that, but not sure it's necessary. It does, though, make me concerned that there may be problems in using N older sibs as a covariate, as it could look causal when it is not. It might be reasonable to use it just within the twin group.

#Parental educational level
par.ed <-aggregate(all.short$mo_educ, by=list(all.short$group8),
                   FUN=mean, na.rm=TRUE)


#Use ranks to convert to normal distribution
#Rw deprivation index is a rank.

depmax<-32844 #N possible ranks

w<-which(is.na(all.short$deprivation_index))
all.short$deprivation_index[w]<-depmax/2
#Assign avg score to those outside england where no dep index
all.short$neighborz<-qnorm(all.short$deprivation_index/depmax)

firsttwin<-min(which(all.short$is_twin==TRUE))
print(paste('N SCT with missing dep index (assigned 0) = ',length(which(w<firsttwin))))
print(paste('N twin with missing dep index (assigned 0) = ',length(which(w>firsttwin))))
all.short$neighborz[w]<-0
dopirate<-0 #toggle to 1 to see pirateplot (not really needed here, but useful as eg of plot labeling)
all.short$singlepar<-0
all.short$singlepar[all.short$fam_structure<3]<-1
singlepar.tab<-table(all.short$group8,all.short$singlepar)
colnames(singlepar.tab)<-c('2 parents','single parent')
prop.table(singlepar.tab,1)


```



```{r dawba_select_data, include=FALSE}

#code those without dawba data
w<-which(is.na(all.short$asd_dsm_r1 )) #If DAWBA done, will have ASD coded
all.short$withdawba<-1
all.short$withdawba[w]<-0
my.dawba1<-all.short[-w,] #short file just has those who did DAWBA

#check if no dawba group is different on language level
#using vocab
temp<-filter(all.short,group8level<7)
temp$withdawba<--1*temp$withdawba #negative;just to make CI print out more legibly in this direction
#check if no dawba group is different on language level
#using vocabulary from WASI
w<-which(temp$wasi_vocab_ss>900)
temp$wasi_vocab_ss[w]<-NA
my.t.dawba <- t.test(temp$wasi_vocab_ss~temp$withdawba)
mysd.dawba<-aggregate(temp$wasi_vocab_ss, by=list(temp$withdawba),
  FUN=sd, na.rm=TRUE)
sd.dawba0 <-round(mysd.dawba[1,2],2)
sd.dawba1 <-round(mysd.dawba[2,2],2)
#compute SRS brief - nb items have been re-ordered so this selects those in SRS brief
w1<-which(colnames(my.dawba1)=='srs_item10')
w2<-which(colnames(my.dawba1)=='srs_item65')
my.dawba1$srsbrief <-rowSums(my.dawba1[,w1:w2])

# Neurodev diag is coded from parental interview only: 0, none, 1, ADHD
# 2, APD (ignored), # 3, ASD, # 4, behav, # 5, dyscalc, # 6, dyslexia
# 7, dyspraxia, # 8, SLI/DLD, # 9, ID
#find those where 3 included in the code (ie community diagnosis of ASD)
temp<-as.character(my.dawba1$neurodev_diag)
w<-grepl('3', temp)
my.dawba1$asd<-0
my.dawba1$asd[w]<-1

#compute yes/no score for presence of emotional disorders or conduct disorders
my.dawba1$dawba_emot <-my.dawba1$generalized_anx_dsm_r1
+my.dawba1$social_anx_dsm_r1
+my.dawba1$separation_anx_dsm_r1
+my.dawba1$other_anx_dsm_r1
+my.dawba1$phobia_dsm_r1
+my.dawba1$depression_dsm_r1
w<-which(my.dawba1$dawba_emot>1)
my.dawba1$dawba_emot[w]<-1

my.dawba1$dawba_conduct <-my.dawba1$opp_def_dsm_r1+my.dawba1$conduct_dsm_r1
w<-which(my.dawba1$dawba_conduct>1)
my.dawba1$dawba_conduct[w]<-1

#compute yes/no score for presence of ADHD
my.dawba1$adhd <-my.dawba1$adhd_comb_dsm_r1+my.dawba1$adhd_hyp_dsm_r1+my.dawba1$adhd_inatt_dsm_r1
w<-which(my.dawba1$adhd>1)
my.dawba1$adhd[w]<-1

#compute yes/no score for presence of ASD and ASD- (here called SCD)
my.dawba1$asd <-my.dawba1$asd_dsm_agree #initialise _ this includes values 1 and 4 for ASD-
my.dawba1$scd=0

w<-c(which(my.dawba1$asd==1),which(my.dawba1$asd==4)) #find the ASD- cases
my.dawba1$scd[w]<-1 #assign ASD- cases to 1
my.dawba1$asd[w]<-0 #assign ASD- cases to 0 for the ASD column
w<-which(my.dawba1$asd==2) #ASD coded as 2 - needs recoding to 1
my.dawba1$asd[w]<-1

#create a 5 digit code for ASD/ASD-/ADHD/Emot/Conduct
my.dawba1$alldsm <-10000*my.dawba1$asd+
                    1000*my.dawba1$scd+
                     100*my.dawba1$adhd+
                      10*my.dawba1$dawba_emot+
                         my.dawba1$dawba_conduct

my.dawba1$alldsm_factor <-as.factor(my.dawba1$alldsm)
levels(my.dawba1$alldsm_factor)<- c('None','Conduct','Emotion','Conduct+Emotion',
                                    'ADHD','ADHD+Conduct','ADHD+Emotion',
                                    'ADHD+Conduct+Emotion','ASD-','ASD-+ADHD',
                                    'ASD-+ADHD+Conduct','ASD-+ADHD+Emotion',
                                    'ASD','ASD+Conduct','ASD+Emotion',
                                    'ASD+ADHD','ASD+ADHD+Conduct',
                                    'ASD+ADHD+Emotion','ASD+ADHD+Emotion+Conduct'
                                    )

#get ranges for SRS
my.dawba1$srs_range <-NA
w<-which(my.dawba1$srs_t_score<91)
my.dawba1$srs_range[w]<-3
w<-which(my.dawba1$srs_t_score<75)
my.dawba1$srs_range[w]<-2
w<-which(my.dawba1$srs_t_score<61)
my.dawba1$srs_range[w]<-1

w<-which(my.dawba1$group8level==8)
my.dawba1$bias[w]<-999
diagtableSRS<-table(my.dawba1$alldsm_factor,my.dawba1$srs_range,my.dawba1$is_twin)


my.dawba<-select(my.dawba1,record_id,Age,female,trisomy,orig_famcode,bias,why_tested,asd,
                 srs_t_score,socaw_ss,soccog_ss,
                 soccomm_ss,socmot_ss,autfeat_ss,srsbrief,
                 dawba_emot,dawba_conduct,
                 asd_dsm_agree,socphob_dsm_agree,
                 cgas_r1,cgas_r2,rater1,rater2, asd_dsm_r1,autism_icd_r1,social_anx_dsm_r1,
                 social_anx_icd_r1,asd_dsm_r2,autism_icd_r2,social_anx_dsm_r2,
                 social_anx_icd_r2,emotional.T,conduct.T,hyperactivity.T,peer.T,prosocial.T,socapt.T,
                 sepanx_concern,socanx_concerns,soc_vs_sep,socfear_level,
                 group8,group8level,ageband,ageSDQ,langfactor,alldsm,alldsm_factor,
                 sepaband,  spphband, sophband,  panband, agoband,  ptsdband, ocdband,  genaband, depband,  adhdbandd, adhdtype,  oddband, cdband,  eatband,  ticbandd,  asdband)

#group of 5 from 'emotional' etc are SDQ scales, added Nov 2017; also social anxiety and sep anxiety added. Sas_total is social awareness scale
#These now converted to T-scores from Goodman norms - see above
my.dawba$Group <- my.dawba$group8 #for compatibility with later routines
```

```{r getages, include=FALSE}
#Add information about mean age for each group, as well as % in ageSDQ bands
mymeanages <- aggregate(my.dawba$Age, by=list(my.dawba$group8level),
  FUN=mean, na.rm=TRUE)
mysdages <- aggregate(my.dawba$Age, by=list(my.dawba$group8level),
  FUN=sd, na.rm=TRUE)
agetab<-table(my.dawba$group8,my.dawba$ageSDQ)

mymeanage.bias <- aggregate(my.dawba$Age, by=list(my.dawba$bias),
  FUN=mean, na.rm=TRUE)
mysdage.bias <- aggregate(my.dawba$Age, by=list(my.dawba$bias),
  FUN=sd, na.rm=TRUE)

```
```{r in.prev.study, include=FALSE}
#count how many SCT cases were in the earlier 2009 study (Bishop et al 2011 paper)
print('')
print('In 2009 study')
w<-which(my.dawba$orig_famcode==1)
Norig<-length(w)
origstudy<-table(my.dawba$bias[w],my.dawba$trisomy[w])
colnames(origstudy)<-c('XXX','XXY','XYY')
rownames(origstudy)<-c('Low Bias','High Bias')


```




```{r socanx, include=FALSE}
#Start with DAWBA diagnoses: focus on ASD and social anxiety; use consensus diagnoses
#ASD is coded 2 for ASD and 1 for partial symptoms (eg SCD)
#SocPhobia is coded 2 for definite and 1 for unsure.
#Unsure for soc phobia is rare, and so recoded here to 0
w<-which(my.dawba$socphob_dsm_agree==1)
my.dawba$socphob_dsm_agree[w]<-0
w<-which(my.dawba$asd_dsm_agree==4) #code used for Asperger (old def) = ASD-
my.dawba$asd_dsm_agree[w]<-1
my.dawba$bicode<- 10*my.dawba$asd_dsm_agree+my.dawba$socphob_dsm_agree
my.dawba$bicode2<-as.factor(my.dawba$bicode)

levels(my.dawba$bicode2)  <- c('Neither','Social Phobia','ASD-','ASD- +Social Phobia','ASD','ASD+Social Phobia')                                                      
mytab<-table(my.dawba$bicode2,my.dawba$group8) 
mytab<-mytab[,1:6] #omit twins
myprop<-prop.table(mytab,2) #proportions by column (2nd parameter denotes row =1 or col=2)
# mytab<-as.data.frame.matrix(mytab)
# myprop<-as.data.frame.matrix(myprop)
nicetab <- data.frame(matrix(NA,nrow=8,ncol=7))
#can't have blank colnames, so using dots for now. Each must be unique
colnames(nicetab)<-c('.','-','Low Bias','..','...','High Bias','....')
nicetab[1,] <- c('Diagnosis','XXX','XXY','XYY','XXX','XXY','XYY')
nicetab[2,]<-c('N',colSums(mytab))
myorder<-c(1,4,2,5,3,6) #reorder the rows to be more logical
for (thisrow in 1:6){
  for (thiscol in 2:7){
    nicetab[(2+myorder[thisrow]),thiscol]<-paste0(mytab[thisrow,(thiscol-1)],' (',
                                                  round(100*myprop[thisrow,(thiscol-1)],1),'%)')
  }
}
nicetab[3:8,1]<-c('Neither','ASD-','ASD','Social Phobia only','Social Phobia+ASD-','Social Phobia+ASD')
#rate of Social Phobia
forchi<-matrix(c(22,21,31,0,7,1),nrow=2,byrow=TRUE)
chisq.test(forchi)

#check if presence of diagnosis influences re-participation
sct.only<-filter(my.dawba,group8level<7) #removetwins
byprevious<-table(sct.only$bicode2,sct.only$orig_famcode)
mat.for.chi<-matrix(c(byprevious[1,1],byprevious[1,2],
                      sum(byprevious[2:5,1]),sum(byprevious[2:5,2])),nrow=2,byrow=TRUE)
chisq.test(mat.for.chi)


```



```{r make.oddsratio, include=FALSE}
do.OR <- function(MeltzN,Meltzp,sctN,sctaff){

Meltzaff <- round(MeltzN*Meltzp/100,0)
Meltznon <-MeltzN-Meltzaff
odds1 <-Meltzaff/Meltznon

odds2 <-sctaff/(sctN-sctaff)
oddsr <-odds2/odds1
logoddsr <-log(oddsr)
se.logoddsr <-sqrt(1/Meltzaff+1/Meltznon+1/sctaff+1/(sctN-sctaff))
ci.low <-round(logoddsr-1.96*se.logoddsr,2)
ci.hi <- round(logoddsr+1.96*se.logoddsr,2)
return(c(oddsr,logoddsr,ci.low,ci.hi))
}
```

```{r dawbabands, include=FALSE}
bandtable <- table(sct.only$sophband,sct.only$group8level)
#Look for evidence of milder social problems using DAWBA bands
bandprop<- prop.table(bandtable,2)
goodmanband <- c(6514,996,208,41,18)
 
#goodmanband <-c(4846,2701,146,84) sep anx

 #Ns with band scores 0, 1, 2 3, 4 in British parent report epi sample
 #From supplementary material to Goodman et al 2011 DAWBA band paper
 #Downloaded from https://link.springer.com/article/10.1007%2Fs00127-010-0219-x
 
 goodmanp <-goodmanband/sum(goodmanband)
 
nicebandtab<-data.frame(matrix(NA, ncol=8, nrow=nrow(bandtable))) 
colnames(nicebandtab)<-c('Band','B-CAMHS','XXX-low','XXY-low','XYY-low',
                         'XXX-high','XXY-high','XYY-high')
nicebandtab[,1]<-rownames(bandtable)
nicebandtab[,2]<-paste0(goodmanband,' (',round(100*goodmanp,1),'%)')
for(i in 1:6){
  nicebandtab[,(i+2)]<-paste0(bandtable[,i],' (',round(100*bandprop[,i],1),'%)')
}
 
lorisk <-filter(sct.only,group8level<4) #analysis applies only to lorisk cases

kband<-kruskal.test(lorisk$sophband~lorisk$group8level) 
#oddsratios for band 2 or over in the low risk group
sophbandOR<-data.frame(matrix(NA, nrow=4,ncol=3)) #initialise
colnames(sophbandOR)<-c('XXX-low','XXY-low','XYY-low')
rownames(sophbandOR)<-c('OR','logOR','low95CIlog','high95CIlog')
MeltzN<-sum(goodmanband)
Meltzp<-sum(goodmanband[3:nrow(bandtable)])/MeltzN

for (sct in 1:3){
  sctN<-sum(bandtable[,sct])
  sctaff<-sum(bandtable[3:nrow(bandtable),sct])
  sophbandOR[,sct]<-round(do.OR(MeltzN,Meltzp,sctN,sctaff),1)
  
}

```

```{r diagnoses, include=FALSE}
#Consider whether prevalence of disorder is above population levels for DAWBA diagnoses, using Table 4.1 of Meltzer and Gatward

mymeltz <- read.csv('dawba diagnosis/meltzer_prevalence.csv',stringsAsFactors=FALSE)
mymt<-table(lorisk$ageSDQ,lorisk$group8level)
nboy_young<-mymt[1,2]+mymt[1,3]
ngirl_young<-mymt[1,1]
nboy_old<-mymt[2,2]+mymt[2,3]
ngirl_old<-mymt[2,1]

meltzdat<-filter(all.short,group8level<4)
meltzrows<-c(21,10,5,3,11,4,12,16,19,18,17,26,27)
meltzcolnames<-c("hyperkinetic_icd_r1","generalized_anx_icd_r1","social_anx_icd_r1",
             "separation_anx_icd_r1","other_anx_icd_r1","phobia_icd_r1", "mod_depression_icd_r1" ,                    "opp_def_icd_r1","soc_conduct_icd_r1",
             "unsoc_conduct_icd_r1" ,"fam_conduct_icd_r1" ,"chronic_tic_icd_r1", "anorexia_icd_r1")

alldiag<-length(meltzrows)
or.output <-data.frame(matrix(NA,nrow=alldiag,ncol=5))
colnames(or.output)<- c('Diagnosis','OR','logOR','loCI','hiCI')
for (i in 1:alldiag){
thiscol<-which(colnames(meltzdat)==meltzcolnames[i])
myrow<-meltzrows[i]
sctN <- sum(mymt)

sctaff<-length(which(meltzdat[,thiscol]==2))

MeltzN <- mymeltz[29,4]+mymeltz[29,7]
Meltzp <- (mymeltz[myrow,2]*nboy_young+mymeltz[myrow,3]*ngirl_young
           +mymeltz[myrow,5]*nboy_old+mymeltz[myrow,6]*ngirl_old)/sum(mymt)
trueMeltzaff <- round((mymeltz[myrow,4]*mymeltz[29,4]+mymeltz[myrow,7]+mymeltz[29,7])/100,0)
ORresult<-do.OR(MeltzN,Meltzp,sctN,sctaff)
or.output[i,1]<-meltzcolnames[i]
or.output[i,2:5]<-ORresult
}
```


```{r beeswarm.function, include=FALSE}
make.beeswarm <- function(mydata,groupcol,subgroupcol,colvalues,varlist,namelist,pngname,pngdim,
                          pngwidth,pngheight,groupcolor,mylims,dodiffmean,addtext,yaxislabel){
  #Now do a grid of beeswarm plots
  
  png_bees<-paste0(pngname,'.png')#name to save png file
  png(png_bees,width=pngwidth,height=pngheight,res=300)
  par(mfrow=pngdim) #5 row and 3 columns
  par(mar=c(5.5,5.1,4.1,1),mgp=c(4, 2, 0)) #mar sets the bottom, left, top and right margins
  #mgp – sets  axis label locations relative to the edge of the inner plot window. The first value represents the location the labels (i.e. xlab and ylab in plot), the second the tick-mark labels, and third the tick marks. The default is c(3, 1, 0). Here need 4,2,0 to avoid 2-line label colliding with frame.
  
  #beeswarm jittered values (last value in jitter statement determines amount of jitter)
  
  grouplevel<-sort(unique(mydata[,groupcol]))
  #list of group values in numeric order
  
  ngroup<-length(grouplevel)
  
  pointcol<-c(16,17) #use filled circle then triangle for ageband 1-2 
  min.val<-min(mydata[,subgroupcol]) #because m/f is coded 0/1 need to check minimum value, so can 
  #add one to code for points if we are using m/f
  min.offset<-0
  if(min.val==0){min.offset <- 1}
  agebit<-pointcol[mydata[,subgroupcol]+min.offset]
  testrange<-which(colnames(mydata)%in%varlist)
  for (i in 1:length(varlist)){
    mytitle<-namelist[i]
    mylabel<-'' #default is to not label y axis unless it is left-most column
    if((i-1)%%pngdim[2]==0){ #heh! gets modulus of i-1 relative to n columns, which is zero if leftmost!
      mylabel<-yaxislabel
    }
    
    beeswarm(jitter(mydata[,testrange[i]],3)~mydata[,groupcol] , xlab='Group',ylab=mylabel,spacing=.8,
             pch = 16,pwpch=agebit,cex=1.5,pwcol = colvalues,main=mytitle,ylim=mylims[1:2],
             cex.axis=1.21,cex.lab=1.5,cex.main=1.5,xaxt='n')
    
       #limits for T-score at +/- 1 SD
    up.lim<-60
    low.lim<-40
    
  #  polygon(c(0,0,(ngroup+1),(ngroup+1)), c(up.lim, low.lim, low.lim, up.lim),
  #          col=adjustcolor("yellow",alpha.f=0.3), border = NA)
    if(addtext==1){
      text(2,(mylims[1]+2),'<------Low Bias------>',cex=1.5)
      text(5,(mylims[1]+2),'<------High Bias------>',cex=1.5)
    }
     #add lines for limits
      abline(a=mylims[3],b=0,col='darkgray',lty=1)
    abline(a=mylims[4],b=0,col='darkgray',lty=2)
    abline(a=mylims[5],b=0,col='darkgray',lty=2)
   
    #add means
    for (g in 1:ngroup){
      myg<-grouplevel[g]
      mm<-mean(mydata[mydata[,groupcol]==myg,testrange[i]],na.rm=TRUE) #means for whole sample by group
      segments(g-.4,mm,g+.4,mm,col = 1,lty=1,lwd=2) #plot straight solid lines at mean for each group
      
      #plot dotted lines for means for age-matched groups, i.e ageband2
      if(dodiffmean==1){
        #identify cases for each group in ageband2
        w1<-which(mydata[,groupcol]==myg)
        w2<-which(mydata$ageband==2)
        shortrange<-intersect(w1,w2)
        mm2<-mean(mydata[shortrange,testrange[i]],na.rm=TRUE)
        segments(g-.4,mm2,g+.4,mm2,col = 1,lty=2,lwd=2) #dotted line for age 6-11
      }
    }
    
  
    axis(1,at=1:6,labels=c('XXX','XXY','XYY','XXX','XXY','XYY'),cex.axis=1.5)
  }
  dev.off()
}

```

```{r beeswarm_sdq, include=F,message=FALSE}
#call the make.beeswarm function - this creates a png file
pngname<-'beeswarm_sdq_dec2018'
varlist<-c('emotional.T','conduct.T','hyperactivity.T','peer.T','prosocial.T','socapt.T')

namelist<-c('Emotional','Conduct','Hyperactivity','Peer Relations',
            'Prosocial','Social Aptitude')
mylims<-c(20,80,50,60,70) #first 2 values are yaxis lims, then values for lines at mean -1sd and -2sd
yaxislabel <- 'T-score (high = impairment)'
groupcol<-which(colnames(my.dawba)=='group8')
subgroupcol<-which(colnames(my.dawba)=='ageSDQ')

pngdim<-c(3,2)
pngwidth<-3000
pngheight<-3000

#We want this order "None","Social Phobia","ASD","ASD-","Social Phobia+ASD","Social Phobia+ASD-")
diaglist<-c(0,2,20,10,22,12)

my.dawba$bicode3<-my.dawba$bicode #initialise
for (d in 1:length(diaglist)){
  w<-which(my.dawba$bicode3==diaglist[d])
  my.dawba$bicode3[w]<-d
}

mycols<- c("grey66", "cornflowerblue","red", "hotpink1","purple","blue")

dodiffmean<-0
addtext<-0
w<-which(my.dawba$group8level <7)
mydata<-my.dawba[w,]
mydata$group8<-droplevels(mydata$group8)#need this to stop it remembering empty level!
#levels(mydata$Group)[4]<-'Twin +\nConcerns'
colvalues<-mycols[mydata$bicode3]
addtext<-1
make.beeswarm(mydata,groupcol,subgroupcol,colvalues,
              varlist,namelist,pngname,pngdim,pngwidth,pngheight,
              groupcolor,mylims,dodiffmean,addtext,yaxislabel)

```



```{r beeswarmSRS,include=FALSE}
pngname<-'beeswarm_srs_T_2018'
pngdim<-c(3,2)
pngwidth<-3000
pngheight<-3000
varlist<-c('socaw_ss','soccog_ss','soccomm_ss','socmot_ss','autfeat_ss','srs_t_score')
namelist<-c('Soc. Awareness','Soc. Cognition','Soc. Communication','Soc. Motivation',
            'Autistic Features','Total SRS')
mylims<-c(0,100,50,60,70) #first 2 values are yaxis lims, then values for lines at mean -1sd and -2sd

make.beeswarm(mydata,groupcol,subgroupcol,colvalues,
              varlist,namelist,pngname,pngdim,pngwidth,pngheight,
              groupcolor,mylims,dodiffmean,addtext,yaxislabel)


```



```{r MANOVAfunctions, include=FALSE, echo=FALSE}
######################################################################################
#define function for running MANOVA here
#Currently can have up to 2 covariates
do.manova <- function(mydata,mycolrange,gpcolumn,gpvalues,covcolumns){
  
  #check multivariate normality for each group
  for (i in 1:length(gpvalues)){
    my.matrix <- as.matrix(mydata[mydata[,gpcolumn]==gpvalues[i],mycolrange])
    print(paste('Check multivariate normality: group',i))
    print(mshapiro.test(t(my.matrix)))
  }
  my.matrix <- as.matrix(mydata[,mycolrange])
  mycov1<-mydata[  ,covcolumns[1]]
  # mycov2<-mydata[  ,covcolumns[2]]
  if (covcolumns==0){ 
    mymanova.fit <- manova(my.matrix ~ as.factor(mydata[,gpcolumn]))}
  else if (length(covcolumns)==2)#not currently used
  { mymanova.fit <- manova(my.matrix ~ as.factor(mydata[,gpcolumn])+mycov1+mycov2)  } 
  else if (length(covcolumns)==1)
  {  mymanova.fit <- manova(my.matrix ~ as.factor(mydata[,gpcolumn])+mycov1) }
  return(mymanova.fit)
}

######################################################################################

######################################################################################
#define function for populating manova.dataframe here
manova.write <- function(mycomparison,effectnames,offset,manova.stats,manova.dataframe){
  for (i in 1:length(effectnames)){
    j<-i+offset
    if(i==1){
      manova.dataframe[j,1]<-mycomparison
    }
    
    manova.dataframe[j,2]<-effectnames[i]
    manova.dataframe[j,3]<-round(manova.stats[i,2],3)
    manova.dataframe[j,4]<-round(manova.stats[i,3],2)
    manova.dataframe[j,5]<-round(manova.stats[i,4],0)
    manova.dataframe[j,6]<-round(manova.stats[i,5],0)
    manova.dataframe[j,7]<-round(manova.stats[i,6],3)
  }
  return(manova.dataframe)
}
######################################################################################

```

```{r run.manova, include=FALSE}
## Now run manovas - use the functions defined above
#create dataframe to hold results from all Manovas
manova.dataframe<-data.frame(matrix(NA,nrow=3,ncol=7))
colnames(manova.dataframe)<-c('Comparison','Effect','Wilks','F','df1','df2','p')
manova.sdqetc.df<-manova.dataframe #just copying basic file

#To run the analyses, can use the sct.only file that was created above 
#First convert values to z-scores in same direction.

#find col numbers for columns to use
#NB intercorrelations within SRS are high and cause problems, so use overall SRS t-score
mycolrange<-which(colnames(sct.only) %in% c("emotional.T","conduct.T", "hyperactivity.T" ,  "peer.T", "prosocial.T"))
#ignore soc aptitude as it has some missing data

gpcolumn<-which(colnames(sct.only)=='group8level')
gpvalues<-c(1,2,3)
covcolumns<-which(colnames(sct.only) %in% c('langfactor'))

#First just do groups 1-3
#NB cases with missing data are excluded, so doesn't matter if you use lobias.sct or lobias.sct2
#Result will be the same, but use lobias.sct2 to get numbers to report
gp.sctlow<-filter(sct.only,group8level<4)

mymanova.fit<-do.manova(gp.sctlow,mycolrange,gpcolumn,gpvalues,covcolumns)
mymanova.sum<-summary(mymanova.fit, test="Wilks")

#Massive failure of multivariate normality in all 3 groups!
#So go instead for many nonparametric tests and use permutation if necessary to establish pvalue
```

```{r permutekruskalfunction,include=FALSE}
do.permute <- function(mydata,mycolrange,gpcolumn,gpvalues,n.iter,mylabs,
                       mytitle,pngname,prevfile,upperquant){
  if(n.iter==0){myp<-prevfile} #save time by reading prior file
  #may need to load this first using load comment
  else{
    #allocate group at random for permutation test
    n.forgp <-table(mydata[,gpcolumn])
    range1start<-1
    range2start<-n.forgp[1]+1
    range3start<-nrow(mydata)+1 #dummy value for case where just 2 groups
    if(length(gpvalues)>2){
      range3start <-n.forgp[1]+n.forgp[2]+1}
    myp<-data.frame(matrix(NA,nrow=n.iter*length(mycolrange),ncol=2))
    thisrow <- 0
    mydata$nugp <-mydata[,gpcolumn] #default is correct group value.
    for (i in 1:n.iter){
      if (i>1){ #first run retains correct group value but then shuffled for subsequent runs
        mydata$thisr<-rank(runif(nrow(mydata)))
        mydata$nugp <- 3
        w1<-which(mydata$thisr < range3start)
        w2<-which(mydata$thisr < range2start)
        mydata$nugp[w1]<-2
        mydata$nugp[w2]<-1
      }
      thisk <- 0
      for (k in mycolrange){
        thisrow<-thisrow+1
        thisk <- thisk+1
        simk<- kruskal.test(mydata[,k]~mydata$nugp) 
        myp[thisrow,1]<-thisk
        myp[thisrow,2]<-simk$statistic
      }
    }
  }
  myptrue<-myp[1:length(mycolrange),]
  mypsim<-myp[-(1:length(mycolrange)),]
  
  #see https://www.r-bloggers.com/whisker-of-boxplot/
  #we want to set the whisker to denote the 97.5% point (ie top end of 95%CI)
  #ideally for each measure separately, but failing that, for the overall dataset
  #This version now allows user to specify which upperquantile is used with upperquant
  myquant<-quantile(mypsim$X2,c(.25,.75,upperquant))
  myfinmultiplier <- (myquant[3]-myquant[2])/(myquant[2]-myquant[1])
  png(pngname,width=2000,height=1200,res=300)
  par(mar=c(4, 12, 1, 1) + 0.1)#bottom, left, top and right margins
  
  boxplot(myp$X2~myp$X1,yaxt='n',xlab='Chi square from randomisation',
          horizontal=TRUE,las=1,outcol='#FFFFFF',outcex=.1,range=myfinmultiplier,
          main='')
  axis(2, at=1:length(mycolrange),labels=mylabs,las=2)
  for (i in 1:length(mycolrange)){
    text(myptrue[i,2],i,'*',col='red',cex=2.5)
  }
  dev.off()
  
  
  return(myp)
}
```


```{r run.kruskal.permute,include=FALSE}
#Same routine will be used for various comparisons; each calls the do.permute function above
load("p1p2.Rdata") #loads in previously created randomised data

#Analysis 1: compare the 3 low bias SCT groups
mydata<-filter(my.dawba,group8level<4)
prevfile<-'' #default value
k.cols <- c("socaw_ss","soccog_ss","soccomm_ss","socmot_ss","autfeat_ss")
mycolrange<-which(colnames(mydata) %in% k.cols)
gpcolumn<-which(colnames(mydata)=='group8level')
gpvalues<-unique(mydata$group8level)
mylabs<-c('SRS: Social Awareness','SRS: Social Cognition','SRS: Social Communication','SRS: Social Motivation','SRS: Autistic Features')
n.iter<-0 #set to 0 if you want to use a saved file; otherwise 10000
if (n.iter==0){prevfile<-myp1} #if n.iter is > 0, this is ignored and new data will be generated
mytitle<-'Comparing three Low Bias SCT groups' #This was used on plot but currently ignored
pngname <-'random99_chiplotSRS_LowBias_karyotype.png'
upperquant <- .99 #specifies upper fins on boxplot
myp1 <- do.permute(mydata,mycolrange,gpcolumn,gpvalues,n.iter,mylabs,
                   mytitle,pngname,prevfile,upperquant)
#the function returns the generated dataset (or just repeats the specified one)

#Next compare high bias and low bias SCTs
#Most parameters are unchanged, but need to specify new dataset and group code
mydata <-filter(my.dawba,group8level<7)
gpcolumn<-which(colnames(mydata)=='bias')
gpvalues<-unique(mydata$bias)
mytitle<-'Low Bias vs High Bias SCT groups'
pngname <-'random99_chiplotSRS_low_vs_hibias.png'
if(n.iter==0){prevfile<-myp2}
myp2 <- do.permute(mydata,mycolrange,gpcolumn,gpvalues,n.iter,mylabs,
                   mytitle,pngname,prevfile,upperquant)


#since it took a while to make these randomised files (about 7 min each), we'll save them
save(myp1,myp2, file = "p1p2.Rdata")
#these variables can then be loaded in with 'load("p1p2.Rdata")' and used for prevfile


```




```{r beeswarm.SRSt.with.diagnoses,include=FALSE}
#########################################################################
#Now just of historic value: this was original attempt with colour coding
srscol<-which(colnames(my.dawba)=='srs_t_score')

#first need to jitter those with SRS of 90 - too many to plot
#add a random number to those with scores of 90
  w<-which(my.dawba[,srscol]>89)
  myr<-runif(w,0,4)
  my.dawba[w,srscol]<-90+myr

#--------------------------------------------------------------------------
#We want a gap between the two blocks of groups.
#Try doing this by creating a datapoint from a fake group
my.dawba2<-filter(my.dawba,group8level<7)
  myn<-nrow(my.dawba2)
dummyrow<-my.dawba2[myn,]
my.dawba2<-rbind(my.dawba2,dummyrow)
my.dawba2$record_id[myn+1]<-'dummy'
my.dawba2[(myn+1),srscol]<-NA
my.dawba2$group8level[myn+1]<-3.5 #intermediate score
my.dawba2$group8level<-as.factor(my.dawba2$group8level)
levels(my.dawba2$group8level)<-c('XXX','XXY','XYY',' ','XXX*','XXY*','XYY*') #* denotes asc bias group
my.dawba2$bicode2[myn+1]<-NA #NA so won't plot

#recode the diagnostic groups to numbers
#the 2 digit number is ASD*10 then Social Phobia

#We want this order "None","Social Phobia","ASD","ASD-","Social Phobia+ASD","Social Phobia+ASD-")
diaglist<-c(0,2,20,10,22,12)

my.dawba2$bicode3<-my.dawba2$bicode #initialise
for (d in 1:length(diaglist)){
  w<-which(my.dawba2$bicode3==diaglist[d])
  my.dawba2$bicode3[w]<-d
}

mycols<- c("grey66", "cornflowerblue","red", "hotpink1","purple","blue")
#quartz()

png_bees<-'beeswarm_srs_T_orig2018.png'
png(png_bees,width=1200,height=500)
par(mfrow=c(1,2)) #one row and 2 columns - one col is for legend
par(mar=c(5.1,4.1,4.1,0.1))
mysrsnames<-c('T-score')

beeswarm(srs_t_score~group8level , data = my.dawba2, xlab='Trisomy',ylab='SRS T-score',
         horizontal=FALSE,ylim=c(35,95),col = 5, pch = 16,cex=1.7,cex.axis=1.5,cex.lab=1.5,
         pwcol = mycols[my.dawba2$bicode3])
par(xpd=FALSE) #confine to plot area
abline(a=60,b=0,col='darkgray',lty=2)
abline(a=75,b=0,col='darkgray',lty=2)
text(4,65,'Mild',cex=1.5)
text(4,80,'Severe',cex=1.5)
par(xpd=NA) #write outside plot area
text(2,25,'Low bias',cex=1.3)
text(6,25,'High bias',cex=1.3)

#legend as separate plot in panel 2
mylegendcols<- mycols

mysrsnames<-c('T-score')
plot(x=NULL, y=NULL , type = "n", axes = F, xlab = "", ylab = "", xlim=c(0,20), ylim=c(0,250)) #set up the plot?
mylabels <- c("None","Social Phobia","ASD","ASD-","Social Phobia+ASD","Social Phobia+ASD-")
#draw rectangles with labels
for (i in 1:6){
  x1<-0;x2<-2;y1<-(i-1)*41;y2<-y1+31;
  rect(x1,y1,x2,y2, col = mylegendcols[(7-i)], border = "transparent") 
  text(x2+.5,y1+15,mylabels[(7-i)],pos=4,cex=1.3)
}
text(0,y2+18,'Diagnosis',pos=4,offset=0,font=2,cex=1.5)
dev.off()

```

```{r srscutoff, include=FALSE}
my.dawba$srscut75<-NA
w<-which(my.dawba$srs_t_score>0)
my.dawba$srscut75[w]<-0
w1<-which(my.dawba$srs_t_score>74)
my.dawba$srscut75[w1]<-1

my.dawbashort<-filter(my.dawba,group8level<7)
srscuttab<-table(my.dawbashort$srscut75,my.dawbashort$asd_dsm_agree)



```

```{r dsmagree, include=FALSE}
#JK suggests we should report JK/AW agreement
#first convert AW code 4 to code 1
w<-which(my.dawbashort$rater2=='JK') # use next line to select those with AW and JK codes
#temp<-my.dawbashort[w,]
temp<-my.dawbashort
temp<-select(temp,rater2,record_id,asd,asdband,asd_dsm_agree,asd_dsm_r1,asd_dsm_r2,social_anx_dsm_r1,social_anx_dsm_r2) #NB asd is 0 or 1 depending on whether parent reported a diagnosis of asd
w1<-which(temp$asd_dsm_r1==4)
temp$asd_dsm_r1[w1]<-1
agreetab<-table(temp$asd_dsm_r1,temp$asd_dsm_r2)
perfectagree <- sum(agreetab[1,1],agreetab[2,2],agreetab[3,3])
disagree2<-sum(agreetab[1,3],agreetab[3,1])
agreeN<-sum(agreetab)
percagree1<-round(100*perfectagree/agreeN,1)
percagree3<-round(100*disagree2/agreeN,1)
percagree2<-100-percagree1-percagree3

agreetabsoc<-table(temp$social_anx_dsm_r1,temp$social_anx_dsm_r2)
percagrees1<-round(100*sum(agreetabsoc[1,1],agreetabsoc[3,2])/agreeN,1)

comm_asd_agreetab<-table(temp$asd,temp$asd_dsm_agree)
bandasd_agreetab<-table(temp$asd,temp$asdband)
bandasd_agreetab2<-table(temp$asd_dsm_agree,temp$asdband)
```
```{r preflowchart.sct, include=FALSE}
#http://rich-iannone.github.io/DiagrammeR/graphviz_and_mermaid.html

# Based on version done for twins by Alex Wilson
# 1st nov 2017 - now with plot of SRS by group by diagnosis
w<-which(sct.data$trisomy==9) #remove isochromosome
sct.data<-sct.data[-w,]
#Referral source
nhscases<-c(70, 212, 213, 214, 215, 218, 221, 222, 223, 225, 229, 231,
            233, 234, 237, 239, 240, 241, 247, 251, 252, 254, 255, 257,
            258, 264, 265, 268, 270, 273, 274, 276, 283, 285, 297, 304,
            306, 307, 310, 318, 319, 321, 322, 323, 327, 328, 329, 331,
            334, 335, 336, 337, 340, 342, 346, 348, 351, 352, 355, 357,
            358, 360, 362, 365, 367, 368, 369, 370)
nhsset<-filter(sct.data,record_id %in% nhscases)
othset<-setdiff(sct.data,nhsset)

y<-table(nhsset$pre_postnatal_diag)
z<-table(othset$pre_postnatal_diag)
y1=y[1]
y2=y[2]
z1=z[1]
z2=z[2]

#Count for diagnosis pre or postnatal
prenatals<-subset(sct.data,pre_postnatal_diag==0)
postnatals<-subset(sct.data,pre_postnatal_diag==1)
label1<-'Prenatal'
label2<-'Postnatal'
label3<-'When SCT identified'

n.A<-nrow(postnatals) #Ns needed for flowchart are now computed
n.B<-nrow(prenatals)

# reason for testing 2 = Behaviour and 3 =Neurodev.
biasgrp<-filter(sct.data,why_tested==3|why_tested==2)
nobiasgrp<-filter(sct.data,why_tested<2|why_tested>3)
n.C <-nrow(nobiasgrp)
n.D <-nrow(biasgrp) 

#subset by karyotype
xxx1<-subset(nobiasgrp,trisomy==1)
xxy1<-subset(nobiasgrp,trisomy==2)
xyy1<-subset(nobiasgrp,trisomy==3)
xxx2<-subset(biasgrp,trisomy==1)
xxy2<-subset(biasgrp,trisomy==2)
xyy2<-subset(biasgrp,trisomy==3)

n.E<-nrow(xxx1)
n.F<-nrow(xxy1)
n.G<-nrow(xyy1)
n.H<-nrow(xxx2)
n.I<-nrow(xxy2)
n.J<-nrow(xyy2)

#now check DAWBA
n.K=length(which(xxx1$asd_dsm_r1> -1))
n.L=length(which(xxy1$asd_dsm_r1> -1))
n.M=length(which(xyy1$asd_dsm_r1> -1))
n.N=length(which(xxx2$asd_dsm_r1> -1))
n.O=length(which(xxy2$asd_dsm_r1> -1))
n.P=length(which(xyy2$asd_dsm_r1> -1))

#now create flow chart ; TB denotes top to bottom
#Need to add labels along the side: top row 'Reason for testing or Time of testing'
#Then trisomy, then with DAWBA data
```

```{r flow_SCTa, include=FALSE}
flow1a<-DiagrammeR::grViz("
digraph a_nice_graph {
        
        # node definitions with substituted label text
        node [shape = plaintext, fontname = Helvetica]
        X[label='@@1']
        Y[label= 'Bias']
        Z[label= 'Trisomy']
        ZZ[label = 'With DAWBA']
        
        node [shape=rectangle, width = 1,height=.6]
        A[label='@@2']
        B[label='@@3']
        node [shape=square, fixedsize = true, width = .6,height=.6]
        C[label='@@4']
        D[label='@@5']
        E[label='@@6']
        F[label='@@7']
        G[label='@@8']
        H[label='@@9']
        I[label='@@10']
        J[label='@@11']
        K[label='@@12']
        L[label='@@13']
        M[label='@@14']
        N[label='@@15']
        O[label='@@16']
        P[label='@@17']
        # edge definitions with the node IDs
        A -> C
        B -> {C D}
        C -> {E F G}
        D -> {H I J}
        E -> K
        F -> L
        G -> M
        H -> N
        I-> O
        J -> P
    
        X -> Y [alpha=0,color='white']
        Y -> Z [alpha=0,color='white']
       Z -> ZZ [alpha=0,color='white']
}

[1]: paste0(label3, ':\\n ',' Recruited from')
[2]: paste0(label1,':\\n', 'NHS: N = ',y1,':\\n', 'Other: N = ',y2)
[3]: paste0(label2,':\\n', 'NHS: N = ',z1,':\\n' ,'Other: N = ',z2)
[4]: paste0('Low',':\\n', 'N = ',n.C)
[5]: paste0('High',':\\n', 'N = ',n.D)
[6]: paste0('XXX',':\\n', 'N = ',n.E)
[7]: paste0('XXY',':\\n', 'N = ',n.F)
[8]: paste0('XYY',':\\n', 'N = ',n.G)
[9]: paste0('XXX',':\\n', 'N = ',n.H)
[10]: paste0('XXY',':\\n', 'N = ',n.I)
[11]: paste0('XYY',':\\n', 'N = ',n.J)
[12]: paste0('N = ',n.K)
[13]: paste0('N = ',n.L)
[14]: paste0('N = ',n.M)
[15]: paste0('N = ',n.N)
[16]: paste0('N = ',n.O)
[17]: paste0('N = ',n.P)
")


flow1a %>% export_svg %>% charToRaw %>% rsvg_png("SCT_asd_flow.png")

```

```{r preflowchart.twin, include=FALSE}
#Now do plot for twin data 
twin.short$exclude<-0
w<-which(twin.short$neurodev_diag %in% c(3,37,389))
twin.short$exclude[w]<-1
w<-which(twin.short$piq<70)
twin.short$exclude[w]<-3
w<-which(twin.short$pass_hearing==0)
twin.short$exclude[w]<-2

myexclude<-table(twin.short$exclude)

y1<-table(twin.short$exclude)

inc1<-y1[[1]] #N individual twins included
exc1<-nrow(twin.short)-inc1 #N excluded bcs ASD/hearing loss/lowIQ

twin.short$withdawba <-0
w<-which(twin.short$asd_dsm_r1 > -1)
twin.short$withdawba[w]<-1

twin.short.inc<-twin.short[twin.short$exclude==0,]

noconcernM<-table(twin.short.inc$parent_conc[twin.short.inc$female==0])[[1]]
concernM<-table(twin.short.inc$parent_conc[twin.short.inc$female==0])[[2]]

noconcernF<-table(twin.short.inc$parent_conc[twin.short.inc$female==1])[[1]]
concernF<-table(twin.short.inc$parent_conc[twin.short.inc$female==1])[[2]]

dawba_LI_Y <- table(twin.short.inc$withdawba[twin.short.inc$parent_conc==1])[[2]]
dawba_LI_N <- table(twin.short.inc$withdawba[twin.short.inc$parent_conc==1])[[1]]

dawba_TD_Y <- table(twin.short.inc$withdawba[twin.short.inc$parent_conc==0])[[2]]
dawba_TD_N <- table(twin.short.inc$withdawba[twin.short.inc$parent_conc==0])[[1]]

#now create flow chart ; TB denotes top to bottom

label3<-' '

flow2<-DiagrammeR::grViz("
            digraph twinflowdawba {
            
            # node definitions with substituted label text
            node [shape = plaintext, fontname = Helvetica]
            X[label='@@1']
             Y[label= 'Exclude:\\n ASD (DAWBA), \\nlow IQ, hearing']
            Z[label= 'Total N twin children']
            Q[label='DAWBA\\n available']
            
           node [shape=square,fixedsize = true, width = 1.4,height=1.2]
            I[label='@@10']
            A[label='@@2',fillcolor=lightBlue]
            B[label='@@3']
            
            C[label='@@4']
            D[label='@@5']
            E[label='@@6']
            F[label='@@7']
            G[label='@@8']
            H[label='@@9']
            
            # edge definitions with the node IDs
            I -> {A B}
            B -> {G H}
            G -> {C D}
            H -> {E F}
            
            X -> Y [alpha=0,color='white']
            Y -> Z [alpha=0,color='white']
            Z -> Q [alpha=0,color='white']
            }
            
            [1]: paste0(label3, ':\\n ',' ')
            [2]: paste0('Excluded:\\n N = ',exc1)
            [3]: paste0('Included:\\n N = ',inc1)
            [4]: paste0('with DAWBA:\\nN = ',dawba_LI_Y)
            [5]: paste0('without DAWBA:\\nN = ',dawba_LI_N)
            [6]: paste0('with DAWBA:\\nN = ',dawba_TD_Y)
            [7]:  paste0('without DAWBA:\\nN = ',dawba_TD_N)
            [8]: paste0('Parental concern \\n or SALT,\\n N (male) = ',concernM,'\\n N (female) = ',concernF)
            [9]: paste0('No concern \\n No SALT,\\n N (male) = ',noconcernM,'\\n N (female) = ',noconcernF)
            [10]: paste0(y1,' twin pairs, \\n 1 selected at \\n random')
            
            ")

flow2 %>% export_svg %>% charToRaw %>% rsvg_png("twins_flow_dawba.png")

```

```{r poweranalysis}
#comparing 3 groups for Low bias only
lobiastot<-n.K+n.L+n.M
meltztot <-5219
meltzhi<-round(meltztot*.05,0)
meltzlo<-meltztot-meltzhi
meltzodds<-meltzhi/meltzlo

#compare difft odds ratios
thisN <- 14 #smallest N in low bias group
#try with N of 1 or more affected group
for (i in 1:5){
simodds<-i/(thisN-i)
simlogOR <-log(simodds/meltzodds)

se.simlogOR <-sqrt(1/meltzhi+1/meltzlo+1/i+1/(thisN-i))
print(paste(i,round(simlogOR,2),round(se.simlogOR,2),round(simlogOR-1.96*se.simlogOR,2)))
}

pwr.anova.test(k = 3,  sig.level = 0.05,n = (n.K+n.L+n.M),power=.8)

pwr.t2n.test(n1 = (n.K+n.L+n.M), n2= (n.N+n.O+n.P),  sig.level =.05, power =.8 )
```


```{r SocialPhobiaN,include=FALSE}

#For SE of log odds ratio: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1127651/
socanx.xxy1 <- sum(mytab[c(2,4,6),2]) #low bias xxy cases with SocAnx
socanxplus.xxy1 <- sum(mytab[c(4,6),2])#low bias xxy cases with SocAnx+ASD/ASD-
socanx.xxy2 <- sum(mytab[c(2,4,6),5])#hi bias xxy cases with SocAnx
socanxplus.xxy2 <- sum(mytab[c(4,6),5])#hi bias xxy cases with SocAnx+ASD/ASD-

MeltzN <- 5219 #size of population from which population estimate BOYS
Meltzp <-.4 #percentage prevalence from Meltzer sample BOYS
Meltzaff <- round(MeltzN*Meltzp/100,0)
Meltznon <-MeltzN-Meltzaff
odds1 <-Meltzaff/Meltznon

xxynon <-  (sum(mytab[,2])-socanx.xxy1)
odds2 <-socanx.xxy1/xxynon
oddsr <-odds2/odds1
logoddsr <-log(oddsr)
se.logoddsr <-sqrt(1/Meltzaff+1/Meltznon+1/socanx.xxy1+1/xxynon)
ci.low <-round(logoddsr-1.96*se.logoddsr,2)
ci.hi <- round(logoddsr+1.96*se.logoddsr,2)
```

```{r asd.dawba, include=FALSE}
#!!!!!!!!!!!!!!!!!!!!!!!!!!
trydoubleprevalence<-0   #! set to zero for final version; set to 1 to inspect data if higher prevalence used
#!!!!!!!!!!!!!!!!!!!!!!!!!!
#For this we combine the ASD- and ASD diagnoses
asd.xxx1 <- sum(mytab[3:6,1])#low bias xxx cases with ASD/ASD-
asd.xxx2 <- sum(mytab[3:6,4])#hi bias xxx cases with ASD/ASD-
asd.xxy1 <- sum(mytab[3:6,2])#low bias xxy cases with ASD/ASD-
asd.xxy2 <- sum(mytab[3:6,5])#hi bias xxy cases with ASD/ASD-
asd.xyy1 <- sum(mytab[3:6,3])#low bias xyy cases with ASD/ASD-
asd.xyy2 <- sum(mytab[3:6,6])#hi bias xyy cases with ASD/ASD-
#Odds ratio for low bias only
xxxnon<-(sum(mytab[,1])-asd.xxx1)
xxynon<-(sum(mytab[,2])-asd.xxy1)
xyynon<-(sum(mytab[,3])-asd.xyy1)

MeltzN <- 5219 #size of population from which population estimate (same for boys and girls)
Meltzp.boy <-.5 #percentage prevalence from Meltzer sample BOYS
Meltzp.girl <-.1 #percentage prevalence from Meltzer sample BOYS


if (trydoubleprevalence==1){
#Meltzer rates seem low by current standards: what are OR like if we double them
Meltzp.boy<-1
Meltzp.girl<-.2
}

Meltzaff.boy <- round(MeltzN*Meltzp.boy/100,0)
Meltznon.boy <-MeltzN-Meltzaff.boy
odds1.boy <-Meltzaff.boy/Meltznon.boy

Meltzaff.girl <- round(MeltzN*Meltzp.girl/100,0)
Meltznon.girl <-MeltzN-Meltzaff.girl
odds1.girl <-Meltzaff.girl/Meltznon.girl

odds.xxx <-asd.xxx1/xxxnon #odds ratios just for the low bias cases
odds.xxy <-asd.xxy1/xxynon
odds.xyy <-asd.xyy1/xyynon
oddsr.xxx <-odds.xxx/odds1.girl
oddsr.xxy <-odds.xxy/odds1.boy
oddsr.xyy <-odds.xyy/odds1.boy
logoddsr.xxx <-log(oddsr.xxx)
logoddsr.xxy <-log(oddsr.xxy)
logoddsr.xyy <-log(oddsr.xyy)
se.logoddsr.xxx <-sqrt(1/Meltzaff.girl+1/Meltznon.girl+1/asd.xxx1+1/xxxnon)
se.logoddsr.xxy <-sqrt(1/Meltzaff.boy+1/Meltznon.boy+1/asd.xxy1+1/xxynon)
se.logoddsr.xyy <-sqrt(1/Meltzaff.boy+1/Meltznon.boy+1/asd.xyy1+1/xyynon)
ci.low.xxx <-round(logoddsr.xxx-1.96*se.logoddsr.xxx,2)
ci.hi.xxx <- round(logoddsr.xxx+1.96*se.logoddsr.xxx,2)
ci.low.xxy <-round(logoddsr.xxy-1.96*se.logoddsr.xxy,2)
ci.hi.xxy <- round(logoddsr.xxy+1.96*se.logoddsr.xxy,2)
ci.low.xyy <-round(logoddsr.xyy-1.96*se.logoddsr.xyy,2)
ci.hi.xyy <- round(logoddsr.xyy+1.96*se.logoddsr.xyy,2)
```
# Things to do before submitting
1. Move tables and figures to end
2. Autoformat references, and ensure left-justified
3. Check heading of tables for extraneous characters
4. Check numbered section that has somehow got its own weird format
5. Add page numbers
6. Abstract, keywords etc change
7. Authors add details (current APA is not compatible)

# Abstract

**Background**: Recent studies suggest that an extra sex chromosome increases the risk of both autism spectrum disorder (ASD) and social anxiety, but it unclear whether these risks are specific to particular karyotypes.  **Methods**: We considered diagnostic data from an online psychiatric assessment (DAWBA) and questionnaire responses completed by parents of children with 47,XXX (N = `r n.K+n.N`), 47,XXY (N = `r n.L+n.O`) and 47,XYY (N = `r n.M+n.P`) karyotypes. Analysis focused mainly on `r n.K+n.L+n.M` children who were diagnosed prenatally or on the basis of other medical concerns in childhood (Low Bias subgroup), to minimise ascertainment bias. **Results**: Children with symptoms of ASD who fell short of meeting DSM-IV criteria were coded as cases of ASD-. The odds ratio of ASD/ASD- in the Low Bias group was computed relative to gender-specific population norms. This gave log odds ratio (95% confidence interval) of `r logoddsr.xxx` (`r ci.low.xxx` - `r ci.hi.xxx`) for XXX girls; `r logoddsr.xxy` (`r ci.low.xxy` - `r ci.hi.xxy`) for XXY boys; and `r logoddsr.xyy` (`r ci.low.xyy` - `r ci.hi.xyy`) for XYY boys. Despite this elevated risk, most children had no autistic features. A diagnosis of DSM-IV Social Phobia was rare, though, in line with prediction, all `r sum(mytab[c(2,4,6),1:3])` Low Bias cases with this diagnosis had 47,XXY karyotype. All three trisomy groups showed increased risk of milder symptoms of social anxiety.  **Conclusions**: An increased risk of ASD was found in girls with 47,XXX karyotype, as well as in boys with 47,XXY or 47,XYY. Symptoms of social anxiety were increased in all three karyotypes. There was wide variation in psychiatric status of children with the same karyotype, suggesting that an extra sex chromosome affects developmental stability in a non-specific way, with a diverse range of possible phenotypes.

# Introduction
Chromosome trisomies arise from an error of cell division during meiosis, so that either the egg or sperm contains two copies of the chromosome rather than one. When a trisomy affects one of the autosomes, this is often lethal, or causes severe physical and mental abnormalities. Trisomies of the sex chromosomes, however, have much milder effects, and often go undetected. In an analysis of 244,848 UK Biobank female participants, researchers discovered 110 cases of trisomy X, none of whom appeared aware of the trisomy {Tuke, 2017 #39165}. Undiagnosed cases make study of the impact of sex chromosome trisomies difficult, because those cases who do come to attention may be atypical, with genetic testing being prompted by developmental abnormalities.

In the 1960s, several centres came together with the aim of evaluating the impact of sex chromosome trisomies in samples identified on newborn screening. The three kinds of trisomy - trisomy X (47,XXX), Klinefelter's syndrome (47,XXY) and 47,XYY karotypes (henceforth referred to as XXX, XXY and XYY) were all found to be associated with neurodevelopmental problems, particularly affecting language and motor functions, though varying in profile and severity both within and between the three karyotypes {Leggett, 2010 #17405}. 

More recent studies, however, have noted an increased risk of autism spectrum disorder (ASD) in males: both those with Klinefelter's syndrome and boys with an extra Y chromosome {Van Rijn, 2008 #16706; Bishop, 2011 #17168; Cordeiro, 2012 #18895; Ross, 2012 #18868; Joseph, 2018 #39159}, and in girls with trisomy X {Wigby, 2016 #39137}. Furthermore, in all three trisomies, Bishop et al (2011) found evidence of milder autistic-like communication problems affecting pragmatic as well as structural domains of communication, even after excluding those with a diagnosis of ASD. 

In one of the few studies to include girls and boys, Van Rijn and colleagues {Van Rijn, 2014 #38662} compared a sample of males and females (aged 9–18 years) with an extra X chromosome to a group of children with an ASD diagnosis, using the Autism Diagnostic Interview - Revised (ADI-R) {Le Couteur, 1989 #6304}. The trisomy samples included both prenatally identified and later-referred children but no differences were found between these subgroups. Furthermore, there were no differences between boys and girls with an extra X chromosome, with around 20% of children scoring above cutoff on the ADI-R in all three domains. Children were also given the Social Responsiveness Scale (SRS) {Constantino, 2005 #14237}.The rationale behind the SRS is that ASD is not a qualitatively distinct disorder, but is the extreme point on a continuum of impairment. The SRS  was developed to quantify social and related impairments characteristic of autism in the broader population, and is scored so that the population mean is 50 (SD = 10), with high scores corresponding to impairment. Van Rijn et al found that total SRS T-scores in the extra X group (XXX M = 63.8, SD = 31.7; XXY M = 68.7, SD = 31.7) were higher than those of a typically developing group (M = 26.3, SD = 16.3), but lower than an ASD comparison group (M = 97.6, SD = 28.8). 44.2% of the extra X group scored above the threshold indicating clinically significant impairment (defined here as 65). This study, then, suggested that rates of autistic symptomatology are far higher in children with an extra X chromosome than was previously thought. However, results from the SRS do need to be interpreted cautiously. There is mounting evidence that while the SRS may be sensitive to ASD, it is not very specific. A number of studies have shown that high scores on the SRS are not restricted to those with a diagnosis of autism, but are also reported for young people with a range of diagnoses, including conduct disorder, ADHD and anxiety, e.g., {Pine, 2008 #39110; Towbin, 2005 #15584; Bölte, 2011 #19563; Settipani, 2012 #39113; Cholemkery, 2014 #19564; Moul, 2015 #39168; South, 2017 #39114}. In a sample of boys with XXY studied by Tartaglia et al {Tartaglia, 2010 #39178}, over 25% scored in the mild-to-moderate or severely impaired range in all domains of social responsiveness with the exception of social awareness, yet, in a subset of 20 children from this sample given a more detailed ASD assessment, only one boy met diagnostic criteria for a Pervasive Developmental Disorder. 

Another intriguing finding from the study by Van Rijn et al (2014) was that the extra X groups also showed high levels of social anxiety on all five subscales of the Social Anxiety Scale (SAS), with mean scores being higher than both the ASD and the typically-developing group. The authors noted that this finding of increased social anxiety was consistent with earlier descriptive accounts of the impact of XXX and XXY karyotypes. Comorbidity between ASD and social anxiety is common in children with typical chromosome complements {Spain, 2018 #39176}, but the particularly high rates of social anxiety in the children with trisomies suggests that this might be a separate condition, not just secondary to ASD. As noted by Van Rijn et al (2014), this suggests that 'children with an extra X chromosome differ from children with ASD in an important aspect, which is the ability to reflect on their own social functioning and to develop a related concern about social rejection, i.e., the opinions, thoughts and expressions of others.' (p. 317)

Most of the focus of studies of children with sex chromosome trisomies using the SRS has been on the impact of an additional X chromosome, Cordeiro and colleagues {Cordeiro, 2012 #18895} compared boys with XXY and XYY chromosome complements, and found that the XXY group had significantly less impairment than the XYY group on the total SRS T-score (XXY M = 62.0, SD = 15.4; XYY M = 72.8, SD = 15.8). There was, however, evidence of ascertainment bias in this study, with prenatally diagnosed children (XXY M = 58.6, SD = 14.2; XYY = 63.5, SD = 15.6) having lower total SRS T-scores than postnatally diagnosed children (XXY M = 66.3, SD = 16.1; XYY M = 77.3, SD = 14.1). 47.1% of all the XXY children and 85% of the XYY children scored above the threshold for social impairment (defined as 60). Scores on individual SRS subscales were very similar to total scores, with significantly more impairment for XYY children compared to XXY children on all subscales, except Social Motivation, on which they scored the same; this scale represented a relative strength for the XYY group in terms of their overall profile. For the XXY group, means were above the 60 cut-off on all subscales except Social Awareness, which represented a relative strength for this group. Interestingly, these patterns for XXY boys to score relatively better on Social Awareness than other subscales and for XYY boys to score relatively better on Social Motivation than other subscales have been replicated in other studies (XXY: {Tartaglia, 2010 #39178} XYY: {Ross, 2015 #19436}). This pattern has suggested that there may be ‘characteristic profiles’ of social difficulties in children with SCTs. This is reminiscent of earlier work that suggested that Klinefelter’s Syndrome was associated with a particular personality style, characterised as shy, sensitive and passive {Herlihy, 2011 #18587}. More recently, Joseph et al {Joseph, 2018 #39159} found that elevated SRS scores were characteristic of males with XYY karyotype, especially those identified postnatally, but the specificity of the SRS was poor, with high scores associated with a range of psychiatric symptomatology.

Taken together, these studies indicate that an increased likelihood of a clinical-range SRS score is seen in children who have an extra X or Y chromosome, though the pattern of results suggests there may be a different profile depending on whether there is an additional X or Y chromosome. Studies of children who do not have trisomies indicate a need for caution in interpreting high SRS scores, as the specificity of this instrument is poor. In particular, children and young people with social anxiety may score within the clinically significant range of the SRS. Disentangling the relationship between ASD and social anxiety is not straightforward because anxiety-related social behaviours may look like autism, but have a different underlying basis {Tyson, 2012 #39118}. 

The goal of the current study was to evaluate profiles of social impairment and autistic features in children with an extra sex chromosome using information from an online diagnostic instrument (the DAWBA) and parental questionnaires, including the SRS.

Specific predictions were: 

1. In line with our previous study {Bishop, 2011 #17168}, we expected to find an increased rate of autistic features in all three trisomies, but with confirmed autism diagnoses elevated only in boys with XXY or XYY.  

2. Social anxiety symptoms were expected to be higher in children with an extra X chromosome (XXY and XXX) than in boys with an extra Y. 

3. We anticipated that ascertainment bias will affect results, and levels of social and psychiatric impairment will be  higher in children identified in the course of investigation for developmental disorders, than in other children with sex chromosome trisomies. 

4. Finally, extending the work by Joseph et al (2018), we expected that children with all three types of trisomy will have elevated scores on the SRS, but this will not necessarily be indicative of ASD.


# Methods

## Participants 
 
The full sample included `r n.K+n.L+n.M+n.N+n.O+n.P` children (`r n.K+n.N` with XXX, `r n.L+n.O` with XXY and `r n.M+n.P` with XYY) whose parents had completed the online Development and Wellbeing Assessment (DAWBA, see below). As shown in Figure 1, these came from a larger sample of `r n.C+n.D` children aged from 5 to 16 years who participated in a study of language and laterality in sex chromosome trisomies; genetic, laterality and language characteristics of these children have been reported in previous papers {Bishop, 2018 #39160; Newbury, 2018 #39109; Wilson, 2018 #38731}. Figure 1 shows recruitment sources as via National Health Service Clinical Genetics centres, or Other (via two support groups: Unique: the Rare Chromosome Support Group, and the Klinefelter Syndrome Association, or from self-referral via social media). A criterion for inclusion was that the child was aware of their trisomy status. Figure 1 distinguishes between cases where the trisomy was discovered prenatally or in the course of medical investigations unrelated to neuropsychiatric problems (e.g., problems affecting hormonal or muscular-skeletal systems), and those discovered in childhood during investigations for behavioural or neurodevelopmental problems. The latter group, referred to as the 'High Bias' subgroup, are likely to have a high rate of disorder that is not  representative of that karyotype. Accordingly, most of the analysis reported here excludes these cases and focuses on the 'Low Bias' subgroup. An unexpected feature of the sample was that the majority of girls with trisomy X were in the Low Bias subgroup, whereas the majority of boys with XXY or XYY were in the High Bias subgroup, suggesting that these karyotypes were more likely to be associated with neurodevelopmental problems. The Low Bias and High Bias subgroups were comparable in age: mean (SD) for Low Bias was `r round((mymeanage.bias[1,2]/12),1)` (`r round((mysdage.bias[1,2]/12),1)`) years, and for High Bias `r round((mymeanage.bias[2,2]/12),1)` (`r round((mysdage.bias[2,2]/12),1)`) years.

`r thisfig <- thisfig+1`

![_Figure `r thisfig`. Flowchart showing numbers of children with each karyotype in the study, in relation to whether prenatally or postnatally diagnosed, whether high ascertainment bias, and whether DAWBA was completed._](SCT_asd_flow.png)
\
One advantage of using DAWBA is that there are comprehensive epidemiological British data for boys and girls available using this instrument {Meltzer, 2000 #39161}. In order to compare rates of psychiatric problems with population prevalence figures from Meltzer et al  the sample was subdivided into a younger (age 5-10 years) and older (age 11-16 years) age group. Numbers in these two age bands were roughly equal for all karyotype groups:  Low Bias: XXX (`r agetab[1,1]` younger, `r agetab[1,2]` older); XXY (`r agetab[2,1]` younger, `r agetab[2,2]` older), XYY (`r agetab[3,1]` younger, `r agetab[3,2]` older); High Bias:   XXX (`r agetab[4,1]` younger, `r agetab[4,2]` older); XXY (`r agetab[5,1]` younger, `r agetab[5,2]` older), XYY (`r agetab[6,1]` younger, `r agetab[6,2]` older).

`r Norig` children had taken part in the previous study by Bishop et al (2011). Information about ASD in that study was available only from parental report: two of the children (8.7%) had been reported by their parents to have a diagnosis of ASD. No information on social anxiety had been collected in the 2011 study. 

In a previous report based on the current sample (Bishop et al, 2018), we noted that the likelihood of parents completing checklists for this study was lower if the child had significant language problems. This prompted us to compare children whose parents did and did not complete the DAWBA, using a vocabulary test score (see below) as an indicator of language level. A t-test revealed that, as before with checklist completion, parents who completed the DAWBA had children with milder language problems: mean vocabulary score for those who completed DAWBA was `r round(my.t.dawba$estimate[2],1)`(SD = `r sd.dawba1`) versus `r round(my.t.dawba$estimate[1],1)`(SD = `r sd.dawba0`) for the remainder, t (`r round(my.t.dawba$parameter,1)`) = `r round(my.t.dawba$statistic,1)`, 95% CI for difference in means = `r round(my.t.dawba$conf.int[1],1)` to `r round(my.t.dawba$conf.int[2],1)`.


## Material\

### Psychiatric evaluation
\
One or both parents were asked to complete the online Development and Wellbeing Assessment (DAWBA)  {Goodman, 2000 #18146} in their own time; `r (n.K+n.L+n.M+n.N+n.O+n.P-1)` families of children with SCT (one with two affected children) complied with this request.

The DAWBA includes a range of questions relating to DSM-IV and ICD-10 diagnostic criteria, risk factors, and impact on the family, which are supplemented by optional free text. 

The DAWBA software codes the items relevant to different diagnoses into bands that indicate the likelihood of the child meeting criteria for a range of psychiatric diagnoses {Goodman, 2011 #39163}, with the final diagnosis being made by a trained rater who assimilates all the quantitative and qualitative information. DAWBA has been used to obtain prevalence estimates for psychiatric disorders for a representative sample of over 10,000 children in Great Britain (Meltzer et al., 2000). Validity of DAWBA for diagnosis of autism has been shown to be acceptable {McEwen, 2016 #39164}.

All interviews with SCT cases were coded against DSM-IV criteria by the first author (AW); 30 interviews were recoded by a second independent rater (JK), with the remainder recoded by the last author (DB). The rater doing the coding was blind to the trisomy status of the child, although occasionally this was mentioned by parents in free text comments. Cases of disagreement were resolved by the last author (DB), who gave a final consensus rating that was used here. 

To address our questions of interest, we focused on DSM-IV diagnoses of Autism Spectrum Disorder (ASD) and Social Anxiety Disorder. ASD was coded as absent (0) or present (2), with a code of 1 given for significant autism symptoms that fell short of diagnostic criteria. Most of the latter group, which we refer to as the ASD- group, showed evidence of social and communicative difficulties, but did not have sufficient repetitive behaviours and restricted interests to merit an ASD diagnosis. There was agreement between raters on ASD coding for `r percagree1`% of cases, disagreement by one point (ratings of 0/1 or 1/2) in `r percagree2`%, and major disagreement (ratings of 0/2) in `r percagree3`%. There were `r comm_asd_agreetab[2,3]` children whose parents reported an existing diagnosis of ASD, and all of these were identified as definite ASD by DAWBA. These included two cases who had taken part in the study by Bishop et al (2011). Meltzer et al (2000) provided prevalence figures for the broader ASD category of PDDNOS, so we combined ASD and ASD- when comparing diagnoses with their sample.

The criteria for DSM-IV Social Phobia (now more commonly known as Social Anxiety Disorder) include a marked and persistent fear of one or more social situations where the individual may be subject to scrutiny by others. For a diagnosis of Social Phobia the symptoms must interfere significantly with everyday life. In children, the anxiety must occur in peer settings, not just with adults, must be disproportionate to the situation, and not be better explained by another disorder such as ASD. Social Phobia is differentiated from Separation Anxiety and General Anxiety Disorder by the situational specificity of the anxiety. Social Phobia was coded as present or absent, with `r percagrees1`%, agreement between raters.

## The Social Responsiveness Scale (SRS) 

The SRS is a 70-item scale designed to measure autism-related impairments as quantitative traits. It is composed of five subscales: social cognition, social awareness, social motivation, social communication and autistic features. Scores on each subscale, and on the total scale, were transformed to T-scores (based on the test manual) with mean 50 and SD of 10, with impairment represented as a positive score. 

## Vocabulary measure used to compare those with and without DAWBA scores  
The Vocabulary subtest from the Wechsler Abbreviated Scale of Intelligence {Wechsler, 1999 #12412} administered to children as part of the language test battery (Bishop et al, 2018) was used to check whether the families who completed DAWBA differed from others (see above). 

## Additional measures not included in this analysis

Children in this study were given additional language and cognitive assessments, which are described fully in the companion paper by Bishop et al (2018), and an assessment of cerebral and manual laterality, as documented by Wilson and Bishop (2018). DNA from saliva samples was analysed, as described by Newbury et al (2018). 

## Procedure  

Ethical approval was obtained for the study in 2011 from the Berkshire NHS Research Ethics Committee (reference 11/SC/0096), and data collection started in August of that year, finishing in October 2016. Families who had expressed interest in the study were interviewed by telephone to assess whether the child met inclusion criteria, and if so, a parent was invited to complete a consent form. Those who gave consent were also asked to view a video about the study with their child to ensure the family understood what was required before asking the child's assent to take part. For those who agreed, an appointment was made to see the child at home or at school, depending on parental preference. Families were widely dispersed around the UK, including Northern Ireland, Scotland, Wales and England. Parents were provided with passwords and a personal link to access the secure DAWBA website. They were asked to complete the DAWBA in their own time, and if they wished could spread the task over several days. The SRS was either completed by the parent while the child was doing language assessments, or was mailed to the parent and returned by post.

## Data analysis  
Study data were analysed using R software {R Core Team, 2016 #19256}, with the main database managed using REDCap electronic data capture tools hosted at the University of Oxford {Harris, 2009 #19562}.(Harris et al., 2009). The original sample size had been determined to give adequate power for genetic analyses reported by Newbury et al. (2018), rather than with phenotypic analyses in mind, and so we conducted additional analyses to consider the power for the current analyses on the reduced sample with DAWBA data.  

For comparing rates of disorder in the Low Bias group with population prevalence figures from Meltzer et al (2000), we used the log odds ratio. For this comparison, the smallest samples used here (N = 14) is sufficiently powered to detect an increase of prevalence of 20% or more, albeit with a large confidence interval. For instance, if the general population prevalence of disorder is 5%, and 3/14 of a trisomy group has disorder, then the log odds ratio is log((3/11)/(5/95)) = 1.64. The standard error of the log odds ratio is the square root of the sum of the reciprocals of the four frequencies {Bland, 2000 #17072}. Since the sample size for Meltzer et al is over 5000 for both boys and girls, then for our example, the standard error would be approximately sqrt(1/250+1/4750+1/3+1/11) = .65, so the 95% confidence interval for the log odds ratio covers `r 1.64-(.65*1.96)` to `r 1.64+(.65*1.96)`.  

For quantitative comparisons between the three Low Bias trisomy groups, we used nonparametric Kruskal Wallis tests.  A rough estimate of power can be obtained using the pwr.anova.test function in R {Champely, 2018 #39191}.  This gives a value of f = .25, corresponding to a medium effect size in {Cohen, 1988 #9165}'s terminology with 80% power and alpha of .05. However, actual power is likely to be lower than this because of the use of unequal group sizes and non-normal measures.  We made a further set of comparisons between the Low Bias and High Bias groups, with all karyotypes combined.  Using the pwr.t2n.test function for computing power of t-test with unequal sizes, gives 80% power to detect a Cohen's d of .61 with alpha of .05.
Thus for karyotype comparisons in the Low Bias trisomy cases, and between the combined Low Bias vs High Bias groups, our sample is powered to detect only relatively large effects. More subtle effects will require larger samples, which are unlikely to be feasible for a single research centre. This emphasises the need for raw data to be available so that future researchers will be able to combine information across studies.
 

# Results

Our research questions focused on the rates of ASD and Social Phobia in XXX, XXY and XYY karyotypes. Because we predicted that these rates would be influenced by ascertainment bias, the data were analysed separately for the Low Bias and High Bias subgroups. Table `r thistab+1` shows the numbers of children with DSM-IV diagnoses of ASD, ASD- or Social Phobia on the Development and Wellbeing Assessment (DAWBA).

**Table `r thistab+1`** 
_Numbers (percentages) by karyotype group for DAWBA diagnosis of ASD, ASD- and Social Phobia. (ASD- indicates with ASD symptoms but falling short of DSM-IV criteria)_
```{r makediagtable,echo=FALSE}
regulartable(nicetab) %>%
  theme_zebra() %>%
  autofit()
# kable(nicetab) %>%
#   kable_styling(bootstrap_options = c("striped", "hover"))
thistab<-thistab+1 #increment table counter
#nicetab has to have distinct column names - currently these are just dots.
```
  
\
Our first prediction was that there would be a high prevalence of ASD or ASD- specifically in boys with XXY or XYY. Meltzer et al (2000) reported prevalence rates for 'Pervasive Developmental Disorder' (encompassing ASD and ASD-) of 0.5% for boys and 0.1% for girls. Compared with these figures, there are elevated rates in all three trisomies. We consider here just the Low Bias groups: the log odds ratio (with 95% confidence interval) is `r logoddsr.xxx` (`r ci.low.xxx` - `r ci.hi.xxx`) for XXX girls; `r logoddsr.xxy` (`r ci.low.xxy` - `r ci.hi.xxy`) for XXY boys; and `r logoddsr.xyy` (`r ci.low.xyy` - `r ci.hi.xyy`) for XYY boys. One of these affected children, a girl with XXX, had been in the original 2011 study, but had not been identified with ASD at that time. Thus, these results are not confounded with those from the previous study. Our prediction of an increased prevalence of ASD was confirmed, but this extended to girls with XXX as well as the two groups of boys. The prevalence rates reported by Meltzer et al (2000) are lower than  subsequent estimates of ASD prevalence in the UK {Baird, 2006 #15032}, but we found that even if we doubled the prevalence rates used to compute odds ratios, the ASD/ASD- frequencies were still well above expected levels. Nevertheless, those with an ASD diagnosis were in the minority in all three trisomies. 

Our second prediction was that there would be a raised prevalence of Social Phobia in those with an extra X chromosome. As shown in Table `r (thistab)`, there were `r numnames[socanx.xxy1+1]` children with this diagnosis in the XXY-Low Bias group and `r numnames[socanx.xxy2+1]` in the XXY-High Bias group, as well as one child in the XYY-High Bias group. All but `r numnames[mytab[2,2]+1]` of these children had additional diagnoses of ASD- or ASD.  Although the numbers of boys with XXY studied here is very small, the finding that 21% in the Low Bias group and 28% in the High Bias group met diagnostic criteria for Social Phobia is well above the prevalence in the general population (log odds ratio for Low Bias group = `r round(logoddsr,2)`, 95% CI = `r ci.low`-`r ci.hi`). It may seem plausible that Social Phobia would be a condition that develops with age, as children's ability to reflect on the reactions of others develop and social demands on them increase. However, the effect of age was small in the epidemiological sample of Meltzer et al. (2000), with prevalence of Social Phobia at 0.4% in young boys, 0.2% in young girls,  0.3% in older boys and 0.4% in older girls. There was no obvious association with age in the trisomy sample: the three XXY boys with Social Phobia in the Low Risk group were aged 6, 10 and 14 years.  

DAWBA categorical diagnoses only detect more severe levels of impairment associated with burden to families. We therefore also examined symptoms reflected in the DAWBA banding for Social Phobia (Goodman et al, 2011), an ordinal scale computed from the profile of social anxiety symptoms. Table `r thistab+1` shows the distribution of DAWBA band scores. In the B-CAMHS sample, 96% of cases fell in band 0 or 1. It is evident by inspection that subclinical levels of symptoms are relatively high in all three types of trisomy. For the Low Bias group, the log odds ratios (95% CI) for scoring in band 2 or above, relative to the B-CAMHS sample is `r sophbandOR[2,1]` (`r sophbandOR[3,1]` - `r sophbandOR[4,1]`) for XXX girls, `r sophbandOR[2,2]` (`r sophbandOR[3,2]` - `r sophbandOR[4,2]`) for XXY boys, and `r sophbandOR[2,3]` (`r sophbandOR[3,3]` - `r sophbandOR[4,3]`) for XYY boys. Although the trend is for a higher rate of problems in those with an extra X chromosome, the difference in rates of scores of band 2 or above by karyotype was not statistically reliable; Kruskal-Wallis chi-squared = `r round(kband$statistic,2)`, df = `r kband$parameter`, p = `r round(kband$p.value,2)`.

**Table `r thistab+1`** 
_Numbers (percentages) by karyotype group in DAWBA bands for Social Phobia, in relation to data from follow-up of the British Child and Adolescent Mental Health Surveys (B-CAMHS; Goodman et al, 2011)_
```{r diagtable,echo=FALSE}
regulartable(nicebandtab) %>%
  theme_zebra() %>%
  autofit()
# kable(nicetab) %>%
#   kable_styling(bootstrap_options = c("striped", "hover"))
thistab<-thistab+1 #increment table counter

```


`r thisfig <- thisfig+1`Figure `r thisfig` shows the T-scores on the SRS. SRS T-scores are scaled with mean of 50 and SD of 10, where a high score indicates impairment. The figure shows individual data points as well as means on the SRS scales, with children meeting diagnostic criteria for ASD and/or Social Phobia colour-coded.

![**Figure `r thisfig`**. _Beeswarm plots: SRS scales and total T-score for the three trisomy groups, subdivided by bias group. Symbol depicts age band (circle 5-10 yr, triangle 11-16 yr. Colour depicts diagnosis: grey = none; light blue = Social Phobia; dark blue = Social Phobia+ASD-; pink = ASD-; red = ASD; purple = Social Phobia+ASD)_](beeswarm_SRS_T_2018.png)

`r srsfig <- thisfig` It is evident from Figure `r (thisfig)` that, while the overall rate of impairment is raised relative to the general population, many children with sex chromosome trisomies score within normal limits.  We had planned to use Multivariate Analysis of Variance (MANOVA) to compare groups on the SDQ and SRS, but assumptions of multivariate normality were not met, and so nonparametric Kruskal-Wallis tests were used instead. A randomisation test with 10,000 iterations was used to estimate the range of chi square values that would be obtained  under the null hypothesis, where group identity was assigned at random. A series of analyses was conducted to address the specific questions raised in the Introduction.

**Test of hypotheses 1-2** (Low Bias group only). The first two hypotheses were that there would be chromosome-specific differences in profiles, with autistic features more prominent in boys with an extra Y chromosome, and other types of social impairment in boys and girls with an additional X chromosome. To reduce the effects of ascertainment bias, these predictions were tested using data from the Low Bias groups only. Scores for the three karyotypes were compared on the five SRS scales. `r thisfig<-thisfig+1` Results are summarised in Figure `r thisfig`, which shows the obtained chi square values in relation to those from the randomised data, with the upper fin of each boxplot corresponding to the 99th centile. Given the small sample sizes, and wide variation within each karyotype, the power is low to detect subtle effects. 

![**Figure `r thisfig`**. *Plot of chi square values obtained on Kruskal-Wallis test comparing the three Low Bias sex chromosome trisomy groups (red asterisks), relative to the range of chi square values when group is randomised. The box shows the interquartile range, and the upper fin of the boxplot corresponds to the 99th centile for randomised values.*](random99_chiplotSRS_LowBias_karyotype.png)




**Test of hypothesis 3**. (Low Bias vs High Bias trisomy groups) 
`r thisfig<-thisfig+1` Hypothesis 3 predicts that children in the High Bias group will have more evidence of impairment than those in the Low Bias group. Again, multiple Kruskal-Wallis tests were used to test this prediction, this time with the three karyotypes combined. Results are shown in Figure `r thisfig`. This analysis confirms the substantial differences between the Low Bias and High Bias groups, who differ markedly on all the scales.

![**Figure `r thisfig`**. *Plot of chi square values obtained on Kruskal-Wallis test comparing Low Bias vs High Bias sex chromosome trisomy cases (all trisomies combined), relative to the range of chi square values when group is randomised. The upper fin of the boxplot corresponds to the 99th centile for randomised values.*](random99_chiplotSRS_low_vs_hibias.png) 


**Test of hypothesis 4**. `r thisfig<-thisfig+1`
Hypothesis 4 proposed that elevated scores on the SRS would be seen in children with sex chromosome trisomies, but would not necessarily be indicative of ASD. As shown in Figure `r srsfig`, the odds of having a score outside the normal range on the SRS scales is elevated in all three trisomies, especially for children from the High Bias Group.  The last panel of Figure `r srsfig` shows T-scores on the SRS total for children with sex chromosome trisomies in relation to DAWBA diagnoses of ASD, ASD- and Social Phobia. These data show that all but one child with a diagnosis of ASD or ASD- score above the 'Severe' cutoff of 70, but most children in the 'Mild' range of 60-70 do not have a diagnosis of either ASD or ASD-. 

A cutoff total score of 75 has been reported as giving sensitivity and specificity of 0.85 and 0.75 for ASD diagnosis in a population of mixed cases of ASD and other developmental disorders {Bölte, 2011 #19563}. In the current sample, if we group together the ASD and ASD- cases, then with this cutoff, we obtain `r srscuttab[1,1]` true negatives (non-autistic < 75), `r srscuttab[2,1]` false positives (non-autistic > 74), `r srscuttab[1,2]+srscuttab[1,3]` false negatives (autistic < 75), and `r srscuttab[2,2]+srscuttab[2,3]` true positive (autistic > 74), which gives sensitivity of `r round((srscuttab[2,2]+srscuttab[2,3])/(srscuttab[1,2]+srscuttab[1,3] +srscuttab[2,2]+srscuttab[2,3]),2)` and specificity of `r round(srscuttab[1,1]/(srscuttab[1,1]+srscuttab[2,1] ),2)`. If, however, we compute predictive power for diagnosis of ASD, and treat above-cutoff ASD- cases as false positives, sensitivity remains high at `r round(srscuttab[2,3]/sum(srscuttab[1:2,3]),2)` but specificity declines to `r round(sum(srscuttab[1,1:2])/sum(srscuttab[1:2,1:2]),2)`. In general, our results are consistent with those of Bölte et al. (2011), in showing that a cutoff of 75 is effective at identifying children with autistic features, though many of those scoring in this range would not meet full DSMIV criteria, but rather correspond to cases of ASD-. 


# Discussion



Before discussing the specific results found here, it is important to consider potential biases in the sample of children with sex chromosome trisomies. We distinguished a High Bias group who were identified during investigation for neurodevelopmental or behavioural problems; these cases may be useful in revealing genetic or other factors that may be associated with variable outcomes, but they will inevitably over-estimate prevalence of problems. Our Low Bias group is more appropriate for establishing risk of various outcomes, but it is important to note that there still will be some bias in those identified on prenatal screening: mothers will be older than average, and will have elected to continue with the pregnancy. Because the prenatally-diagnosed cases were identified via National Health Services Clinical Genetics departments, they are less likely than those from US samples to be biased to more affluent families. However, another bias that is impossible to control is determined by who volunteers to take part in the research. In this study, children could be included only if they knew about their chromosome status; our prior research has shown that parents are more likely to disclose this information when the child is experiencing difficulties {Gratton, 2016 #19429}. Children were also required to assent to take part in the study, after viewing a video that provided information about what was involved. It is plausible that this would deter children with significant social anxiety. In addition, only 89 of 142 (62%) of participating families completed the DAWBA, and our analysis showed that the child's language level was a factor affecting this, with parents of more mildly affected children being more likely to complete the interview. Factors such as these will inevitably affect who takes part in any study of children with genetic conditions - even the original newborn screening studies - as one cannot compel people to participate. The best we can do in this situation is to be transparent about sources of bias, and consider how far results are consistent across different studies where specific influences on who volunteers may assume different importance. Bearing this in mind, we turn to consider the specific results obtained here, which lead to six broad conclusions.  

## 1. Sex chromosome trisomy as risk factor for ASD ##
The current results add to those of previous research {Van Rijn, 2008 #16706; Van Rijn, 2014 #19565; Cordeiro, 2012 #18895; Ross, 2012 #18868; Joseph, 2018 #39159} in indicating that the rate of ASD is increased in children with an extra sex chromosome. On the basis of previous results by Bishop et al (2011), we predicted that elevated rates of ASD would be found only in boys with XXY and XYY. However, our results were consistent with the more recent studies of Van Rijn et al (2014) and Wigby et al (2016), who found increased levels of ASD in girls with trisomy X. In the current sample, two of 25 girls with XXX in the Low Bias group, as well as one of four girls in the High Bias group met DSM-IV criteria for ASD.  Furthermore, if we adopt the broader definition of ASD-, which includes cases with significant autism symptoms but falling short of ASD (usually because of insufficient evidence of repetitive behaviours) then we can compare rates with the gender-specific population norms for PDDNOS of Meltzer et al (2011). Although those with ASD/ASD- were a minority of children, and the sample was too small to give a confident estimate of prevalence, these results show that the increased prevalence of ASD applies to all three trisomies. Note that a minority of children in the current study had taken part in the earlier study by Bishop et al (2011), and all of those with ASD in the current sample were new cases. 

## 2. Sex chromosome trisomy as risk factor for social anxiety disorder ##
Very few children in this study met diagnostic criteria for DSM-IV Social Phobia: all but one of those who did were cases of XXY. Only two children (both from the Low Risk XXY group, N = 14) had Social Phobia in the absence of any autistic features. While this represents a substantial increase over expected values (relative to Meltzer et al's epidemiological prevalence of 0.4% in boys), numbers are too small for reliable estimates. When we considered milder symptoms of social anxiety, using DAWBA bands, a rather different picture emerged, with 11/25 (44%) of XXX girls, 6/14 (43%) of XXY boys and 3/13 (23%) of XYY boys being placed in band 2 or over, compared with a prevalence of 3.4% in the B-CAMHS epidemiological sample. Thus all three types of sex chromosome trisomy confer a risk of increased social anxiety, though this generally falls short of meeting diagnostic criteria for Social Phobia. As noted above, however, a child with severe Social Phobia is unlikely to have agreed to participate in the study, as this involved an individual assessment with an unfamiliar adult.

## 3. Many children show no signs of either ASD or Social anxiety  

We used parental responses to the SRS to capture mild, subclinical aspects of psychiatric problems, and it did reveal elevated levels of impairment, even when children with diagnoses of ASD/ASD- or Social Phobia were excluded. However, the SRS  also confirmed the wide range of outcomes in all three trisomies, with around half the children in the Low Bias group scoring within the normal range. 

## 4. The effects of ascertainment bias are substantial  
As anticipated, children from the High Bias group had more severe problems than those from the Low Bias group. Although most studies that have looked at this question have found similar patterns, this is not always the case (cf. {Van Rijn, 2014 #19565}). The method for dividing cases into Low Bias and High Bias subgroups may affect results: if the latter group is defined purely in terms of late diagnosis, then it may include some whose trisomy was discovered in the course of medical investigations, rather than, as in our sample, being restricted to those where behavioural or cognitive impairments had prompted genetic investigation.

## 5. More similarities than differences in the impact of an extra X or extra Y chromosome
The notion that supplementary X and Y chromosomes might have different effects is attractive as well as plausible {Green, 2019 #39167; Skuse, 2018 #39029}, particularly since it has potential to throw light on sex differences in psychiatric disorders in general. However, data from the current study indicate that insofar as such effects occur, they can be swamped by other influences, leading to a highly heterogeneous picture. 

We should not dismiss the idea that there could be complementary effects (more social anxiety with an extra X and more ASD with an extra Y) on the basis of data from a single study. Our sample sizes were small, especially when attention was restricted to the Low Bias group, so we lacked statistical power to detect any but large associations. Insofar as there were trends in the data, they agreed with the prediction of higher levels of social anxiety in boys and girls with an extra X chromosome, and higher levels of ASD and reduced prosocial behaviour in boys with an extra Y.  But we can conclude that, insofar as such effects occur, risks for specific disorders are probabilistic rather than deterministic. A simple model that treats social anxiety and lack of social awareness as points at either end of a single dimension is also complicated by the finding of cases who met diagnostic criteria for both Social Phobia and ASD/ASD-. 

## 6. Variability in outcomes suggests that an extra sex chromosome creates general developmental instability

The more we discover about genetic causes of neurodevelopmental disorders, the more we are confronted with the fact that the same aetiology can lead to very variable outcomes. This is true for a wide range of genetic conditions that involve copy-number variants or point mutations, as well as chromosome aneuploidy {Mitchell, 2015 #39173}. Conditions such as tuberous sclerosis, 22q11.2 deletion syndrome, 15q11.2 microdeletion syndrome are well-established as risk factors for neurodevelopmental disorders, but the clinical manifestations can vary widely from person to person, even within a family. The phenotypic outcomes of these conditions do not respect traditional diagnostic boundaries: the same genetic aetiology can lead to ASD, schizophrenia, epilepsy or ADHD. It is often assumed that this variability is due either to interactions with the genetic background or to environmental factors that moderate gene effects. This kind of explanation cannot, however, explain why in twin studies, one can often see marked phenotypic variability in concordant MZ twins: this has been shown by LeCouteur et al {Le Couteur, 1996 #12114} for ASD and by Corey et al {Corey, 2011 #39172} for epilepsy. Given that MZ twins have identical genes and closely similar environments, we need another explanation for phenotypic variability. One candidate is stochastic variation {Molenaar, 1993 #39171; Wolf, 1997 #10890}. Purely random events early in neurodevelopment can have cascading effects so that quite minor differences in the prenatal brain lead to long-lasting differences in function. This alone, however, does not explain why there should be higher rates of impairment in those with chromosome trisomies. To account for that, we need to introduce the idea of developmental instability, i.e., a disturbance of the usual homeostatic processes that make the organisation robust to external influences {Nijhout, 2003 #16303}. This would mean that small stochastic effects would have larger impact than would otherwise be the case.  

Evidence that aneuploidy may be linked to developmental instability was obtained by {Beach, 2017 #19484}, who found high variability in cell cycle progression among yeast cells with identical aneuploidies. Furthermore, these cells showed variable response to environmental stress. These authors went on to demonstrate wide phenotypic variation in inbred mice with genetically engineered trisomy 19, despite a uniform genetic background and environment. 
Mitchell {Mitchell, 2018 #39169} proposed that a wide range of genetically-based neurodevelopmental disorders may involve a process of developmental instability. This can make sense of the wide range of outcomes seen with an extra sex chromosome - both in terms of the severity of impairments, and in terms of the wide range in symptoms that are observed. If the main effect of an additional chromosome is to increase neurodevelopmental instability, then the developing brain will be less robust to perturbations, so that small differences in early neurodevelopmental processes will have cascading effects that can lead to very different clinical pictures. 

## Clinical implications  

When a sex chromosome trisomy is identified prenatally, it is important that parents are given accurate information about the implications for the child's development; the quality of information provided is one factor that influences whether the mother continues with the pregnancy {Jeon, 2012 #39166}. It is crucial that parents are provided with information from unbiased samples: The recommendations of Linden, Bender and Robinson {Linden, 2002 #18611} are generally still valid: as they noted, children with sex chromosome trisomies 'are at an increased risk for developmental problems but.. most are in the normal range of development, and marked abnormality is not usually seen' (p. 4). However, at the time they were writing, there was no awareness of the risk of ASD, possibly because the diagnostic criteria for autism were far more stringent at the time when the neonatal screening studies were conducted {Bishop, 2011 #17168}. The companion paper on this sample confirms the recommendations that parents should be informed that presence of a sex chromosome trisomy raises the probability that the child will have educational difficulties {Bishop, 2018 #39160}, but the current analysis indicates that in addition, although the majority of children will not merit a clinical diagnosis for ASD and/or Social Phobia, the likelihood of these conditions is increased.

Advice on management of children with a sex chromosome trisomy has mostly focused on the provision of speech and language therapy and educational interventions to address problems with speech, language and educational progress that may arise. The current study emphasises that symptoms of social anxiety are common in all three trisomies, suggesting that early involvement of Child and Adolescent Mental Health Services may be warranted in some cases. The outcomes of children with sex chromosome trisomies are very varied, making it difficult to predict in advance the exact type of support that will be needed, but it is clear that problems often do not fit neatly into single diagnostic categories, and optimal support may require collaboration between different services.

\newpage
# Acknowledgements
We offer warmest thanks to the families who took part in the study, and school staff who helped facilitate assessment arrangements. The study would not have been possible without the hard work and dedication of a series of research assistants who conducted the assessments, often travelling all over the UK to do so: Eleanor Payne, Nikki Gratton, Georgina Holt, Annie Brookman, Elaine Gray, Louise Atkins, Holly Thornton and Sarah Morris. We also thank Paul A. Thompson for expert advice on statistical analysis. This work was funded by Wellcome Trust Programme Grants no 082498/Z/07/Z and 082498/Z/07/C.
\newpage

# References

# Data availability statement
Data and scripts relating to this project, as well as a STROBE checklist, are available on Open Science Framework at this address: https://osf.io/u2c7d/ (DOI 10.17605/OSF.IO/U2C7D). Data are available under the terms of the Creative Commons Zero "No rights reserved" data waiver (CC0 1.0 Public domain dedication).

# Competing interests
No competing interests were disclosed. 

# Grant information
This work was funded by Wellcome Trust Programme Grant 082498.
The funders had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript. 


\setlength{\parindent}{-0.5in}
\setlength{\leftskip}{0.5in}


```{r sessinfo, include=FALSE}
sessionInfo()
```